---
title: "BRCA1_SGE2_global_20171119.Rmd"
author: "Greg Findlay"
date: "11/19/2017"
output: html_document
---

Updated global analysis scripts to apply filters and get final functional scores with updated pipeline for both SGE1 and SGE2, includes mixture modeling to define thresholds and filter off concordance, 
(previously have run the XrL4_all_ggplot_removed scripts and the positional modeling v6 scripts)
```{r}
#MUST RUN POSITIONAL CORRECTION MARKDOWN SCRIPTS
#i.e. positional_modeling_SGE2_v6_20171012.Rmd
# tHDR_pos_merge_df are built with XyyrL41_prog_map_thresh5_df_rL42_int_ord_df -- prog_map means:                all_hdr_variant == TRUE, !=WT ,POS!= REF, and > threshold of 0.00001
#add a position from PAM based on exon...
# merge df's built off the prog_map_thresh_5 data sets.
#previous change in code in sam_to_edits script dictated that 'hdr5' is always the cut-proximal site, and 'hdr3' is always the cut-distal site, so here even though X2, X5 and X15 are cut at the 3' PAM site, 'hdr5' is still used to indicate we want the data at the cut site.

require(ggplot2)
require(gridExtra)
require(mclust)
require(VennDiagram)
library(gam)
library(mixtools)
library(plotROC)
named_conseq_colors_old = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#0B3DB9", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#3885DB", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#9D0012","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "REF" = "#000000")
named_conseq_colors_full = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#0B3DB9", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#CF7700", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#9D0012","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "REF" = "#000000", "absent" = "#BFBCBC","Benign" = "#013DD3","Conflicting interpretations of pathogenicity" = "#FFCB00","Likely benign" = "#2360F9","Likely pathogenic" = "#FA2039","Pathogenic" = "#A40000","Uncertain significance" = "#00A54D" )
named_conseq_colors_full_simple = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#9D0012","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "REF" = "#000000", "absent" = "#BFBCBC","Benign" = "#013DD3","Conflicting interpretations of pathogenicity" = "#FFC927","Likely benign" = "#2360F9","Likely pathogenic" = "#FA2039","Pathogenic" = "#A40000","Uncertain significance" = "#31947B","Gray" = "#000000" )
named_conseq_colors = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#6793F8", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#002372", "SYNONYMOUS" = "#0C4EE5","STOP_LOST" = "#9D0012","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "REF" = "#000000")

X2rL4_tHDR_pos_merge_df$hdr_cut <- X2rL4_tHDR_pos_merge_df$hdr5
X3rL4_tHDR_pos_merge_df$hdr_cut <- X3rL4_tHDR_pos_merge_df$hdr5
X4rL4_tHDR_pos_merge_df$hdr_cut <- X4rL4_tHDR_pos_merge_df$hdr5
X5rL4_tHDR_pos_merge_df$hdr_cut <- X5rL4_tHDR_pos_merge_df$hdr5
X15rL4_tHDR_pos_merge_df$hdr_cut <- X15rL4_tHDR_pos_merge_df$hdr5
X16rL4_tHDR_pos_merge_df$hdr_cut <- X16rL4_tHDR_pos_merge_df$hdr5
X17rL4_tHDR_pos_merge_df$hdr_cut <- X17rL4_tHDR_pos_merge_df$hdr5
X18rL4_tHDR_pos_merge_df$hdr_cut <- X18rL4_tHDR_pos_merge_df$hdr5
X19rL4_tHDR_pos_merge_df$hdr_cut <- X19rL4_tHDR_pos_merge_df$hdr5
X20rL4_tHDR_pos_merge_df$hdr_cut <- X20rL4_tHDR_pos_merge_df$hdr5
X21rL4_tHDR_pos_merge_df$hdr_cut <- X21rL4_tHDR_pos_merge_df$hdr5
X22rL4_tHDR_pos_merge_df$hdr_cut <- X22rL4_tHDR_pos_merge_df$hdr5
X23rL4_tHDR_pos_merge_df$hdr_cut <- X23rL4_tHDR_pos_merge_df$hdr5
#calling these again in case old script re-assigned without the 0-indexing adjustment
X2_cut_pos <- 41276033.5+1
X3_cut_pos <- 41267763.5+1
X4_cut_pos <- 41258518.5+1 
X5_cut_pos <- 41256892.5+1
X15_cut_pos <- 41222951.5+1
X16_cut_pos <- 41219676.5+1
X17_cut_pos <- 41215950.5+1
X18_cut_pos <- 41215377.5+1
X19_cut_pos <- 41209129.5+1
X20_cut_pos <- 41203127.5+1
X21_cut_pos <- 41201207.5+1
X22_cut_pos <- 41199706.5+1
X23_cut_pos <- 41197794.5+1

#binding all of the merge df's together...
XrL4_all_tHDR_pos_merge_master_df <- rbind(X2rL4_tHDR_pos_merge_df,X3rL4_tHDR_pos_merge_df,X4rL4_tHDR_pos_merge_df,X5rL4_tHDR_pos_merge_df,X15rL4_tHDR_pos_merge_df,X16rL4_tHDR_pos_merge_df,X17rL4_tHDR_pos_merge_df,X18rL4_tHDR_pos_merge_df,X19rL4_tHDR_pos_merge_df,X20rL4_tHDR_pos_merge_df,X21rL4_tHDR_pos_merge_df,X22rL4_tHDR_pos_merge_df,X23rL4_tHDR_pos_merge_df)

XrL4_all_tHDR_pos_merge_master_df$pos_alt <- paste(XrL4_all_tHDR_pos_merge_master_df$pos, XrL4_all_tHDR_pos_merge_master_df$alt, sep='')
#add a clinvar transcript annotation here
#add a clinvar mutation string i.e. c.
#add some sort of intronic indication (i.e. c. 5017 -3 C -> A)

write.table(XrL4_all_tHDR_pos_merge_master_df, "/mount/SGE/BRCA1/XrL4_all_tHDR_pos_merge_v6_df_20171015.txt", sep="\t")
XrL4_all_tHDR_pos_merge_master_df <- read.table("/mount/SGE/BRCA1/XrL4_all_tHDR_pos_merge_v6_df_20171015.txt", sep="\t", header=TRUE)

#import this table back in as a data frame now that all analysis is final... re-run from here.
#can everything be re-run from this df?

### Detect outliers in data set by comparing post-pre selection and pre-lib selection
#accounting for position again, but emphasizing outliers
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),y=rL41rL42_tHDR_pre_lib_loess_rL41rL42, color=conseq,size = rL41rL42_tHDR_post_lib_loess_rL41rL42))+geom_point(alpha=0.6)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Pre/lib loess2 corr. log-2')+xlab('Post/pre log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_size_continuous(range = c(0.1, 2))+geom_text(aes(label=paste(exon,pos),alpha=abs(log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm)-rL41rL42_tHDR_pre_lib_loess_rL41rL42)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black')
#same plot as above but fitting curves by conseq, as well as total fit
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),y=rL41rL42_tHDR_pre_lib_loess_rL41rL42, color=conseq,size = rL41rL42_tHDR_post_lib_loess_rL41rL42))+geom_point(alpha=0.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('Pre/lib loess2 corr. log-2')+xlab('Post/pre log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_size_continuous(range = c(0.1, 2))+geom_smooth(aes(color = XrL4_all_tHDR_pos_merge_master_df$conseq), method = 'lm',se=FALSE)+geom_smooth(aes(color='REF'),method = 'lm')


#does post/lib give a different result than pre/lib+post/pre?
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm) + rL41rL42_tHDR_pre_lib_loess_rL41rL42, y=rL41rL42_tHDR_post_lib_loess_rL41rL42,color=conseq))+geom_point(alpha=0.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Post/lib corr. log-2')+xlab('Post/pre + pre/lib corr. log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+scale_size_continuous(range = c(0.1, 2))

#does post/lib give a different result than pre/lib+post/pre (labelled with exon)
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm) + rL41rL42_tHDR_pre_lib_loess_rL41rL42, y=rL41rL42_tHDR_post_lib_loess_rL41rL42,color=conseq))+geom_point(alpha=0.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Post/lib corr. log-2')+xlab('Post/pre + pre/lib corr. log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_size_continuous(range = c(0.1, 2))+geom_text(aes(label=paste(exon,paste(pos,alt,sep=''))),hjust=0.5, vjust=0.5, size = 1.4,colour='Black')

#and with a density contour...
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),y=rL41rL42_tHDR_pre_lib_loess_rL41rL42))+geom_point(aes(color=conseq),alpha=0.3)+stat_density2d(aes(alpha=..level..), geom="polygon") +scale_alpha_continuous(limits=c(0.2,5), breaks=seq(0,0.05,by=0.002)) + theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('Pre/lib loess2 corr. log-2')+xlab('Post/pre log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#ID outliers with high pre/lib:
XrL4_all_tHDR_pos_merge_master_df[which(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df$rL41rL42_tHDR_pre_lib_loess_rL41rL42)) > 1.2),c('exon','pos','conseq','rL41rL42_tHDR_pre_lib_loess_rL41rL42','tHDR_post_pre_ratio_rL41rL42_mean_synnorm')]
#ID outliers with lost pre/lib
XrL4_all_tHDR_pos_merge_master_df[which(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df$rL41rL42_tHDR_pre_lib_loess_rL41rL42)) < -2 ),c('exon','pos','conseq','rL41rL42_tHDR_pre_lib_loess_rL41rL42','tHDR_post_pre_ratio_rL41rL42_mean_synnorm')]

#global plot vs. error:
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=corr_pre_pseudo/pre_pseudo,y=pre_pseudo_freq, color=conseq,size=corr_pre))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('pre freq.2')+xlab('fraction reads est. from error')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#do any variants drop out completely? lib to pre
#with size being correction rate:
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(y=log10((pre_pseudo_freq+rL42_pre_pseudo_freq)/2),x=log10(lib_freq), color=conseq,size=(1-(corr_pre_pseudo/pre_pseudo+rL42_corr_pre_pseudo/rL42_pre_pseudo)/2)))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('lib freq. log-10')+ylab('mean. pre freq. (cSNV) log-10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_text(aes(label=paste(exon,paste(pos,alt)),alpha=abs(log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm)-rL41rL42_tHDR_pre_lib_loess_rL41rL42)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black')

#tHDR reads
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(y=log10((tHDR_pre_pseudo_freq+rL42_tHDR_pre_pseudo_freq)/2),x=log10(lib_freq), color=conseq,size=rL41rL42_tHDR_post_lib_loess_rL41rL42))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('lib_freq')+ylab('tHDR_pre_lib_ratio_mean')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_text(aes(label=paste(exon,paste(pos,alt)),alpha=abs(log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm)-rL41rL42_tHDR_pre_lib_loess_rL41rL42)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black')

#do any variants drop out completely? pre to post
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(y=log10((post_pseudo_freq+rL42_post_pseudo_freq)/2),x=log10((pre_pseudo_freq+rL42_pre_pseudo_freq)/2), color=conseq,size=rL41rL42_tHDR_post_lib_loess_rL41rL42))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('mean cSNV pre freq. log-10')+ylab('mean cSNV post freq. log-10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_text(aes(label=paste(exon,paste(pos,alt)),alpha=abs(log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm)-rL41rL42_tHDR_pre_lib_loess_rL41rL42)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black')



#variant filtering:
#rL41 frequency already must be >= .00001
length(XrL4_all_tHDR_pos_merge_master_df$tHDR_lib_pseudo_freq)
length(which(XrL4_all_tHDR_pos_merge_master_df$tHDR_lib_pseudo_freq > 10^-4))
length(which(XrL4_all_tHDR_pos_merge_master_df$rL42_pre_freq >= 0.00001))

#NOTE -- DEMAND that only the one matching the exon's cut end is yes!! This will solve all NAG / NTG issues...
length(which(XrL4_all_tHDR_pos_merge_master_df$c_hdr_snv == 'True'))
length(which(XrL4_all_tHDR_pos_merge_master_df$hdr3 =='yes')) #distal edits
length(which(XrL4_all_tHDR_pos_merge_master_df$hdr5 =='yes')) #proximal edits
#want this one!!
length(which(XrL4_all_tHDR_pos_merge_master_df$hdr_cut =='yes' & XrL4_all_tHDR_pos_merge_master_df$tHDR_lib_pseudo_freq > 10^-4 ))

XrL4_all_filter_lib_hdr_cut_df <- XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$hdr_cut =='yes'),]

#high sequencing error contribution -- average of two replicates must be > 0.5 in cSNV context
length(which((XrL4_all_tHDR_pos_merge_master_df$corr_pre_pseudo/XrL4_all_tHDR_pos_merge_master_df$pre_pseudo + XrL4_all_tHDR_pos_merge_master_df$rL42_corr_pre_pseudo/XrL4_all_tHDR_pos_merge_master_df$rL42_pre_pseudo)/2 <0.5))

length(which((XrL4_all_filter_lib_hdr_cut_df$corr_pre_pseudo/XrL4_all_filter_lib_hdr_cut_df$pre_pseudo + XrL4_all_filter_lib_hdr_cut_df$rL42_corr_pre_pseudo/XrL4_all_filter_lib_hdr_cut_df$rL42_pre_pseudo)/2 <0.5))

#which points are poorly represented in cSNV context compared to potential seq error? not worth using!
XrL4_all_filter_lib_hdr_cut_error_df <- XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$tHDR_lib_pseudo_freq > 10^-4 & XrL4_all_tHDR_pos_merge_master_df$rL42_pre_freq >= 0.00001 & XrL4_all_tHDR_pos_merge_master_df$hdr_cut =='yes' & (XrL4_all_tHDR_pos_merge_master_df$corr_pre_pseudo/XrL4_all_tHDR_pos_merge_master_df$pre_pseudo + XrL4_all_tHDR_pos_merge_master_df$rL42_corr_pre_pseudo/XrL4_all_tHDR_pos_merge_master_df$rL42_pre_pseudo)/2 <0.5),]

#KEEP THIS!!!
XrL4_all_filter_lib_hdr_cut_df <- XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$tHDR_lib_pseudo_freq > 10^-4 & XrL4_all_tHDR_pos_merge_master_df$rL42_pre_freq >= 0.00001 & XrL4_all_tHDR_pos_merge_master_df$hdr_cut =='yes'),]

#created a new column pasting the pos and alt rows together above -- all of these were manually identified as outliers in the pre/lib loess.2-adjsuted tHDR mean values.
#list of NCGG variants at cut PAMs by pos_alt:
exclude_NCGG_active <- c('41256887C','41222959G','41209124C','41199701C','41197789C')
#list of NCGG PAMs 
exclude_NCGG_all <- c('41256887C','41222959G','41209124C','41199701C','41197789C','41267771G','41215385G','41203121C','41201202C')
#list of NXG variants at cut PAMs (already exlcuded by requirement of prox pam present):
exclude_NXG_all <- c('41276039A','41267769A','41267769T','41222957A','41222957T','41215383A','41215383T','41209126T','41209126A','41203123T','41203123A','41201204A','41201204T','41199703T','41199703A','41197791T','41197791A')
#list of X5 NAG variants at cut PAM and PAM+1 variants (poor fit):
exclude_X5_PAM <- c('41256890G','41256890C','41256890T','41256887C','41256887G','41256887A')
#list of other variants near cut site depleted heavily, including X2 pam, and X16 ps var's.
exclude_other_ps_pam <- c('41276038A','41219672G')
#list of variants that may have splice difference in context of PAM edit:
#ones with RNA evidence of depletion in edited PAM context (must go)
#these are filtered out later with SNVs not in CADD (i.e. not true SNVs)
#exclude_pam_splice_low_rna <- c('41256962A','41199702G')
#exclude_pam_splice_no_test_GGT_donor <- c('41258513A','41215945A','41203122A','41203067A','41201127A')
#exclude_pam_splice_no_test_HGT_donor <- c('41219675C','41209125A','41199702A','41197790A','41197769A')
#fall substantially above:
length(which(XrL4_all_tHDR_pos_merge_master_df$rL41rL42_tHDR_pre_lib_loess_rL41rL42 > 1))
exclude_high_pre_lib <- XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$rL41rL42_tHDR_pre_lib_loess_rL41rL42 > 1),c('pos_alt')]
XrL4_all_exclude_list <- c(exclude_NCGG_all,exclude_NXG_all,exclude_X5_PAM,exclude_other_ps_pam,exclude_high_pre_lib)
#only removing re-editing outliers and inflated pre/lib's
XrL4_all_tHDR_pos_merge_exclude <- XrL4_all_tHDR_pos_merge_master_df[! XrL4_all_tHDR_pos_merge_master_df$pos_alt %in% XrL4_all_exclude_list,]

#removing all points flagged.
XrL4_all_tHDR_pos_merge_all_filters <- XrL4_all_filter_lib_hdr_cut_df[! XrL4_all_filter_lib_hdr_cut_df$pos_alt %in% XrL4_all_exclude_list,]

#filtered sets compared to full set: XrL4_all_tHDR_pos_merge_all_filters
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),y=rL41rL42_tHDR_pre_lib_loess_rL41rL42, color=conseq))+geom_point(alpha=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='bottom')+ylab('Pre / lib corrected (log-2)')+xlab('Post / pre (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim = c(-6,2.5),xlim=c(-6,2.5))

ggplot(XrL4_all_tHDR_pos_merge_all_filters, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),y=rL41rL42_tHDR_pre_lib_loess_rL41rL42, color=conseq,size = rL41rL42_tHDR_post_lib_loess_rL41rL42))+geom_point(alpha=0.6)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Pre/lib loess2 corr. log-2')+xlab('Post/pre log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_size_continuous(range = c(0.1, 2))+geom_text(aes(label=paste(exon,paste(pos,alt,sep='')),alpha=abs(log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm)-rL41rL42_tHDR_pre_lib_loess_rL41rL42)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black')+coord_cartesian(ylim = c(-5,2.5),xlim=c(-5,2.5))

#testing which points are drastically different between pre/lib+post/pre and post/lib:
#does post/lib give a different result than pre/lib+post/pre (labelled with exon)
grid.arrange(ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm) + rL41rL42_tHDR_pre_lib_loess_rL41rL42, y=rL41rL42_tHDR_post_lib_loess_rL41rL42,color=conseq))+geom_point(alpha=0.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Post/lib corr. log-2')+xlab('Post/pre + pre/lib corr. log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_size_continuous(range = c(0.1, 2)),ggplot(XrL4_all_tHDR_pos_merge_all_filters, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm) + rL41rL42_tHDR_pre_lib_loess_rL41rL42, y=rL41rL42_tHDR_post_lib_loess_rL41rL42,color=conseq))+geom_point(alpha=0.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Post/lib corr. log-2')+xlab('Post/pre + pre/lib corr. log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_size_continuous(range = c(0.1, 2)),ncol = 2)

#unfiltered vs. CADD:
cor(log2(XrL4_all_tHDR_pos_merge_master_df_CADD$tHDR_post_pre_ratio_rL41rL42_mean_synnorm) + XrL4_all_tHDR_pos_merge_master_df_CADD$rL41rL42_tHDR_pre_lib_loess_rL41rL42,as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_tHDR_pos_merge_master_df_CADD$rL41rL42_tHDR_post_lib_loess_rL41rL42,as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_tHDR_pos_merge_master_df_CADD$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df_CADD$CADD.phred)),method='spearman')
#filtered vs. CADD (very slight improvement)
cor(log2(XrL4_all_tHDR_pos_merge_all_filters_CADD$tHDR_post_pre_ratio_rL41rL42_mean_synnorm) + XrL4_all_tHDR_pos_merge_all_filters_CADD$rL41rL42_tHDR_pre_lib_loess_rL41rL42,as.numeric(as.character(XrL4_all_tHDR_pos_merge_all_filters_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_tHDR_pos_merge_all_filters_CADD$rL41rL42_tHDR_post_lib_loess_rL41rL42,as.numeric(as.character(XrL4_all_tHDR_pos_merge_all_filters_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_tHDR_pos_merge_all_filters_CADD$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,as.numeric(as.character(XrL4_all_tHDR_pos_merge_all_filters_CADD$CADD.phred)),method='spearman')

#RNA analysis (excluding exon 18 for now):
XrL4_all_tHDR_pos_merge_all_filters_rna_syn <- XrL4_all_tHDR_pos_merge_all_filters[which((XrL4_all_tHDR_pos_merge_all_filters$conseq == 'SYNONYMOUS' & XrL4_all_tHDR_pos_merge_all_filters$c_hdr_snv == 'True' & XrL4_all_tHDR_pos_merge_all_filters$exon != 'X18' ) | (XrL4_all_tHDR_pos_merge_all_filters$conseq == 'SPLICE_SITE' & XrL4_all_tHDR_pos_merge_all_filters$c_hdr_snv == 'True' & XrL4_all_tHDR_pos_merge_all_filters$CDSpos != 'NA'& XrL4_all_tHDR_pos_merge_all_filters$exon != 'X18')),]  

#missense and syn
XrL4_all_tHDR_pos_merge_all_filters_rna_syn_mis_spl <- XrL4_all_tHDR_pos_merge_all_filters[which((XrL4_all_tHDR_pos_merge_all_filters$conseq == 'SYNONYMOUS' & XrL4_all_tHDR_pos_merge_all_filters$c_hdr_snv == 'True' & XrL4_all_tHDR_pos_merge_all_filters$exon != 'X18') | (XrL4_all_tHDR_pos_merge_all_filters$conseq == 'NON_SYNONYMOUS' & XrL4_all_tHDR_pos_merge_all_filters$c_hdr_snv == 'True' & XrL4_all_tHDR_pos_merge_all_filters$exon != 'X18') | (XrL4_all_tHDR_pos_merge_all_filters$conseq == 'SPLICE_SITE' & XrL4_all_tHDR_pos_merge_all_filters$c_hdr_snv == 'True' & XrL4_all_tHDR_pos_merge_all_filters$CDSpos != 'NA' & XrL4_all_tHDR_pos_merge_all_filters$exon != 'X18')),]  

RNA_intervals <- seq(-5,2,length.out=71)
LOF_SNV_fractions <- c()
SNVs_below_RNA_thresh_count <- c()
for (x in RNA_intervals){
  snvs_below <- which(log2(XrL4_all_tHDR_pos_merge_all_filters_rna_syn_mis_spl$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm) <= x)
  count_snvs_below <- length(snvs_below)
  SNVs_below_RNA_thresh_count <- append(SNVs_below_RNA_thresh_count,count_snvs_below)
  low_fit_count <- length(which(XrL4_all_tHDR_pos_merge_all_filters_rna_syn_mis_spl[snvs_below,'rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean'] <= -1.25)) #threshold for fitness arbitrarily set here as rough bimodal point
  LOF_SNV_fractions <- append(LOF_SNV_fractions,low_fit_count/count_snvs_below)
}
rna_fitness_df <- data.frame(fraction_snvs_LOF = LOF_SNV_fractions, log2_rna_expression_threshold = RNA_intervals, snvs_below_RNA_thresh = SNVs_below_RNA_thresh_count)
#plot showing percent SNVs tolerated below given RNA level
ggplot(rna_fitness_df)+geom_point(aes(x=log2_rna_expression_threshold,y=1-fraction_snvs_LOF,size=SNVs_below_RNA_thresh_count))












#comparison of different enrichment score metrics across all exons:


XrL4_all_tHDR_pos_merge_all_filters_mis_clinvar_p_b <- XrL4_all_tHDR_pos_merge_all_filters[which((XrL4_all_tHDR_pos_merge_all_filters$conseq == 'NON_SYNONYMOUS' & XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple == 'Pathogenic') | (XrL4_all_tHDR_pos_merge_all_filters$conseq == 'NON_SYNONYMOUS' & XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple == 'Benign')),]  

ggplot(XrL4_all_tHDR_pos_merge_all_filters_mis_clinvar_p_b, aes(x=rL41rL42_tHDR_post_lib_loess_rL41rL42, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(XrL4_all_tHDR_pos_merge_all_filters_mis_clinvar_p_b, aes(x=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)


XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b <- XrL4_all_tHDR_pos_merge_all_filters[which( XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple == 'Pathogenic' | XrL4_all_tHDR_pos_merge_all_filters$conseq == 'NON_SYNONYMOUS' & XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple == 'Benign'),] 

ggplot(XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b, aes(x=rL41rL42_tHDR_post_lib_loess_rL41rL42, fill=clinvar_simple,color=conseq))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)
ggplot(XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b, aes(x=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=clinvar_simple,color=conseq))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b_lp_lb <- XrL4_all_tHDR_pos_merge_all_filters[which( XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple == 'Pathogenic' | XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple == 'Benign' | XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple == 'Likely pathogenic' | XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple == 'Likely benign'),]  
clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b_lp_lb <- XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b_lp_lb
clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b_lp_lb$clinvar_simple <- factor(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b_lp_lb$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

ggplot(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b_lp_lb, aes(x=rL41rL42_tHDR_post_lib_loess_rL41rL42, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b_lp_lb, aes(x=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_clinvar_p_b_lp_lb, aes(x=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=clinvar_simple,color=conseq))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#all missense variants ordered by clinvar
XrL4_all_tHDR_pos_merge_all_filters_mis <- XrL4_all_tHDR_pos_merge_all_filters[which((XrL4_all_tHDR_pos_merge_all_filters$conseq == 'NON_SYNONYMOUS')),]  
clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_mis <- XrL4_all_tHDR_pos_merge_all_filters_mis
clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_mis$clinvar_simple <- factor(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_mis$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

ggplot(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_mis, aes(x=rL41rL42_tHDR_post_lib_loess_rL41rL42, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)
ggplot(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters_mis, aes(x=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#all variants ordered by ClinVar annotation
clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters <- XrL4_all_tHDR_pos_merge_all_filters
clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple <- factor(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

ggplot(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters, aes(x=rL41rL42_tHDR_post_lib_loess_rL41rL42, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(clinvar_ord_XrL4_all_tHDR_pos_merge_all_filters, aes(x=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

##NEED TO FILTER THESE CALL SETS IN FINAL ANALYSIS BASED ON GLOBAL FILTERS ABOVE, and perform global min/max normalization and scale each exon by median nonsense depletion across all.
X2rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X2'),]
X3rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X3'),]
X4rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X4'),]
X5rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X5'),]
X15rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X15'),]
X16rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X16'),]
X17rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X17'),]
X18rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X18'),]
X19rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X19'),]
X20rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X20'),]
X21rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X21'),]
X22rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X22'),]
X23rL4_final_df <- XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$exon == 'X23'),]

X2rL4_final_exon_df <- X2rL4_final_df[which(X2rL4_final_df[,'cDNApos'] !='REF' & X2rL4_final_df[,'cDNApos']!='NA'),]
X3rL4_final_exon_df <- X3rL4_final_df[which(X3rL4_final_df[,'cDNApos'] !='REF' & X3rL4_final_df[,'cDNApos']!='NA'),]
X4rL4_final_exon_df <- X4rL4_final_df[which(X4rL4_final_df[,'cDNApos'] !='REF' & X4rL4_final_df[,'cDNApos']!='NA'),]
X5rL4_final_exon_df <- X5rL4_final_df[which(X5rL4_final_df[,'cDNApos'] !='REF' & X5rL4_final_df[,'cDNApos']!='NA'),]
X15rL4_final_exon_df <- X15rL4_final_df[which(X15rL4_final_df[,'cDNApos'] !='REF' & X15rL4_final_df[,'cDNApos']!='NA'),]
X16rL4_final_exon_df <- X16rL4_final_df[which(X16rL4_final_df[,'cDNApos'] !='REF' & X16rL4_final_df[,'cDNApos']!='NA'),]
X17rL4_final_exon_df <- X17rL4_final_df[which(X17rL4_final_df[,'cDNApos'] !='REF' & X17rL4_final_df[,'cDNApos']!='NA'),]
X18rL4_final_exon_df <- X18rL4_final_df[which(X18rL4_final_df[,'cDNApos'] !='REF' & X18rL4_final_df[,'cDNApos']!='NA'),]
X19rL4_final_exon_df <- X19rL4_final_df[which(X19rL4_final_df[,'cDNApos'] !='REF' & X19rL4_final_df[,'cDNApos']!='NA'),]
X20rL4_final_exon_df <- X20rL4_final_df[which(X20rL4_final_df[,'cDNApos'] !='REF' & X20rL4_final_df[,'cDNApos']!='NA'),]
X21rL4_final_exon_df <- X21rL4_final_df[which(X21rL4_final_df[,'cDNApos'] !='REF' & X21rL4_final_df[,'cDNApos']!='NA'),]
X22rL4_final_exon_df <- X22rL4_final_df[which(X22rL4_final_df[,'cDNApos'] !='REF' & X22rL4_final_df[,'cDNApos']!='NA'),]
X23rL4_final_exon_df <- X23rL4_final_df[which(X23rL4_final_df[,'cDNApos'] !='REF' & X23rL4_final_df[,'cDNApos']!='NA'),]

#plots for RNA analysis!
require(gridExtra)

#X2
X2_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X2rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X2_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X2rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X2_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X2_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#same plots with cDNA position instead of CDSpos
X2_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_cDNA_pos <- ggplot(X2rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X2_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_cDNA_pos <- ggplot(X2rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X2_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_cDNA_pos, X2_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_cDNA_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X2rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X2rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X2rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X2rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#which correlates best to pre-ratios?
cor(log2(X2rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X2rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X2rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X2rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X2rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X2rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X2rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X2rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X3
X3_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X3rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X3_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X3rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X3_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X3_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X3rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X3rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X3rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X3rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X3rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X3rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X3rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X3rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X3rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X3rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X3rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X3rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X4
X4_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X4rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X4_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X4rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X4_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X4_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X4rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X4rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X4rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X4rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X4rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X4rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X4rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X4rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X4rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X4rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X4rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X4rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X5
X5_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X5rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X5_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X5rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X5_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X5_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X5rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X5rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X5rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X5rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X5rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X5rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X5rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X5rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X5rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X5rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X5rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X5rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X15
X15_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X15rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X15_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X15rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X15_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X15_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X15rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X15rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X15rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X15rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X15rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X15rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X15rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X15rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X15rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X15rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X15rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X15rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X16
X16_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X16rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+scale_y_continuous(breaks=c(-2,-1,0))
X16_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X16rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X16_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X16_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X16rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X16rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X16rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X16rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X16rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X16rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X16rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X16rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X16rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X16rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X16rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X16rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X17
X17_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X17rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X17_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X17rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X17_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X17_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X17rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X17rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X17rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X17rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X17rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X17rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X17rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X17rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X17rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X17rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X17rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X17rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X18
X18_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X18rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X18_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X18rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X18_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X18_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X18rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X18rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X18rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X18rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X18rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X18rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X18rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X18rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X18rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X18rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X18rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X18rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X19
X19_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X19rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
#modified code to scale limits -- excludes the PAM points
X19_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X19rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+scale_y_continuous(breaks=c(-3,-2,-1,0,1), limits=c(-3,2))

grid.arrange(X19_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X19_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X19rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X19rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X19rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X19rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X19rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X19rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X19rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X19rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X19rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X19rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X19rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X19rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X20
X20_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X20rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X20_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X20rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X20_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X20_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X20rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X20rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X20rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X20rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X20rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X20rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X20rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X20rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X20rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X20rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X20rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X20rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X21
X21_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X21rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X21_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X21rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X21_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X21_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X21rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X21rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X21rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X21rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X21rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X21rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X21rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X21rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X21rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X21rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X21rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X21rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X22
X22_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X22rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X22_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X22rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X22_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X22_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X22rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X22rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X22rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X22rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X22rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X22rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X22rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X22rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X22rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X22rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X22rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X22rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')

#X23
X23_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos <- ggplot(X23rL4_final_exon_df, aes(y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 fitness score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')
X23_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos <- ggplot(X23rL4_final_exon_df, aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour='Gray',alpha=0.15),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Log-2 RNA')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

grid.arrange(X23_tHDR_post_lib_loess_e.sns_rL41rL42_mean_by_pos, X23_tHDR_rna_pre_ratio_rL41rL42_mean_synnorm_by_pos, nrow=2)

#rna_pre ratio across replicates:
ggplot(X23rL4_final_exon_df, aes(x=log2(tHDR_rna_pre_ratio_synnorm),y=log2(rL42_tHDR_rna_pre_ratio_synnorm)))+geom_point(aes(colour=conseq),size=1)+geom_abline(intercept=0,slope=1,lty=2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Rep. 2 rna/pre')+xlab('Rep. 1 rna/pre')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL41
ggplot(X23rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq),y=log2(tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#rna vs. pre by replicate, rL42
ggplot(X23rL4_final_exon_df, aes(x=log2(rL42_tHDR_pre_pseudo_freq),y=log2(rL42_tHDR_rna_pseudo_freq)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
#average rna vs. average pre
ggplot(X23rL4_final_exon_df, aes(x=log2(tHDR_pre_pseudo_freq/2+rL42_tHDR_pre_pseudo_freq/2),y=log2(tHDR_rna_pseudo_freq/2+rL42_tHDR_rna_pseudo_freq/2)))+geom_point(aes(colour=conseq,size=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('RNA freq. (log-2)')+xlab('DNA freq. D5 (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#which correlates best to pre-ratios?
cor(log2(X23rL4_final_exon_df$tHDR_pre_pseudo_freq),log2(X23rL4_final_exon_df$tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X23rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq),log2(X23rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq),method='spearman')
cor(log2(X23rL4_final_exon_df$tHDR_pre_pseudo_freq/2+X23rL4_final_exon_df$rL42_tHDR_pre_pseudo_freq/2),log2(X23rL4_final_exon_df$tHDR_rna_pseudo_freq/2+X23rL4_final_exon_df$rL42_tHDR_rna_pseudo_freq/2),method='spearman')
### end of RNA section

# add a new column to:
#compute the exonic position of each variant (make a string with negative or positive annotation instead of NA --> will have to order this to be able to plot (plot one order, and then add vector of strings for labels, maybe?)
# clinvar nucleotide positions / other categorical translations between data sets...
# conseq without 'splice' but either exonic or intronic

#FINAL NORMALIZATION SECTION (SHOULD COME BEFORE RNA PLOTS above)
# min/max normalization on filtered data...
#use conseq or conseq_simp to not include splice?

#general formula:  b/(a-c)*(A-C) where b is snv, a is exon med syn, c is exon med nonsense, A is global med syn, C is global med nonsense.

#do this first for the repl. mean fits and scores :
XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn <- median(XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])

XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns <- median(XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])

X2_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X2rL4_final_df[which(X2rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X2_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X2rL4_final_df[which(X2rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X2rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X2rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X2_tHDR_post_lib_loess_rL41rL42_median_syn-X2_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X3_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X3rL4_final_df[which(X3rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X3_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X3rL4_final_df[which(X3rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X3rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X3rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X3_tHDR_post_lib_loess_rL41rL42_median_syn-X3_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X4_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X4rL4_final_df[which(X4rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X4_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X4rL4_final_df[which(X4rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X4rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X4rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X4_tHDR_post_lib_loess_rL41rL42_median_syn-X4_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X5_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X5rL4_final_df[which(X5rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X5_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X5rL4_final_df[which(X5rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X5rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X5rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X5_tHDR_post_lib_loess_rL41rL42_median_syn-X5_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X15_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X15rL4_final_df[which(X15rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X15_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X15rL4_final_df[which(X15rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X15rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X15rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X15_tHDR_post_lib_loess_rL41rL42_median_syn-X15_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X16_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X16rL4_final_df[which(X16rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X16_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X16rL4_final_df[which(X16rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X16rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X16rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X16_tHDR_post_lib_loess_rL41rL42_median_syn-X16_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X17_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X17rL4_final_df[which(X17rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X17_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X17rL4_final_df[which(X17rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X17rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X17rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X17_tHDR_post_lib_loess_rL41rL42_median_syn-X17_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X18_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X18rL4_final_df[which(X18rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X18_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X18rL4_final_df[which(X18rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X18rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X18rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X18_tHDR_post_lib_loess_rL41rL42_median_syn-X18_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X19_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X19rL4_final_df[which(X19rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X19_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X19rL4_final_df[which(X19rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X19rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X19rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X19_tHDR_post_lib_loess_rL41rL42_median_syn-X19_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X20_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X20rL4_final_df[which(X20rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X20_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X20rL4_final_df[which(X20rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X20rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X20rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X20_tHDR_post_lib_loess_rL41rL42_median_syn-X20_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X21_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X21rL4_final_df[which(X21rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X21_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X21rL4_final_df[which(X21rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X21rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X21rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X21_tHDR_post_lib_loess_rL41rL42_median_syn-X21_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X22_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X22rL4_final_df[which(X22rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X22_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X22rL4_final_df[which(X22rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X22rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X22rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X22_tHDR_post_lib_loess_rL41rL42_median_syn-X22_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

X23_tHDR_post_lib_loess_rL41rL42_median_syn <- median(X23rL4_final_df[which(X23rL4_final_df$conseq == 'SYNONYMOUS'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X23_tHDR_post_lib_loess_rL41rL42_median_ns <- median(X23rL4_final_df[which(X23rL4_final_df$conseq == 'STOP_GAINED'),c('rL41rL42_tHDR_post_lib_loess_rL41rL42')])
X23rL4_final_df$tHDR_post_lib_loess_rL41rL42_sns_norm <- (X23rL4_final_df$rL41rL42_tHDR_post_lib_loess_rL41rL42/(X23_tHDR_post_lib_loess_rL41rL42_median_syn-X23_tHDR_post_lib_loess_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_rL41rL42_median_ns)

#repeat the above for the e.sns replicate means... what does this look like keeping the replicate normalized scored separate?

XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(XrL4_all_tHDR_pos_merge_all_filters[which(XrL4_all_tHDR_pos_merge_all_filters$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])

X2_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X2rL4_final_df[which(X2rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X2_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X2rL4_final_df[which(X2rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X2rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X2rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X2_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X2_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X3_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X3rL4_final_df[which(X3rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X3_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X3rL4_final_df[which(X3rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X3rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X3rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X3_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X3_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X4_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X4rL4_final_df[which(X4rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X4_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X4rL4_final_df[which(X4rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X4rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X4rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X4_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X4_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X5_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X5rL4_final_df[which(X5rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X5_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X5rL4_final_df[which(X5rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X5rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X5rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X5_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X5_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X15_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X15rL4_final_df[which(X15rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X15_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X15rL4_final_df[which(X15rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X15rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X15rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X15_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X15_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X16_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X16rL4_final_df[which(X16rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X16_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X16rL4_final_df[which(X16rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X16rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X16rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X16_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X16_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X17_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X17rL4_final_df[which(X17rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X17_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X17rL4_final_df[which(X17rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X17rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X17rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X17_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X17_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X18_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X18rL4_final_df[which(X18rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X18_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X18rL4_final_df[which(X18rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X18rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X18rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X18_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X18_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X19_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X19rL4_final_df[which(X19rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X19_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X19rL4_final_df[which(X19rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X19rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X19rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X19_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X19_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X20_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X20rL4_final_df[which(X20rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X20_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X20rL4_final_df[which(X20rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X20rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X20rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X20_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X20_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X21_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X21rL4_final_df[which(X21rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X21_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X21rL4_final_df[which(X21rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X21rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X21rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X21_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X21_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X22_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X22rL4_final_df[which(X22rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X22_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X22rL4_final_df[which(X22rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X22rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X22rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X22_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X22_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X23_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn <- median(X23rL4_final_df[which(X23rL4_final_df$conseq == 'SYNONYMOUS'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X23_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns <- median(X23rL4_final_df[which(X23rL4_final_df$conseq == 'STOP_GAINED'),c('rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean')])
X23rL4_final_df$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- (X23rL4_final_df$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean/(X23_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X23_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

#make the same calculations on a per replicate score basis...:
#keeping all scalers as medians of replicate averages (i.e. global syn/ns and per exon syn/ns will be called from medians, but )
X2rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X2rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X2_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X2_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X2rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X2rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X2_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X2_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X3rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X3rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X3_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X3_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X3rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X3rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X3_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X3_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X4rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X4rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X4_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X4_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X4rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X4rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X4_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X4_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X5rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X5rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X5_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X5_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X5rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X5rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X5_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X5_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X15rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X15rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X15_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X15_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X15rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X15rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X15_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X15_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X16rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X16rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X16_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X16_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X16rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X16rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X16_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X16_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X17rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X17rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X17_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X17_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X17rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X17rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X17_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X17_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X18rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X18rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X18_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X18_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X18rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X18rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X18_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X18_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X19rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X19rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X19_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X19_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X19rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X19rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X19_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X19_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X20rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X20rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X20_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X20_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X20rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X20rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X20_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X20_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X21rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X21rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X21_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X21_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X21rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X21rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X21_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X21_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X22rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X22rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X22_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X22_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X22rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X22rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X22_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X22_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

X23rL4_final_df$tHDR_post_lib_loess_e.sns_rL41_sns_norm <- (X23rL4_final_df$rL41_tHDR_post_lib_loess_rL41_e.sns/(X23_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X23_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)
X23rL4_final_df$tHDR_post_lib_loess_e.sns_rL42_sns_norm <- (X23rL4_final_df$rL42_tHDR_post_lib_loess_rL42_e.sns/(X23_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-X23_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns))*(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn-XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns)

#binding all of the sns-normalized df's together...
#this data frame has been variant filtered and then normalized!!!
XrL4_all_final_sns_norm <- rbind(X2rL4_final_df,X3rL4_final_df,X4rL4_final_df,X5rL4_final_df,X15rL4_final_df,X16rL4_final_df,X17rL4_final_df,X18rL4_final_df,X19rL4_final_df,X20rL4_final_df,X21rL4_final_df,X22rL4_final_df,X23rL4_final_df)
### calculate replicate dependent post-pre + lo.e.sns pre-lib scores for all points
#calculate a post-pre + pre-lib metric per replicate and compare to replicate reproducibility with other functional scores
XrL4_all_final_sns_norm$rL41_tHDR_pre_lib_lo_plus_post_pre <- XrL4_all_final_sns_norm$rL41_tHDR_pre_lib_loess_rL41+log2(XrL4_all_final_sns_norm$tHDR_post_pre_ratio_synnorm)
XrL4_all_final_sns_norm$rL42_tHDR_pre_lib_lo_plus_post_pre <- XrL4_all_final_sns_norm$rL42_tHDR_pre_lib_loess_rL41+log2(XrL4_all_final_sns_norm$rL42_tHDR_post_pre_ratio_synnorm)
XrL4_all_final_sns_norm$tHDR_pre_lib_lo_plus_post_pre_rL41rL42_mean <- (XrL4_all_final_sns_norm$rL42_tHDR_pre_lib_lo_plus_post_pre+XrL4_all_final_sns_norm$rL41_tHDR_pre_lib_lo_plus_post_pre)/2
#weighted by pre reads...
XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41rL42w_sns_norm <- XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm*XrL4_all_final_sns_norm$tHDR_pre_weight + XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm*XrL4_all_final_sns_norm$rL42_tHDR_pre_weight

#write this out:
#write.table(XrL4_all_final_sns_norm, "/mount/SGE/BRCA1/XrL4_all_final_sns_norm_20171016.txt", sep="\t")
#XrL4_all_final_sns_norm <- read.table("/mount/SGE/BRCA1/XrL4_all_final_sns_norm_20171016.txt", sep="\t", header=TRUE)


conseq_ord_XrL4_all_final_sns_norm <- XrL4_all_final_sns_norm
conseq_ord_XrL4_all_final_sns_norm$conseq <- factor(conseq_ord_XrL4_all_final_sns_norm$conseq, levels = c('STOP_GAINED','CANONICAL_SPLICE','SYNONYMOUS','INTRONIC','SPLICE_SITE','5PRIME_UTR','NON_SYNONYMOUS'))

#all scoring metrics compared...


grid.arrange(ggplot(conseq_ord_XrL4_all_final_sns_norm, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm), fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3,1)),ggplot(conseq_ord_XrL4_all_final_sns_norm, aes(x=log2(tHDR_post_lib_ratio_rL41rL42_mean_synnorm), fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3,1)),ggplot(conseq_ord_XrL4_all_final_sns_norm, aes(x=rL41rL42_tHDR_post_lib_loess_rL41rL42, fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3,1)),ggplot(conseq_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3,1)),ggplot(conseq_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42w_sns_norm, fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3,1)),nrow=5)

XrL4_all_final_sns_norm_mis_clinvar_p_b <- XrL4_all_final_sns_norm[which((XrL4_all_final_sns_norm$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm$clinvar_simple == 'Pathogenic') | (XrL4_all_final_sns_norm$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm$clinvar_simple == 'Benign')),]  

ggplot(XrL4_all_final_sns_norm_mis_clinvar_p_b, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(XrL4_all_final_sns_norm_mis_clinvar_p_b, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)


XrL4_all_final_sns_norm_clinvar_p_b <- XrL4_all_final_sns_norm[which( XrL4_all_final_sns_norm$clinvar_simple == 'Pathogenic' | XrL4_all_final_sns_norm$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm$clinvar_simple == 'Benign'),]  

ggplot(XrL4_all_final_sns_norm_clinvar_p_b, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(XrL4_all_final_sns_norm_clinvar_p_b, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

XrL4_all_final_sns_norm_clinvar_p_b_lp_lb <- XrL4_all_final_sns_norm[which( XrL4_all_final_sns_norm$clinvar_simple == 'Pathogenic' | XrL4_all_final_sns_norm$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm$clinvar_simple == 'Benign' | XrL4_all_final_sns_norm$clinvar_simple == 'Likely pathogenic' | XrL4_all_final_sns_norm$clinvar_simple == 'Likely benign'),]  
clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb <- XrL4_all_final_sns_norm_clinvar_p_b_lp_lb
clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb$clinvar_simple <- factor(clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

#with colored boundaries for conseq
grid.arrange(ggplot(clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=clinvar_simple,color=conseq))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple),ggplot(clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple,color=conseq))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple),nrow=2)

#without colored boundaries for conseq
ggplot(clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#all missense variants ordered by clinvar
XrL4_all_final_sns_norm_mis <- XrL4_all_final_sns_norm[which((XrL4_all_final_sns_norm$conseq == 'NON_SYNONYMOUS')),]  
clinvar_ord_XrL4_all_final_sns_norm_mis <- XrL4_all_final_sns_norm_mis
clinvar_ord_XrL4_all_final_sns_norm_mis$clinvar_simple <- factor(clinvar_ord_XrL4_all_final_sns_norm_mis$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

ggplot(clinvar_ord_XrL4_all_final_sns_norm_mis, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(clinvar_ord_XrL4_all_final_sns_norm_mis, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#clinvar_ord_XrL4_all_final_sns_norm_mis filtered on p/lp b/lb
clinvar_ord_XrL4_all_final_sns_norm_mis_p_b_lp_lb <- clinvar_ord_XrL4_all_final_sns_norm_mis[which( clinvar_ord_XrL4_all_final_sns_norm_mis$clinvar_simple == 'Pathogenic' | clinvar_ord_XrL4_all_final_sns_norm_mis$clinvar_simple == 'Benign' | clinvar_ord_XrL4_all_final_sns_norm_mis$clinvar_simple == 'Likely pathogenic' | clinvar_ord_XrL4_all_final_sns_norm_mis$clinvar_simple == 'Likely benign'),]

#only missense
ggplot(clinvar_ord_XrL4_all_final_sns_norm_mis_p_b_lp_lb, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#all variants ordered by ClinVar annotation
clinvar_ord_XrL4_all_final_sns_norm <- XrL4_all_final_sns_norm
clinvar_ord_XrL4_all_final_sns_norm$clinvar_simple <- factor(clinvar_ord_XrL4_all_final_sns_norm$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

ggplot(clinvar_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)
ggplot(clinvar_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#isolate all synonymous and missense variants and test how rna_pre_ratio correlates with survival (c_hdr_snv only, so RNA levels can be detected.
#only syn
XrL4_all_final_sns_norm_syn <- XrL4_all_final_sns_norm[which((XrL4_all_final_sns_norm$conseq == 'SYNONYMOUS' & XrL4_all_final_sns_norm$c_hdr_snv == 'True' & XrL4_all_final_sns_norm$exon != 'X18') | (XrL4_all_final_sns_norm$conseq == 'SPLICE_SITE' & XrL4_all_final_sns_norm$c_hdr_snv == 'True' & XrL4_all_final_sns_norm$CDSpos != 'NA' & XrL4_all_final_sns_norm$exon != 'X18')),]

XrL4_all_final_sns_norm_ns <- XrL4_all_final_sns_norm[which((XrL4_all_final_sns_norm$conseq == 'STOP_GAINED' & XrL4_all_final_sns_norm$c_hdr_snv == 'True')),]
XrL4_all_final_sns_norm_ns_no18 <- XrL4_all_final_sns_norm[which((XrL4_all_final_sns_norm$conseq == 'STOP_GAINED' & XrL4_all_final_sns_norm$c_hdr_snv == 'True' & XrL4_all_final_sns_norm$exon != 'X18')),]



#log2 median and sd for rna ratios
XrL4_all_final_sns_norm_syn_rna_median <- median(log2(XrL4_all_final_sns_norm_syn$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm))
XrL4_all_final_sns_norm_syn_rna_sd <- sd(log2(XrL4_all_final_sns_norm_syn$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm))
#defined as being within 1 SD from median (log2 taken here to compare
XrL4_all_final_sns_norm_syn_norm_exp <- XrL4_all_final_sns_norm_syn[which(XrL4_all_final_sns_norm_syn_rna_median + XrL4_all_final_sns_norm_syn_rna_sd >= log2(XrL4_all_final_sns_norm_syn$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm) & log2(XrL4_all_final_sns_norm_syn$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm) >= XrL4_all_final_sns_norm_syn_rna_median-XrL4_all_final_sns_norm_syn_rna_sd),]


ggplot(XrL4_all_final_sns_norm_syn, aes(y=tHDR_post_lib_loess_rL41rL42_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), fill=conseq))+geom_point(aes(colour=conseq),alpha=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Fitness (D11/lib, log-2 corr.)')+xlab('D5 RNA expression (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

ggplot(XrL4_all_final_sns_norm_syn, aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), fill=conseq))+geom_point(aes(colour=conseq),alpha=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Fitness (D11/lib, log-2 corr.)')+xlab('D5 RNA expression (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#with labels to understand outliers (post-lib discordancy label)... seems to be problem with one replicate's 'post' value?
ggplot(XrL4_all_final_sns_norm_syn, aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), fill=conseq))+geom_point(aes(colour=conseq),alpha=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Fitness (D11/lib, log-2 corr.)')+xlab('D5 RNA expression (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_text(aes(label=paste(paste(exon,pos),paste(tHDR_post_lib_loess_e.sns_rL41_sns_norm,tHDR_post_lib_loess_e.sns_rL42_sns_norm),'  ') ,alpha=abs(tHDR_post_lib_loess_e.sns_rL41_sns_norm-tHDR_post_lib_loess_e.sns_rL42_sns_norm)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black')


### global correlations for different scores, and global comparisons of 'final' scores
### this is still a hanger on call to y=tHDR_post_lib_loess_e.sns_rL41rL42
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_pre_lib_lo_plus_post_pre_rL41rL42_mean, y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=conseq))+geom_point(aes(colour=conseq),alpha=0.3,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('e.sns functional score')+xlab('pre.lib_plus_post.pre functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
cor(XrL4_all_final_sns_norm$tHDR_pre_lib_lo_plus_post_pre_rL41rL42_mean,XrL4_all_final_sns_norm$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,method='spearman')

ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_pre_lib_lo_plus_post_pre_rL41rL42_mean, y=rL41rL42_tHDR_post_lib_loess_rL41rL42, fill=conseq))+geom_point(aes(colour=conseq),alpha=0.3,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('mean-fit score')+xlab('pre.lib_plus_post.pre score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
cor(XrL4_all_final_sns_norm$tHDR_pre_lib_lo_plus_post_pre_rL41rL42_mean,XrL4_all_final_sns_norm$rL41rL42_tHDR_post_lib_loess_rL41rL42,method='spearman')

ggplot(XrL4_all_final_sns_norm, aes(x=rL41rL42_tHDR_post_lib_loess_rL41rL42, y=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=conseq))+geom_point(aes(colour=conseq),alpha=0.3,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('e.sns functional score')+xlab('mean-fit score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
cor(XrL4_all_final_sns_norm$rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean,XrL4_all_final_sns_norm$rL41rL42_tHDR_post_lib_loess_rL41rL42,method='spearman')

#normalized scores (two different ways).
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=conseq))+geom_point(aes(colour=conseq),alpha=0.3,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('mean-fit score')+xlab('pre.lib_plus_post.pre score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)
cor(XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,XrL4_all_final_sns_norm$tHDR_post_lib_loess_rL41rL42_sns_norm,method='spearman')


### global correlations with CADD to determine optimal 'fitness score' from these candidates
#has annotation in CADD
XrL4_variants_with_CADD <- XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$CADD.phred != 'NA' & XrL4_all_tHDR_pos_merge_master_df$CADD.phred != 'REF' ),c('pos_alt')]
#filter XrL4_all_final_sns_norm on CADD variants:
XrL4_all_final_sns_norm_CADD <- XrL4_all_final_sns_norm[XrL4_all_final_sns_norm$pos_alt %in% XrL4_variants_with_CADD,]
#compare correlations of top three metrics...
cor(XrL4_all_final_sns_norm_CADD$tHDR_post_lib_loess_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_final_sns_norm_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_final_sns_norm_CADD$rL41rL42_tHDR_post_lib_loess_rL41rL42,as.numeric(as.character(XrL4_all_final_sns_norm_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_final_sns_norm_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_final_sns_norm_CADD$tHDR_post_lib_loess_e.sns_rL41rL42w_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_CADD$CADD.phred)),method='spearman')
#individual scores vs. cadd scores
cor(XrL4_all_final_sns_norm_CADD$rL41_tHDR_post_lib_loess_rL41_e.sns,as.numeric(as.character(XrL4_all_final_sns_norm_CADD$CADD.phred)),method='spearman')
cor(XrL4_all_final_sns_norm_CADD$rL42_tHDR_post_lib_loess_rL42_e.sns,as.numeric(as.character(XrL4_all_final_sns_norm_CADD$CADD.phred)),method='spearman')

### WORKING IN THESE UPDATES!
### final filtering strategy to reduce discordant calls between replicates?
### with remaining points, stats to get P-values
### global QC correlations:
#with labels to understand outliers (post-lib discordancy label)... seems to be problem with one replicate's 'post' value?

#replicate discrepancy plot w/o labels:  which scoring system is more reproducible? about same
#pre sns normalization across exons...
grid.arrange(ggplot(XrL4_all_final_sns_norm, aes(x=rL41_tHDR_post_lib_loess_rL41_e.sns, y=rL42_tHDR_post_lib_loess_rL42_e.sns, fill=conseq))+geom_point(aes(colour=conseq),alpha=0.3,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Functional score rep. 2')+xlab('Functional score rep. 1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-5,2),ylim=c(-5,2)),ggplot(XrL4_all_final_sns_norm, aes(x=rL41_tHDR_pre_lib_lo_plus_post_pre, y=rL42_tHDR_pre_lib_lo_plus_post_pre, fill=conseq))+geom_point(aes(colour=conseq),alpha=0.3,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Functional score rep. 2')+xlab('Functional score rep. 1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-5,2),ylim=c(-5,2)),nrow=2)
cor(XrL4_all_final_sns_norm$rL41_tHDR_post_lib_loess_rL41_e.sns,XrL4_all_final_sns_norm$rL42_tHDR_post_lib_loess_rL42_e.sns,method='spearman')
cor(XrL4_all_final_sns_norm$rL41_tHDR_pre_lib_lo_plus_post_pre,XrL4_all_final_sns_norm$rL42_tHDR_pre_lib_lo_plus_post_pre,method='spearman')

#post sns normalization across exons...
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm, fill=conseq))+geom_point(aes(colour=conseq),alpha=0.3,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Functional score rep. 2')+xlab('Functional score rep. 1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)
cor(XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')

#replicate discrepancy plot with labels:
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm, fill=conseq))+geom_point(aes(colour=conseq,size=pre_weight),alpha=0.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Functional score rep. 2')+xlab('Functional score rep. 1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_text(aes(label=paste(paste(exon,pos),paste(tHDR_post_lib_loess_e.sns_rL41_sns_norm,tHDR_post_lib_loess_e.sns_rL42_sns_norm),'  ') ,alpha=abs(tHDR_post_lib_loess_e.sns_rL41_sns_norm-tHDR_post_lib_loess_e.sns_rL42_sns_norm)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black')

#fitness scores of SNVs with with normal expression
ggplot(XrL4_all_final_sns_norm_syn_norm_exp, aes(y=tHDR_post_lib_loess_e.sns_rL41_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), fill=conseq))+geom_point(aes(colour=conseq),alpha=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Fitness (D11/lib, log-2 corr.)')+xlab('D5 RNA expression (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#histogram comparing snv's with normal expression values and those with all expression
null_compare_tHDR_post_lib_loess_rL41rL42_sns_norm <- grid.arrange(ggplot(XrL4_all_final_sns_norm_syn, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=conseq))+geom_histogram(aes(colour=conseq),alpha=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Fitness (D11/lib, log-2 p.snsn)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3.2,1.2)),ggplot(XrL4_all_final_sns_norm_syn_norm_exp, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=conseq))+geom_histogram(aes(colour=conseq),alpha=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Fitness (D11/lib, log-2 p.snsn)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3.2,1.2)),nrow=2)
null_compare_tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- grid.arrange(ggplot(XrL4_all_final_sns_norm_syn, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=conseq))+geom_histogram(aes(colour=conseq),alpha=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Fitness (D11/lib, e.sns.mean log-2)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3.2,1.2)),ggplot(XrL4_all_final_sns_norm_syn_norm_exp, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=conseq))+geom_histogram(aes(colour=conseq),alpha=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Fitness (D11/lib, e.sns.mean log-2)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3.2,1.2)),nrow=2)
grid.arrange(null_compare_tHDR_post_lib_loess_rL41rL42_sns_norm,null_compare_tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,nrow=1)

mean(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)
sd(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)
mean(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)
sd(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)

mean(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)-2*sd(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)
mean(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)+2*sd(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)
mean(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)-3*sd(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)
mean(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)+3*sd(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)


#add nonsense variants:
XrL4_all_final_sns_norm_ns_syn_normex <- rbind(XrL4_all_final_sns_norm_ns,XrL4_all_final_sns_norm_syn_norm_exp)

#three ways to compare functional / non-functional split, all similar:
bimodal_compare_tHDR_pre_lib_lo_plus_post_pre_rL41rL42_mean <- ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_pre_lib_lo_plus_post_pre_rL41rL42_mean, fill=conseq))+geom_histogram(aes(fill=conseq),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Functional score (pr.lib.lo+po.pr)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1.2))
bimodal_compare_tHDR_post_lib_loess_rL41rL42_sns_norm <- ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=conseq))+geom_histogram(aes(fill=conseq),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Functional score (mean.p.sns)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1.2))
bimodal_compare_tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm <- ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=conseq))+geom_histogram(aes(fill=conseq),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Functional score (e.sns.mean)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1.2))
#it makes virtually no difference if this is weighted by 'rL41 or rL42 pre frequency'
bimodal_compare_tHDR_post_lib_loess_e.sns_rL41rL42w_sns_norm <- ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42w_sns_norm, fill=conseq))+geom_histogram(aes(fill=conseq),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Functional score (e.sns.mean)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1.2))
grid.arrange(bimodal_compare_tHDR_pre_lib_lo_plus_post_pre_rL41rL42_mean,bimodal_compare_tHDR_post_lib_loess_rL41rL42_sns_norm,bimodal_compare_tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,bimodal_compare_tHDR_post_lib_loess_e.sns_rL41rL42w_sns_norm,nrow=2)

#missense and syn
XrL4_all_final_sns_norm_syn_mis_spl_noX18 <- XrL4_all_final_sns_norm[which((XrL4_all_final_sns_norm$conseq == 'SYNONYMOUS' & XrL4_all_final_sns_norm$c_hdr_snv == 'True' & XrL4_all_final_sns_norm$exon != 'X18') | (XrL4_all_final_sns_norm$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm$c_hdr_snv == 'True' & XrL4_all_final_sns_norm$exon != 'X18') | (XrL4_all_final_sns_norm$conseq == 'SPLICE_SITE' & XrL4_all_final_sns_norm$c_hdr_snv == 'True' & XrL4_all_final_sns_norm$CDSpos != 'NA' & XrL4_all_final_sns_norm$exon != 'X18')),]  


#### modeling section

library(gam)
#define 'lof' as a 1 or 0 based on nonsense SNVs / normal expression SYN SNVs.

assign_lof <- function(conseq){
  if (conseq == 'STOP_GAINED') return(1) else return(0)
}

XrL4_all_final_sns_norm_ns_syn_normex_gam <- XrL4_all_final_sns_norm_ns_syn_normex
XrL4_all_final_sns_norm_ns_syn_normex_gam$lof <- sapply(XrL4_all_final_sns_norm_ns_syn_normex_gam$conseq,assign_lof)

#modeling with GAM -- use interacting terms and make this better? risk of overfitting
XrL4_all_gam_out <- gam(lof ~ tHDR_post_lib_loess_e.sns_rL41_sns_norm+tHDR_post_lib_loess_e.sns_rL42_sns_norm, data = XrL4_all_final_sns_norm_ns_syn_normex_gam, family = "binomial")
XrL4_all_gam_out$coefficients
XrL4_all_gam_pred <- predict(XrL4_all_gam_out, newdata = XrL4_all_final_sns_norm, se=TRUE)
XrL4_all_gam_pred$fit

XrL4_all_gam_out <- gam(lof ~ tHDR_post_lib_loess_e.sns_rL41_sns_norm+tHDR_post_lib_loess_e.sns_rL42_sns_norm, data = XrL4_all_final_sns_norm_ns_syn_normex_gam, family = "binomial")
XrL4_all_gam_out$coefficients
XrL4_all_gam_pred <- predict(XrL4_all_gam_out, newdata = XrL4_all_final_sns_norm, se=TRUE)
XrL4_all_gam_pred$fit


#MclustDA approach for Gaussian Mixture Modelling...
#using class from above... LOF = 1, F = 0
require(mclust)
XrL4_all_mclust.train <- MclustDA(data = XrL4_all_final_sns_norm_ns_syn_normex_gam[,c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')], class = XrL4_all_final_sns_norm_ns_syn_normex_gam$lof, G = 1, modelNames = c('V'))
summary(XrL4_all_mclust.train, parameters = TRUE)
XrL4_all_mclust_lof_out <- rep('all_snvs',dim(XrL4_all_final_sns_norm)[1])
summary(XrL4_all_mclust.train, newdata = XrL4_all_final_sns_norm[, c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')], newclass = XrL4_all_mclust_lof_out)
XrL4_all_mclust.predict <- predict.MclustDA(XrL4_all_mclust.train, newdata = XrL4_all_final_sns_norm[, c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')], newclass = XrL4_all_mclust_lof_out,prior = c(0.5,0.5))
XrL4_all_final_sns_norm$mclust_z_lof <- XrL4_all_mclust.predict$z[,2]
XrL4_all_final_sns_norm$mclust_class_lof <- XrL4_all_mclust.predict$classification


max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_class_lof == 0),c('mclust_z_lof')])
min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_class_lof == 0),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_class_lof == 1),c('mclust_z_lof')])
max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_class_lof == 1),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])

mclust_functional_thresh <- mean(c(min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_class_lof == 0),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_class_lof == 1),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])))

mclust_lof_95 <- max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_z_lof > 0.95),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
mclust_lof_99 <- max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_z_lof > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])

mclust_f_05 <- min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_z_lof < 0.05),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
mclust_f_01 <- min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_z_lof < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$mclust_z_lof < 0.001),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])

#How do these compare to the threshold using normalmixEM?



#frequentist, two normal distributions:
#redo null distributions with this...
#get all the P-values using syn_norm_exp 
XrL4_test_p_syn <- pnorm(XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,mean = mean(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm), sd = sd(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm))
#get all the P-values using ns
XrL4_test_p_ns <- pnorm(XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,mean = mean(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm), sd = sd(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm), lower.tail = FALSE)
XrL4_ratio_ns_syn <- XrL4_test_p_ns/XrL4_test_p_syn
XrL4_log_ratio_ns_syn <- log10(XrL4_ratio_ns_syn)

XrL4_all_final_sns_norm$XrL4_test_p_syn <- XrL4_test_p_syn
XrL4_all_final_sns_norm$XrL4_test_p_ns <- XrL4_test_p_ns
XrL4_all_final_sns_norm$XrL4_ratio_ns_syn <- XrL4_ratio_ns_syn
XrL4_all_final_sns_norm$XrL4_log_ratio_ns_syn <- XrL4_log_ratio_ns_syn
XrL4_all_final_sns_norm$XrL4_bh_p_syn <- p.adjust(XrL4_all_final_sns_norm$XrL4_test_p_syn, method = 'BH')
XrL4_all_final_sns_norm$XrL4_bh_p_ns <- p.adjust(XrL4_all_final_sns_norm$XrL4_test_p_ns, method = 'BH')

#suggested thresholds...
#uncorrected threshold for less than functional...
max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_test_p_syn < 0.05),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
#corrected threshold for less than functional...
XrL4_bh_p_syn_max05 <- max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_bh_p_syn < 0.05),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_bh_p_syn < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])

#uncorrected threshold for greater than nonsense...
min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_test_p_ns < 0.05),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
#corrected threshold for greater than nonsense...
XrL4_bh_p_ns_min05 <- min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_bh_p_ns < 0.05),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_bh_p_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])

#calculate the 'inflection point' threshold in the data... where P(ns) and P(syn) are equal...
#take the mean between the two points on either side of the threshold..
XrL4_bh_p_binary_cutoff <- mean(c(max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_bh_p_ns > XrL4_all_final_sns_norm$XrL4_bh_p_syn),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_bh_p_ns < XrL4_all_final_sns_norm$XrL4_bh_p_syn),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])))


#order the dataframe in descending order by P(SYN)

#histograms of training data and threshold for p-value FDR approach:
grid.arrange(ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=conseq))+geom_histogram(aes(fill=conseq),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('Functional score (e.sns.mean)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1.2))+geom_vline(xintercept=XrL4_bh_p_syn_max05 ,lty=2)+geom_vline(xintercept=XrL4_bh_p_ns_min05,lty=2),ggplot(conseq_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Functional score (e.sns.mean)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1.2))+geom_vline(xintercept=XrL4_bh_p_syn_max05 ,lty=2)+geom_vline(xintercept=XrL4_bh_p_ns_min05,lty=2),nrow=2)


#BAYSEIAN MODELING SECTION

library(mixtools)
library(plotROC)
##mixtools normalmixEM --> Basic fit on known distributions... re-try this using fixed
ns_syn_normex_mm.opt <- normalmixEM(x = XrL4_all_final_sns_norm_ns_syn_normex$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, lambda = c(dim(XrL4_all_final_sns_norm_syn_norm_exp)[1]/dim(XrL4_all_final_sns_norm_ns_syn_normex)[1],dim(XrL4_all_final_sns_norm_ns)[1]/dim(XrL4_all_final_sns_norm_ns_syn_normex)[1]),mean.constr = c(mean(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm),mean(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)), sd.constr = c(sd(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm),sd(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)))
ns_syn_normex_mm.opt$mu
ns_syn_normex_mm.opt$sigma

#try calling on the complete data set --> means and sds start at defined locations, then move.
normalmixEM(x = XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, mu = c(mean(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm),mean(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)),sigma = c(sd(XrL4_all_final_sns_norm_syn_norm_exp$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm),sd(XrL4_all_final_sns_norm_ns$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)),lambda = c(0.5,0.5))

#try calling on the complete data set, using fixed mu and sigma...with lambda = c(0.5,0.5) (arbitrary) --> starting from 0.5, 0.5.
XrL4_all_mm.out <- normalmixEM(x = XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, mean.constr = ns_syn_normex_mm.opt$mu, sd.constr = ns_syn_normex_mm.opt$sigma, lambda = c(0.5,0.5))

#try calling on the complete data set, using fixed mu and sigma...with lambda from syn/(ns+syn) (arbitrary) --> usually converges on the exact same values as above for lambda...
#XrL4_all_mm.out <- normalmixEM(x = XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, mean.constr = ns_syn_normex_mm.opt$mu, sd.constr = ns_syn_normex_mm.opt$sigma, lambda = c(dim(XrL4_all_final_sns_norm_syn_norm_exp)[1]/dim(XrL4_all_final_sns_norm_ns_syn_normex)[1],dim(XrL4_all_final_sns_norm_ns)[1]/dim(XrL4_all_final_sns_norm_ns_syn_normex)[1]))


#try calling on the complete data set, using starting mu and sigma...with lambda = c(0.5,0.5) --> still gives same output parameters...
#XrL4_all_mm.out <- normalmixEM(x = XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, mean = ns_syn_normex_mm.opt$mu, sd = ns_syn_normex_mm.opt$sigma, lambda = c(0.5,0.5))

XrL4_all_final_sns_norm$posterior_ns <- XrL4_all_mm.out$posterior[,2]
XrL4_all_final_sns_norm$posterior_syn <- XrL4_all_mm.out$posterior[,1]
XrL4_all_final_sns_norm$posterior_OR <- XrL4_all_final_sns_norm$posterior_ns/XrL4_all_final_sns_norm$posterior_syn
XrL4_all_OR_cutoff <- mean(c(max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > .50),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < .50),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])))

#max value likely LOF
XrL4_all_llof_thresh <- max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > .95),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
#max value that has 99% chance to be LOF
XrL4_all_lof_thresh <- max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > .99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
#max value that is 10,000x more likely to be LOF
max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > .999),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
#min value that has <5% chance of being LOF
XrL4_all_lf_thresh <- min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < .05),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
#min value that has <1% chance of being LOF
XrL4_all_f01_thresh <- min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
XrL4_all_f_thresh <- min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.001),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])

#!!!! fix this call to plot the posterior probability of lof as a function of assay score and run two scripts below
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=posterior_ns))+geom_jitter(aes(colour=conseq),alpha=0.1,height=0.05)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('P(NS)')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=1)

#zooming in on inflection in OR
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=posterior_ns))+geom_jitter(aes(colour=conseq),alpha=0.3,height=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('P(NS)')+xlab('Functional score (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=1)+coord_cartesian(ylim=c(-0.12,1),xlim=c(-1.5,-0.5))

#comparing all 3 boundaries at 99% and 1% (mixtools;black, Mclust;blue) / or .05 FDR (red)
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=posterior_ns))+geom_jitter(aes(colour=conseq),alpha=0.3,height=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('P(NS)')+xlab('Functional score (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+coord_cartesian(ylim=c(-0.12,1),xlim=c(-1.5,-0.5))+geom_vline(xintercept=max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=1)+geom_vline(xintercept=XrL4_bh_p_binary_cutoff,lty=1,color='red')+geom_vline(xintercept=XrL4_bh_p_syn_max05,color='red',lty=2)+geom_vline(xintercept=XrL4_bh_p_ns_min05,color='red',lty=2)+geom_vline(xintercept=mclust_functional_thresh,lty=1,color='darkblue')+geom_vline(xintercept=mclust_f_01,color='darkblue',lty=2)+geom_vline(xintercept=mclust_lof_99,color='darkblue',lty=2)

ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=posterior_ns))+geom_jitter(aes(colour=conseq),alpha=0.3,height=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('P(NS)')+xlab('Functional score (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+coord_cartesian(ylim=c(-0.12,1),xlim=c(-1.5,-0.5))+geom_vline(xintercept=max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=1)+geom_vline(xintercept=XrL4_bh_p_binary_cutoff,lty=1,color='red')+geom_vline(xintercept=XrL4_bh_p_syn_max05,color='red',lty=2)+geom_vline(xintercept=XrL4_bh_p_ns_min05,color='red',lty=2)+geom_vline(xintercept=mclust_functional_thresh,lty=1,color='darkblue')+geom_vline(xintercept=mclust_f_01,color='darkblue',lty=2)+geom_vline(xintercept=mclust_lof_99,color='darkblue',lty=2)


#comparing thresholds to tHDR_post_pre ratios
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm)))+geom_jitter(aes(colour=conseq),alpha=0.5,height=0.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('log-2 post-pre')+xlab('Functional score (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+coord_cartesian(ylim=c(-2,1),xlim=c(-2,-0.0))+geom_vline(xintercept=max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=1)+geom_vline(xintercept=XrL4_bh_p_binary_cutoff,lty=1,color='red')+geom_vline(xintercept=XrL4_bh_p_syn_max05,color='red',lty=2)+geom_vline(xintercept=XrL4_bh_p_ns_min05,color='red',lty=2)+geom_vline(xintercept=mclust_functional_thresh,lty=1,color='darkblue')+geom_vline(xintercept=mclust_f_01,color='darkblue',lty=2)+geom_vline(xintercept=mclust_lof_99,color='darkblue',lty=2)

#comparing thresholds to tHDR_post_pre ratios (only for training data)
ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm)))+geom_jitter(aes(colour=conseq),alpha=0.5,height=0.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('log-2 post-pre')+xlab('Functional score (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+coord_cartesian(ylim=c(-2,1),xlim=c(-2,-0.0))+geom_vline(xintercept=max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=1)+geom_vline(xintercept=XrL4_bh_p_binary_cutoff,lty=1,color='red')+geom_vline(xintercept=XrL4_bh_p_syn_max05,color='red',lty=2)+geom_vline(xintercept=XrL4_bh_p_ns_min05,color='red',lty=2)+geom_vline(xintercept=mclust_functional_thresh,lty=1,color='darkblue')+geom_vline(xintercept=mclust_f_01,color='darkblue',lty=2)+geom_vline(xintercept=mclust_lof_99,color='darkblue',lty=2)

#color by ClinVar
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm)))+geom_jitter(aes(colour=clinvar_simple),alpha=1,height=0.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('log-2 post-pre')+xlab('Functional score (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+coord_cartesian(ylim=c(-2,1),xlim=c(-2,-0.0))+geom_vline(xintercept=max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=1)+geom_vline(xintercept=XrL4_bh_p_binary_cutoff,lty=1,color='red')+geom_vline(xintercept=XrL4_bh_p_syn_max05,color='red',lty=2)+geom_vline(xintercept=XrL4_bh_p_ns_min05,color='red',lty=2)+geom_vline(xintercept=mclust_functional_thresh,lty=1,color='darkblue')+geom_vline(xintercept=mclust_f_01,color='darkblue',lty=2)+geom_vline(xintercept=mclust_lof_99,color='darkblue',lty=2)

ggplot(clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_llof_thresh,lty=3)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_lf_thresh,lty=3)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=1)


#bimodal_compare and conseq_all plots with normalmixEM posterior set.
ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=conseq))+geom_histogram(aes(fill=conseq),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+xlab('Functional score (e.sns.mean)')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1.2))+geom_vline(xintercept=XrL4_all_llof_thresh,lty=3)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_lf_thresh,lty=3)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f_thresh,lty=5)
ggplot(conseq_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_rL41rL42_sns_norm, fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Functional score (e.sns.mean)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1.2))+geom_vline(xintercept=XrL4_all_llof_thresh,lty=3)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_lf_thresh,lty=3)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
##drawing lines on the ClinVar plots
ggplot(clinvar_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('SNVs')+xlab('Enrichment (Log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_llof_thresh,lty=3)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_lf_thresh,lty=3)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#replicate 1 vs. 2 with lines showing replicate concordance (gray) and similarity boundaries
ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, color=conseq,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('function rL42 e.sns')+xlab('function rL41 e.sns')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+ geom_abline(intercept=2*XrL4_all_llof_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lf_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_OR_cutoff,lty=1,slope = -1,color="darkred") +geom_vline(xintercept=XrL4_all_OR_cutoff,lty=2,color="gray")+geom_hline(yintercept=XrL4_all_OR_cutoff,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope = 1,color="gray") + geom_abline(intercept=1,lty=3,slope = 1,color="gray")+ geom_abline(intercept=-2,lty=3,slope = 1,color="gray") + geom_abline(intercept=-1,lty=3,slope = 1,color="gray")


#replicate 1 vs. 2 with lines!
ggplot(conseq_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, color=conseq,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(alpha=0.4)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('function rL42 e.sns')+xlab('function rL41 e.sns')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_llof_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lf_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_OR_cutoff,lty=1,slope = -1,color="darkred") +geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray")+geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray")+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray")+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope = 1,color="gray") + geom_abline(intercept=1,lty=3,slope = 1,color="gray")+ geom_abline(intercept=-2,lty=3,slope = 1,color="gray") + geom_abline(intercept=-1,lty=3,slope = 1,color="gray")

ggplot(clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, color=clinvar_simple,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('function rL42 e.sns')+xlab('function rL41 e.sns')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_llof_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lf_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_OR_cutoff,lty=1,slope = -1,color="darkred") + geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope=1,color="gray") + geom_abline(intercept=1,lty=3,slope=1,color="gray") + geom_abline(intercept=-2,lty=3,slope = 1,color="gray") + geom_abline(intercept=-1,lty=3,slope = 1,color="gray")

ggplot(clinvar_ord_XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, color=clinvar_simple,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('function rL42 e.sns')+xlab('function rL41 e.sns')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_llof_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lf_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_OR_cutoff,lty=1,slope = -1,color="darkred") +geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray")+geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray")+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray")+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope = 1,color="gray") + geom_abline(intercept=1,lty=3,slope = 1,color="gray")+ geom_abline(intercept=-2,lty=3,slope = 1,color="gray") + geom_abline(intercept=-1,lty=3,slope = 1,color="gray")

ggplot(clinvar_ord_XrL4_all_final_sns_norm_clinvar_p_b_lp_lb, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, color=clinvar_simple,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('function rL42 e.sns')+xlab('function rL41 e.sns')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_llof_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lf_thresh,lty=3,slope = -1) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_OR_cutoff,lty=1,slope = -1,color="darkred") +geom_vline(xintercept=XrL4_all_OR_cutoff,lty=2,color="gray")+geom_hline(yintercept=XrL4_all_OR_cutoff,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope = 1,color="gray") + geom_abline(intercept=1,lty=3,slope = 1,color="gray")+ geom_abline(intercept=-2,lty=3,slope = 1,color="gray") + geom_abline(intercept=-1,lty=3,slope = 1,color="gray")

### ASSIGN CATEGORIES TO ALL VARIANTS BASED ON THE POSTERIOR OF NS FROM mixtools MM:
XrL4_all_final_sns_norm$XrL4_mmfunc_class <- 'NA'
assign_mmfunc <- function(posterior_ns){
  if (posterior_ns > 0.99) return('LOF') else if (posterior_ns < 0.01) return('FUNC') else return('INT')
}
XrL4_all_final_sns_norm$XrL4_mmfunc_class <- sapply(XrL4_all_final_sns_norm$posterior_ns, assign_mmfunc)

#assigned functional classes to all variants in database (unfiltered)

### add in the real transcript annotation and amino acid changes (subtract 63 nt from CADD, 21 AA if in exon 15-23...)
assign_clinvar_positions <- function(cDNApos,CDSpos,protPos,exon){
  if (exon == 'X2' | exon == 'X3' | exon == 'X4' | exon == 'X5' | is.na(cDNApos)) return(c(as.numeric(as.character(cDNApos)),as.numeric(as.character(CDSpos)),as.numeric(as.character(protPos)))) else return(c(as.numeric(as.character(cDNApos))-63,as.numeric(as.character(CDSpos))-63,as.numeric(as.character(protPos))-21))
}
clinvar_pos_values = data.frame()
clinvar_pos_values <- mapply(assign_clinvar_positions,XrL4_all_final_sns_norm$cDNApos,XrL4_all_final_sns_norm$CDSpos,XrL4_all_final_sns_norm$protPos,XrL4_all_final_sns_norm$exon)
XrL4_all_final_sns_norm$clinvar_cDNApos <- clinvar_pos_values[1,]
XrL4_all_final_sns_norm$clinvar_CDSpos <- clinvar_pos_values[2,]
XrL4_all_final_sns_norm$clinvar_protPos <- clinvar_pos_values[3,]

XrL4_all_final_sns_norm$pHGVS <- paste(paste('p.',XrL4_all_final_sns_norm$oAA,sep=''),paste(XrL4_all_final_sns_norm$clinvar_protPos,XrL4_all_final_sns_norm$nAA,sep=''),sep='')

# define this like above, but need the rev_comp and the alt XrL4_all_final_sns_norm$cHGVS <- 

#which points are in aGVGD:
aGVGD.df <- read.csv('/Users/Greg/Documents/Align_GVGD_scores_GF.csv',header = TRUE)
protPos_alt_in_aGVGD <- aGVGD.df$protPos_nAA

XrL4_all_final_sns_norm$clinvar_protPos_nAA <- paste(XrL4_all_final_sns_norm$clinvar_protPos,XrL4_all_final_sns_norm$nAA,sep='')

#two functions, one for assigning aGVGD_class (i.e. 'C65') and one for the numeric diff (GV-GD)
assign_aGVGD_class <- function(protPos_nAA){
  if (protPos_nAA %in% protPos_alt_in_aGVGD) return(as.numeric(as.character(aGVGD.df[which(aGVGD.df$protPos_nAA == protPos_nAA),]$prediction_number))) else return(NA)
}
XrL4_all_final_sns_norm$aGVGD_class <- sapply(XrL4_all_final_sns_norm$clinvar_protPos_nAA,assign_aGVGD_class)

assign_aGVGD_GV.GD <- function(protPos_nAA){
  if (protPos_nAA %in% protPos_alt_in_aGVGD) return(aGVGD.df[which(aGVGD.df$protPos_nAA == protPos_nAA),c('GV.GD')]) else return(NA)
}
XrL4_all_final_sns_norm$GV.GD <- sapply(XrL4_all_final_sns_norm$clinvar_protPos_nAA,assign_aGVGD_GV.GD)


#which points are in flossies?
flossies.df <- read.csv('/Users/Greg/Documents/flossies_download_20171009_w_pos_alt.csv',header = TRUE)
variants_in_flossies <- flossies.df$pos_alt
assign_flossie_score <- function(snv_pos_alt){
  if (snv_pos_alt %in% variants_in_flossies) return(flossies.df[which(flossies.df$pos_alt == snv_pos_alt),c('Overall.Frequency')]) else return(0)
}
XrL4_all_final_sns_norm$flossies_score <- sapply(XrL4_all_final_sns_norm$pos_alt,assign_flossie_score)

#which points are in BRAVO?
#note -- used BRAVO Freeze 5, and converted to hg19 coordinates using https://genome.ucsc.edu/cgi-bin/hgLiftOver
bravo.df <- read.csv('/Users/Greg/Documents/BRAVO_BRCA1_ENSG00000012048_11-19-2017.csv',header = TRUE)
variants_in_bravo <- bravo.df$hg19_pos_alt
variants_in_exac <- XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$ExAC.ObsAlleles != 'NA'),]$pos_alt
variants_in_clinvar <- XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$clinvar_simple != 'absent'),]$pos_alt

assign_bravo_AF <- function(snv_pos_alt){
  if (snv_pos_alt %in% variants_in_bravo) return(bravo.df[which(bravo.df$hg19_pos_alt == snv_pos_alt),c('allele_freq')]) else return('NA')
}
assign_bravo_count <- function(snv_pos_alt){
  if (snv_pos_alt %in% variants_in_bravo) return(bravo.df[which(bravo.df$hg19_pos_alt == snv_pos_alt),c('allele_count')]) else return('NA')
}
XrL4_all_final_sns_norm$bravo_AF <- sapply(XrL4_all_final_sns_norm$pos_alt,assign_bravo_AF)
XrL4_all_final_sns_norm$bravo_count <- sapply(XrL4_all_final_sns_norm$pos_alt,assign_bravo_count)

##re-do bravo scripts for gnomAD (12/26/2017 download):
#which points are in gnomAD?
#note -- should be in hg19. NHLBI TopMed contributed exomes to gnomAD
gnomad.df <- read.csv('/Users/Greg/Documents/gnomad_ENSG00000012048_2017_12_26_GF.csv',header = TRUE)
variants_in_gnomad <- gnomad.df$pos_alt

assign_gnomad_AF <- function(snv_pos_alt){
  if (snv_pos_alt %in% variants_in_gnomad) return(gnomad.df[which(gnomad.df$pos_alt == snv_pos_alt),c('Allele.Frequency')]) else return('NA')
}
assign_gnomad_count <- function(snv_pos_alt){
  if (snv_pos_alt %in% variants_in_gnomad) return(gnomad.df[which(gnomad.df$pos_alt == snv_pos_alt),c('Allele.Count')]) else return('NA')
}
assign_gnomad_number <- function(snv_pos_alt){
  if (snv_pos_alt %in% variants_in_gnomad) return(gnomad.df[which(gnomad.df$pos_alt == snv_pos_alt),c('Allele.Number')]) else return('NA')
}

XrL4_all_final_sns_norm$gnomad_AF <- sapply(XrL4_all_final_sns_norm$pos_alt,assign_gnomad_AF)
XrL4_all_final_sns_norm$gnomad_count <- sapply(XrL4_all_final_sns_norm$pos_alt,assign_gnomad_count)
XrL4_all_final_sns_norm$gnomad_number <- sapply(XrL4_all_final_sns_norm$pos_alt,assign_gnomad_number)

## filter now, and re-run all scripts, then save db (might want to do python for pooling)
#3958 total
dim(XrL4_all_final_sns_norm)[1]
#3923 with a difference in scores between replicates of less than 4-fold
dim(XrL4_all_final_sns_norm[which(abs(XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm-XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm)<2),])

#define filters
#most inclusive -->  just ensures variant is not classified oppositely across replicates -- 3798
XrL4_not_opposite_filter <- which(XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm >= XrL4_all_f01_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm > XrL4_all_lof_thresh  | XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm < XrL4_all_f01_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm <= XrL4_all_lof_thresh | XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm >= XrL4_all_f01_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm > XrL4_all_lof_thresh  | XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm < XrL4_all_f01_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm <= XrL4_all_lof_thresh  )

dim(XrL4_all_final_sns_norm[XrL4_not_opposite_filter,])

#requires both replicates to point to either functional / not -- 3659
XrL4_concordance_filter <- which( XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm> XrL4_all_OR_cutoff &  XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm> XrL4_all_OR_cutoff | XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm< XrL4_all_OR_cutoff &  XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm< XrL4_all_OR_cutoff)

#requires both replicates to have same classification -- 3307
XrL4_same_class_filter <- which((XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm >= XrL4_all_f01_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm >= XrL4_all_f01_thresh)  | (XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm <= XrL4_all_lof_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm <= XrL4_all_lof_thresh))

#requires both replicates to have same classification, incl. int.
XrL4_same_class_w_int_filter <- which((XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm >= XrL4_all_f01_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm >= XrL4_all_f01_thresh)  | (XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm <= XrL4_all_lof_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm <= XrL4_all_lof_thresh) | ((XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm < XrL4_all_f01_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm > XrL4_all_lof_thresh) & (XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm < XrL4_all_f01_thresh & XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm > XrL4_all_lof_thresh)))


XrL4_similarity_filter_1 <- which(abs(XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm -  XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm) < 1)
XrL4_similarity_filter_2 <- which(abs(XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL41_sns_norm -  XrL4_all_final_sns_norm$tHDR_post_lib_loess_e.sns_rL42_sns_norm) < 2)

XrL4_all_final_sns_norm_conc <- XrL4_all_final_sns_norm[union(XrL4_not_opposite_filter,XrL4_similarity_filter_2),]
#take all points that are either not opposite or whose scores differ by less than 2 - 3944, and that are not splice altering due to PAM
#list of variants that may have splice difference in context of PAM edit:
#ones with RNA evidence of depletion in edited PAM context (must go)
#these are filtered out later with SNVs not in CADD (i.e. not true SNVs)
exclude_pam_splice_low_rna <- c('41256962A','41199702G')
exclude_pam_splice_no_test_GGT_donor <- c('41258513A','41215945A','41203122A','41203067A','41201127A')
exclude_pam_splice_no_test_HGT_donor <- c('41219675C','41209125A','41199702A','41197790A','41197769A')
exclude_pam_splice_low_rna_or_no_test_GGT <- c('41256962A','41199702G','41258513A','41215945A','41203122A','41203067A','41201127A')
exclude_pam_splice_low_rna_or_no_test_NGT <- c('41256962A','41199702G','41258513A','41215945A','41203122A','41203067A','41201127A','41219675C','41209125A','41199702A','41197790A','41197769A')

#exclude_pam_splice_low_rna_or_no_test_NGT splicing positions to be flagged.
XrL4_all_final_sns_norm_filt <- XrL4_all_final_sns_norm_conc[! XrL4_all_final_sns_norm_conc$pos_alt %in% exclude_pam_splice_low_rna_or_no_test_NGT,]

#SNVs in CADD...
XrL4_all_final_sns_norm_filt_CADD <- XrL4_all_final_sns_norm_filt[XrL4_all_final_sns_norm_filt$pos_alt %in% XrL4_variants_with_CADD,]

XrL4_variants_in_exon <- XrL4_all_tHDR_pos_merge_master_df[which(is.na(XrL4_all_tHDR_pos_merge_master_df$cDNApos) == FALSE),c('pos_alt')]
XrL4_all_final_sns_norm_filt_EXON <- XrL4_all_final_sns_norm_filt[XrL4_all_final_sns_norm_filt$pos_alt %in% XrL4_variants_in_exon,]

XrL4_variants_in_CDS <- XrL4_all_tHDR_pos_merge_master_df[which(is.na(XrL4_all_tHDR_pos_merge_master_df$CDSpos) == FALSE),c('pos_alt')]
XrL4_all_final_sns_norm_filt_CDS <- XrL4_all_final_sns_norm[XrL4_all_final_sns_norm_filt$pos_alt %in% XrL4_variants_in_CDS,]

#call variants on inclusive list (XrL4_all_tHDR_pos_merge_master_df), then select subset of filtered list for now...
XrL4_variants_with_RNA <- XrL4_all_tHDR_pos_merge_master_df[which(is.na(XrL4_all_tHDR_pos_merge_master_df$cDNApos) == FALSE & XrL4_all_tHDR_pos_merge_master_df$exon != 'X18'),c('pos_alt')]
XrL4_all_final_sns_norm_filt_EXON_no18 <- XrL4_all_final_sns_norm_filt[XrL4_all_final_sns_norm_filt$pos_alt %in% XrL4_variants_with_RNA,]
XrL4_variants_with_RNA_no_ns <- XrL4_all_tHDR_pos_merge_master_df[which(is.na(XrL4_all_tHDR_pos_merge_master_df$cDNApos) == FALSE & XrL4_all_tHDR_pos_merge_master_df$exon != 'X18' & XrL4_all_tHDR_pos_merge_master_df$exon != 'X18'),c('pos_alt')]

XrL4_all_final_sns_norm_filt_EXON_CADD <- XrL4_all_final_sns_norm_filt_EXON[XrL4_all_final_sns_norm_filt_EXON$pos_alt %in% XrL4_variants_with_CADD,]
### XrL4_variants_with_RNA (no X18)
XrL4_all_final_sns_norm_filt_EXON_no18 <- XrL4_all_final_sns_norm_filt[XrL4_all_final_sns_norm_filt$pos_alt %in% XrL4_variants_with_RNA,]
XrL4_all_final_sns_norm_filt_CADD_EXON_no18 <- XrL4_all_final_sns_norm_filt_EXON_no18[XrL4_all_final_sns_norm_filt_EXON_no18$pos_alt %in% XrL4_variants_with_RNA,]

#filter complete df on CADD:
XrL4_all_tHDR_pos_merge_master_df_CADD <- XrL4_all_tHDR_pos_merge_master_df[XrL4_all_tHDR_pos_merge_master_df$pos_alt %in% XrL4_variants_with_CADD,]

#ClinVar path / benign
XrL4_variants_P.B <- XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Pathogenic' | XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Benign'),c('pos_alt')]
XrL4_variants_P.B.LP.LB <- XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Pathogenic' | XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Likely benign' | XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Likely pathogenic' | XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Benign'),c('pos_alt')]
XrL4_all_final_sns_norm_filt_CADD_P.B <- XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% XrL4_variants_P.B,]
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB <- XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% XrL4_variants_P.B.LP.LB,]

XrL4_all_final_sns_norm_filt_CADD_bravo

XrL4_ns_syn_normex_variants <- append(XrL4_all_final_sns_norm_syn_norm_exp$pos_alt,XrL4_all_final_sns_norm_ns$pos_alt)
XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex <- XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% XrL4_ns_syn_normex_variants,]
#for syn / ns roc
#for all roc's -- define cuts based on every possible point in data set...
#editing here -- fixed the syn/ns call set... now re-run code below and see if it works...

XrL4_roc_cuts <- append(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm-0.00001,max(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)+0.00001)

XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_sens <- c()
XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_fpr <- c()
for (roc_cut in XrL4_roc_cuts){
  cut_off_sens <- length(which(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < roc_cut & XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex$conseq == 'STOP_GAINED'))/length(which(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex$conseq == 'STOP_GAINED'))
  cut_off_fpr <- length(which(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < roc_cut & XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex$conseq != 'STOP_GAINED'))/length(which(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex$conseq != 'STOP_GAINED'))
  XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_sens <- append(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_sens,cut_off_sens)
  XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_fpr <- append(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_fpr,cut_off_fpr)
}

XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc <- data.frame(cbind(XrL4_roc_cuts,XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_fpr,XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_sens))

#get points where spec and sens. is optimally maximized
XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc_youden <-XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc[which(1-XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc$XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_fpr+XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc$XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_sens == max(1-XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc$XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_fpr+XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc$XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_sens)),]
#median of optimal cutoffs:
median(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc_youden$XrL4_roc_cuts)

#ROC curve for nonsense vs. synonymous variants.
ggplot(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_roc)+geom_point(aes(x=XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_fpr,y=XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_sens))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Sensitivity (nonsense)')+xlab('False pos. rate (syn.)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=0,lty=2,slope = 1)

#trying this again with geom_roc: #using negative here...
XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_geom_roc <- XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex
XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_geom_roc$D_ns_syn_normex <- 1
assign_ns_syn_normex <- function(conseq){
  if (conseq == 'STOP_GAINED') return(1) else return(0)}
XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_geom_roc$D_ns_syn_normex <- sapply(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_geom_roc$conseq,assign_ns_syn_normex)
ns_syn_normex_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_ns_syn_normex_geom_roc, aes(d = D_ns_syn_normex, m = -tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)) +geom_roc(n.cuts=20) + style_roc(xlab = "1 - Specificity")
#with AUC:
ns_syn_normex_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(ns_syn_normex_roc)$AUC, 3)))


#roc analysis for firm path and benign:
XrL4_all_final_sns_norm_filt_CADD_P.B_sens <- c()
XrL4_all_final_sns_norm_filt_CADD_P.B_fpr <- c()
for (roc_cut in XrL4_roc_cuts){
  cut_off_sens <- length(which(XrL4_all_final_sns_norm_filt_CADD_P.B$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < roc_cut & XrL4_all_final_sns_norm_filt_CADD_P.B$clinvar_simple == 'Pathogenic'))/length(which(XrL4_all_final_sns_norm_filt_CADD_P.B$clinvar_simple == 'Pathogenic'))
  cut_off_fpr <- length(which(XrL4_all_final_sns_norm_filt_CADD_P.B$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < roc_cut & XrL4_all_final_sns_norm_filt_CADD_P.B$clinvar_simple != 'Pathogenic'))/length(which(XrL4_all_final_sns_norm_filt_CADD_P.B$clinvar_simple != 'Pathogenic'))
  XrL4_all_final_sns_norm_filt_CADD_P.B_sens <- append(XrL4_all_final_sns_norm_filt_CADD_P.B_sens,cut_off_sens)
  XrL4_all_final_sns_norm_filt_CADD_P.B_fpr <- append(XrL4_all_final_sns_norm_filt_CADD_P.B_fpr,cut_off_fpr)
}

XrL4_all_final_sns_norm_filt_CADD_P.B_roc <- data.frame(cbind(XrL4_roc_cuts,XrL4_all_final_sns_norm_filt_CADD_P.B_fpr,XrL4_all_final_sns_norm_filt_CADD_P.B_sens))

#get points where spec and sens. is optimally maximized
XrL4_all_final_sns_norm_filt_CADD_P.B_roc_youden <-XrL4_all_final_sns_norm_filt_CADD_P.B_roc[which(1-XrL4_all_final_sns_norm_filt_CADD_P.B_roc$XrL4_all_final_sns_norm_filt_CADD_P.B_fpr+XrL4_all_final_sns_norm_filt_CADD_P.B_roc$XrL4_all_final_sns_norm_filt_CADD_P.B_sens == max(1-XrL4_all_final_sns_norm_filt_CADD_P.B_roc$XrL4_all_final_sns_norm_filt_CADD_P.B_fpr+XrL4_all_final_sns_norm_filt_CADD_P.B_roc$XrL4_all_final_sns_norm_filt_CADD_P.B_sens)),]
#median of optimal cutoffs:
median(XrL4_all_final_sns_norm_filt_CADD_P.B_roc_youden$XrL4_roc_cuts)

#ROC curve for Pathogenic vs. Benign variants.
ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B_roc)+geom_point(aes(x=XrL4_all_final_sns_norm_filt_CADD_P.B_fpr,y=XrL4_all_final_sns_norm_filt_CADD_P.B_sens))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Sensitivity (nonsense)')+xlab('False pos. rate (syn.)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=0,lty=2,slope = 1)

#trying this again with geom_roc:
XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc <- XrL4_all_final_sns_norm_filt_CADD_P.B
XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc$D_P.B <- 1
assign_P.B <- function(clinvar_simple){
  if (clinvar_simple == 'Pathogenic') return(1) else return(0)}
XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc$D_P.B <- sapply(XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc$clinvar_simple,assign_P.B)
P.B_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc, aes(d = D_P.B, m = -tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)) +geom_roc() + style_roc(xlab = "1 - Specificity")
#with AUC:
P.B_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_roc)$AUC, 3)))

#roc analysis for firm path and benign, missense only:
XrL4_all_final_sns_norm_filt_CADD_P.B.mis <- XrL4_all_final_sns_norm_filt_CADD_P.B[which(XrL4_all_final_sns_norm_filt_CADD_P.B$conseq == 'NON_SYNONYMOUS'),]
XrL4_all_final_sns_norm_filt_CADD_P.B.mis_sens <- c()
XrL4_all_final_sns_norm_filt_CADD_P.B.mis_fpr <- c()
for (roc_cut in XrL4_roc_cuts){
  cut_off_sens <- length(which(XrL4_all_final_sns_norm_filt_CADD_P.B.mis$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < roc_cut & XrL4_all_final_sns_norm_filt_CADD_P.B.mis$clinvar_simple == 'Pathogenic'))/length(which(XrL4_all_final_sns_norm_filt_CADD_P.B.mis$clinvar_simple == 'Pathogenic'))
  cut_off_fpr <- length(which(XrL4_all_final_sns_norm_filt_CADD_P.B.mis$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < roc_cut & XrL4_all_final_sns_norm_filt_CADD_P.B.mis$clinvar_simple != 'Pathogenic'))/length(which(XrL4_all_final_sns_norm_filt_CADD_P.B.mis$clinvar_simple != 'Pathogenic'))
  XrL4_all_final_sns_norm_filt_CADD_P.B.mis_sens <- append(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_sens,cut_off_sens)
  XrL4_all_final_sns_norm_filt_CADD_P.B.mis_fpr <- append(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_fpr,cut_off_fpr)
}

XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc <- data.frame(cbind(XrL4_roc_cuts,XrL4_all_final_sns_norm_filt_CADD_P.B.mis_fpr,XrL4_all_final_sns_norm_filt_CADD_P.B.mis_sens))

#get points where spec and sens. is optimally maximized
XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc_youden <-XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc[which(1-XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc$XrL4_all_final_sns_norm_filt_CADD_P.B.mis_fpr+XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc$XrL4_all_final_sns_norm_filt_CADD_P.B.mis_sens == max(1-XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc$XrL4_all_final_sns_norm_filt_CADD_P.B.mis_fpr+XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc$XrL4_all_final_sns_norm_filt_CADD_P.B.mis_sens)),]
#median of optimal cutoffs:
median(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc_youden$XrL4_roc_cuts)

#ROC curve for Pathogenic vs. Benign missense variants.
ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_roc)+geom_point(aes(x=XrL4_all_final_sns_norm_filt_CADD_P.B.mis_fpr,y=XrL4_all_final_sns_norm_filt_CADD_P.B.mis_sens))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Sensitivity (nonsense)')+xlab('False pos. rate (syn.)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=0,lty=2,slope = 1)

#trying this again with geom_roc:
XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc <- XrL4_all_final_sns_norm_filt_CADD_P.B.mis
XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc$D_P.B.mis <- 1
assign_P.B.mis <- function(clinvar_simple){
  if (clinvar_simple == 'Pathogenic') return(1) else return(0)}
XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc$D_P.B.mis <- sapply(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc$clinvar_simple,assign_P.B.mis)

P.B.mis_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc, aes(d = D_P.B.mis, m = -tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)) +geom_roc(n.cuts=0, labels=FALSE) + style_roc(xlab = "1 - Specificity")
#with AUC:
P.B.mis_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.mis_roc)$AUC, 3)))

#what about CADD, though?
P.B.mis_cadd_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc, aes(d = D_P.B.mis, m = as.numeric(as.character(CADD.phred)))) + geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity")
P.B.mis_cadd_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.mis_cadd_roc)$AUC, 3)))

#what about grantham, though?
P.B.mis_grantham_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc, aes(d = D_P.B.mis, m = as.numeric(as.character(Grantham)))) + geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity")
P.B.mis_grantham_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.mis_grantham_roc)$AUC, 3)))

#what about aGVGD classes, though?
P.B.mis_aGVGDc_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc, aes(d = D_P.B.mis, m = as.numeric(as.character(aGVGD_class)))) + geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity")
P.B.mis_aGVGDc_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.mis_aGVGDc_roc)$AUC, 3)))

#what about aGVGD differences, though?
P.B.mis_GV.GD_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.mis_geom_roc, aes(d = D_P.B.mis, m = -as.numeric(as.character(GV.GD)))) + geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity") 
P.B.mis_GV.GD_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.mis_GV.GD_roc)$AUC, 3)))

#missense comparison for SGE vs aGVGD vs CADD
grid.arrange(P.B.mis_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.mis_roc)$AUC, 3))),P.B.mis_cadd_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.mis_cadd_roc)$AUC, 3))),P.B.mis_GV.GD_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.mis_GV.GD_roc)$AUC, 3))),nrow=1)

#roc analysis for path and benign incl. 'likely':
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_sens <- c()
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_fpr <- c()
for (roc_cut in XrL4_roc_cuts){
  cut_off_sens <- length(which(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < roc_cut & (XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$clinvar_simple == 'Pathogenic' | XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$clinvar_simple == 'Likely pathogenic' )))/length(which((XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$clinvar_simple == 'Pathogenic' | XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$clinvar_simple == 'Likely pathogenic' )))
  cut_off_fpr <- length(which(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < roc_cut & (XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$clinvar_simple != 'Pathogenic' & XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$clinvar_simple != 'Likely pathogenic' )))/length(which((XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$clinvar_simple != 'Pathogenic' & XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB$clinvar_simple != 'Likely pathogenic' )))
  XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_sens <- append(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_sens,cut_off_sens)
  XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_fpr <- append(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_fpr,cut_off_fpr)
}

XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc <- data.frame(cbind(XrL4_roc_cuts,XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_fpr,XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_sens))

#get points where spec and sens. is optimally maximized
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc_youden <-XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc[which(1-XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc$XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_fpr+XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc$XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_sens == max(1-XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc$XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_fpr+XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc$XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_sens)),]
#median of optimal cutoffs:
median(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc_youden$XrL4_roc_cuts)

#ROC curve for pathogenic vs. benign variants, including likely.
ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc)+geom_point(aes(x=XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_fpr,y=XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_sens))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Sensitivity (nonsense)')+xlab('False pos. rate (syn.)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=0,lty=2,slope = 1)
ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_roc)+geom_line(aes(x=XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_fpr,y=XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_sens))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Sensitivity (Path. / Likely path.)')+xlab('False pos. (Ben. / Likely benign)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=0,lty=2,slope = 1)

#trying this again with geom_roc:
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc <- XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc$D_P.B.LP.LB <- 1
assign_P.B.LP.LB <- function(clinvar_simple){
  if (clinvar_simple == 'Pathogenic' | clinvar_simple == 'Likely pathogenic') return(1) else return(0)}
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc$D_P.B.LP.LB <- sapply(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc$clinvar_simple,assign_P.B.LP.LB)
P.B.LP.LB_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc, aes(d = D_P.B.LP.LB, m = -tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity")
#with AUC:
P.B.LP.LB_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_roc)$AUC, 3)))
#what about CADD, though?
P.B.LP.LB_cadd_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(CADD.phred)))) + geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity")
P.B.LP.LB_cadd_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_cadd_roc)$AUC, 3)))


#Global plots for for replicate discordancies
#plotting all points that are either concordant (in direction) or within (4-fold)
#how filtering scheme looks
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(aes(color=conseq),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('rep. 2 function')+xlab('rep. 1 function')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope=1,color="gray")  + geom_abline(intercept=-2,lty=3,slope = 1,color="gray")

#facet above plot to exon level, specifying order of exons and placing into 4 columns
XrL4_all_final_sns_norm_filt_CADD_exon_ordered <- XrL4_all_final_sns_norm_filt_CADD
XrL4_all_final_sns_norm_filt_CADD_exon_ordered$exon <- factor(XrL4_all_final_sns_norm_filt_CADD_exon_ordered$exon, levels =  c('X2','X3','X4','X5','X15','X16','X17','X18','X19','X20','X21','X22','X23'))

#candidate figure for supplemental showing exons, both replicates...
ggplot(XrL4_all_final_sns_norm_filt_CADD_exon_ordered, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(aes(color=conseq),alpha=0.5,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('rep. 2 function')+xlab('rep. 1 function')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope=1,color="gray")  + geom_abline(intercept=-2,lty=3,slope = 1,color="gray") + facet_wrap( ~ exon, ncol=4)

#performance on control sets:
ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(aes(color=conseq),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('rep. 2 function')+xlab('rep. 1 function')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope=1,color="gray")  + geom_abline(intercept=-2,lty=3,slope = 1,color="gray")

#facet above control plots over 'exon' (not filtered as above is)
XrL4_all_final_sns_norm_ns_syn_normex_exon_ordered <- XrL4_all_final_sns_norm_ns_syn_normex
XrL4_all_final_sns_norm_ns_syn_normex_exon_ordered$exon <- factor(XrL4_all_final_sns_norm_ns_syn_normex_exon_ordered$exon, levels =  c('X2','X3','X4','X5','X15','X16','X17','X18','X19','X20','X21','X22','X23'))
ggplot(XrL4_all_final_sns_norm_ns_syn_normex_exon_ordered, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(aes(color=conseq),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('rep. 2 function')+xlab('rep. 1 function')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope=1,color="gray")  + geom_abline(intercept=-2,lty=3,slope = 1,color="gray") + facet_wrap( ~ exon, ncol=4)

#performance on clinvar (total):
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent'),], aes(color=clinvar),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple != 'absent'),], aes(color=clinvar),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign'),], aes(color=clinvar),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('rep. 2 function')+xlab('rep. 1 function')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope=1,color="gray")  + geom_abline(intercept=-2,lty=3,slope = 1,color="gray")

#performance on clinvar (same as above but faceting on conseq):
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent'),], aes(color=clinvar),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple != 'absent'),], aes(color=clinvar),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign'),], aes(color=clinvar),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('rep. 2 function')+xlab('rep. 1 function')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="gray") + geom_abline(intercept=2,lty=3,slope=1,color="gray")  + geom_abline(intercept=-2,lty=3,slope = 1,color="gray") + facet_wrap( ~ conseq, ncol=3)

## downstream analysis...

#3905 XrL4_variants total
length(which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'))

#counts of each by conseq
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',]$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',]$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',]$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',]$XrL4_mmfunc_class == 'LOF'))

length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq != 'NON_SYNONYMOUS',]$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq != 'NON_SYNONYMOUS',]$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq != 'NON_SYNONYMOUS',]$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq != 'NON_SYNONYMOUS',]$XrL4_mmfunc_class == 'LOF'))

length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS',]$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS',]$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS',]$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS',]$XrL4_mmfunc_class == 'LOF'))

length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE',]$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE',]$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE',]$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE',]$XrL4_mmfunc_class == 'LOF'))

length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE',]$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE',]$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE',]$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE',]$XrL4_mmfunc_class == 'LOF'))

length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'STOP_GAINED',]$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'STOP_GAINED',]$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'STOP_GAINED',]$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'STOP_GAINED',]$XrL4_mmfunc_class == 'LOF'))

length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR',]$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR',]$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR',]$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR',]$XrL4_mmfunc_class == 'LOF'))

length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC',]$XrL4_mmfunc_class != 'NA'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC',]$XrL4_mmfunc_class == 'INT'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC',]$XrL4_mmfunc_class == 'FUNC'))
length(which(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC',]$XrL4_mmfunc_class == 'LOF'))

### FACET OVER CONSEQUENCE AND COLOR BY CLINVAR (plotting absent on bottom)
ggplot(XrL4_all_final_sns_norm_filt_EXON_no18, aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), fill=clinvar_simple))+geom_point(data=XrL4_all_final_sns_norm_filt_EXON_no18[XrL4_all_final_sns_norm_filt_EXON_no18$clinvar_simple == 'absent',],aes(colour=clinvar_simple),alpha=0.6)+geom_point(data=XrL4_all_final_sns_norm_filt_EXON_no18[XrL4_all_final_sns_norm_filt_EXON_no18$clinvar_simple != 'absent',],aes(colour=clinvar_simple),alpha=0.6)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Functional score (e.sns)')+xlab('RNA expression')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray")+geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray")+geom_vline(xintercept=-2,lty=3,color="gray")+geom_vline(xintercept=log2(0.125),lty=3,color="gray")+ facet_wrap( ~ conseq, ncol=3)

#is expression down in intermediate category?
#make a geom_violin for this... doesn't really work.
ggplot(XrL4_all_final_sns_norm_filt_EXON_no18, aes(fill=conseq))+geom_violin(aes(x=XrL4_mmfunc_class, y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Functional category')+ylab('RNA expression')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#box plot
ggplot(XrL4_all_final_sns_norm_filt_EXON_no18, aes(fill=conseq))+geom_boxplot(aes(x=XrL4_mmfunc_class, y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Functional category')+ylab('RNA expression')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

#by exon, looking at only 'stop_gained' (NMD):
#box plot
exon_ord_XrL4_all_final_sns_norm_filt_EXON_no18_ns <- XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'STOP_GAINED'),]
exon_ord_XrL4_all_final_sns_norm_filt_EXON_no18_ns$exon <- factor(exon_ord_XrL4_all_final_sns_norm_filt_EXON_no18_ns$exon, levels = c('X2','X3','X4','X5','X15','X16','X17','X19','X20','X21','X22','X23'))

ggplot(exon_ord_XrL4_all_final_sns_norm_filt_EXON_no18_ns, aes(fill=conseq))+geom_boxplot(aes(x=exon, y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+xlab('Exon')+ylab('RNA expression')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'NON_SYNONYMOUS'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])
median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'NON_SYNONYMOUS'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])
median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'NON_SYNONYMOUS'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])
median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SYNONYMOUS'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])
median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SYNONYMOUS'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])
median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SYNONYMOUS'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])
median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SPLICE_SITE'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])
median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SPLICE_SITE'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])
median(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SPLICE_SITE'),c('tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')])


#looking at how expression levels predict LOF using OR cutoff
ggplot(XrL4_all_final_sns_norm_filt_EXON_no18, aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), fill=conseq))+geom_point(aes(colour=conseq),alpha=0.4)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Functional score (e.sns)')+xlab('D5 RNA expression (log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray")+geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray")+geom_vline(xintercept=-2,lty=3,color="gray")+geom_vline(xintercept=log2(0.125),lty=3,color="gray")
#redo with clinvar colors -- find way to put 'absent down first, then all others on top...
ggplot(XrL4_all_final_sns_norm_filt_EXON_no18, aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), fill=clinvar_simple))+geom_point(aes(colour=clinvar_simple),alpha=0.5,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Functional score (e.sns)')+xlab('RNA expression (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray")+geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray")+geom_vline(xintercept=-2,lty=3,color="gray")+geom_vline(xintercept=log2(0.125),lty=3,color="gray")

# CANDIDATE FIGURE:  same call but with subset first ('absent')
ggplot(XrL4_all_final_sns_norm_filt_EXON_no18, aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), fill=clinvar_simple))+geom_point(data = XrL4_all_final_sns_norm_syn_mis_spl_noX18[which(XrL4_all_final_sns_norm_syn_mis_spl_noX18$clinvar_simple == 'absent'),], aes(colour=clinvar_simple),alpha=0.5,size=1)+geom_point(data = XrL4_all_final_sns_norm_syn_mis_spl_noX18[which(XrL4_all_final_sns_norm_syn_mis_spl_noX18$clinvar_simple != 'absent'),], aes(colour=clinvar_simple),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Functional score (e.sns)')+xlab('RNA expression (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray")+geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray")+geom_vline(xintercept=-2,lty=3,color="gray")+geom_vline(xintercept=log2(0.125),lty=3,color="gray")

#how many VUS or Conflicting are re-assigned to classes vs. not:
length(which(XrL4_all_final_sns_norm$clinvar_simple == 'Uncertain significance'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'Uncertain significance'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'Uncertain significance'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'Uncertain significance'))

length(which(XrL4_all_final_sns_norm$clinvar_simple == 'Conflicting interpretations of pathogenicity'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'Conflicting interpretations of pathogenicity'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'Conflicting interpretations of pathogenicity'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'Conflicting interpretations of pathogenicity'))

length(which(XrL4_all_final_sns_norm$clinvar_simple == 'absent'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'absent'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'absent'))
length(which(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'absent'))

### consider heatmap of %LOF for a matrix of variables (clinvar category x consequence, for instance)
### correlation to predictive measurements: (change data frame reference and score...)
#CADD.raw 
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=as.numeric(as.character(CADD.raw)),colour=conseq))+geom_point(alpha=0.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('CADD score (raw)')+xlab('Enrichment (post-lib corr., log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#which points have an exac frequency over a given threshold? -- by score?
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(ExAC.AF))),colour=conseq))+geom_jitter(alpha=1,width=0, height = 0.2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('ExAC AF (log-10)')+xlab('Enrichment (post-lib corr., log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
#same as above but colored by ClinVar
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(ExAC.AF))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.1,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('ExAC AF (log-10)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
#make an exac database:
XrL4_all_final_sns_norm_filt_CADD_EXAC <- XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$ExAC.AF != 'NA'),]
XrL4_all_final_sns_norm_filt_CADD_EXAC_minAF <- XrL4_all_final_sns_norm_filt_CADD_EXAC[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_EXAC$ExAC.AF)) == min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_EXAC$ExAC.AF)))),]
XrL4_all_final_sns_norm_filt_CADD_EXAC_notminAF <- XrL4_all_final_sns_norm_filt_CADD_EXAC[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_EXAC$ExAC.AF)) != min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_EXAC$ExAC.AF)))),]
#here are the actual allele counts
XrL4_all_final_sns_norm_filt_CADD_EXAC_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_EXAC_notminAF$XrL4_mmfunc_class != 'LOF'),c('ExAC.ObsAlleles')]
XrL4_all_final_sns_norm_filt_CADD_EXAC_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_EXAC_notminAF$XrL4_mmfunc_class == 'LOF'),c('ExAC.ObsAlleles')]

#make a gnomad DB:
XrL4_all_final_sns_norm_filt_CADD_gnomad <- XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$gnomad_AF != 'NA'),]
XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF <- XrL4_all_final_sns_norm_filt_CADD_gnomad[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_gnomad$gnomad_AF)) <= 0.00001),]
XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF <- XrL4_all_final_sns_norm_filt_CADD_gnomad[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_gnomad$gnomad_AF)) > 0.00001),]
#here are the actual allele counts
XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF$XrL4_mmfunc_class != 'LOF'),c('gnomad_count')]
XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF$XrL4_mmfunc_class == 'LOF'),c('gnomad_count')]

#which points have an bravo frequency over a given threshold? -- by score?
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(gnomad_AF))),colour=conseq))+geom_jitter(alpha=1,width=0, height = 0.2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('gnomAD AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

###editing here to do bravo instead of gnomad:
#make an bravo database:
XrL4_all_final_sns_norm_filt_CADD_bravo <- XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$bravo_AF != 'NA'),]
XrL4_all_final_sns_norm_filt_CADD_bravo_minAF <- XrL4_all_final_sns_norm_filt_CADD_bravo[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_bravo$bravo_AF)) == min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_bravo$bravo_AF)))),]
XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF <- XrL4_all_final_sns_norm_filt_CADD_bravo[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_bravo$bravo_AF)) != min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_bravo$bravo_AF)))),]
#here are the actual allele counts
XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF$XrL4_mmfunc_class == 'LOF'),c('bravo_count')]
XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF$XrL4_mmfunc_class != 'LOF'),c('bravo_count')]


#which points have an bravo frequency over a given threshold? -- by score?
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(bravo_AF))),colour=conseq))+geom_jitter(alpha=1,width=0, height = 0.2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Bravo AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
#same as above but colored by ClinVar
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(bravo_AF))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.1,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Bravo AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#intersect variants in clinvar, bravo and gnomad!!
#length of in any of the three:
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad))
#clinvar + gnomad
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad))
#clinvar + bravo
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo))
#gnomad or bravo
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo))

#gnomad AND bravo
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo))

#exac + bravo
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_exac | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo))

XrL4_all_final_sns_norm_filt_CADD_cv_gn_br <- XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad),]
#this plot only shows variants in both databases
ggplot(XrL4_all_final_sns_norm_filt_CADD_cv_gn_br, aes(x=log2(as.numeric(as.character(gnomad_count))+1), y=log2(as.numeric(as.character(bravo_count))+1),color=clinvar_simple))+geom_jitter(alpha=1,width=0.2, height = 0.2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Bravo AF (log-10)')+xlab('ExAC AF (log-10)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

XrL4_all_final_sns_norm_filt_CADD_br_not_gn <- XrL4_all_final_sns_norm_filt_CADD[which( XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo & (XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad == FALSE)),]
ggplot(XrL4_all_final_sns_norm_filt_CADD_br_not_gn, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(bravo_AF))),color=conseq))+geom_jitter(alpha=1,width=0, height = 0.02,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Bravo AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
ggplot(XrL4_all_final_sns_norm_filt_CADD_br_not_gn, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(bravo_AF))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.01,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Bravo AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#clinvar variants unique to ClinVar: 403
XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br <- XrL4_all_final_sns_norm_filt_CADD[which( XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar & ((XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad == FALSE)&(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo == FALSE))),]

#clinvar variants NOT unique to Clinvar: 
XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br <- XrL4_all_final_sns_norm_filt_CADD[which( XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar & ((XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo))),]


#KEEP THIS ONE MAYBE?!
#clinvar variant scores (not seen in other db's):
ggplot(XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, y=tHDR_post_lib_loess_e.sns_rL42_sns_norm,color=conseq))+geom_jitter(alpha=1,width=0, height = 0.0,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Functional score R2')+xlab('Functional score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1)

ggplot(XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, y=tHDR_post_lib_loess_e.sns_rL42_sns_norm,color=conseq))+geom_jitter(alpha=1,width=0, height = 0.0,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Functional score R2')+xlab('Functional score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1)


#are any variants unique to FLOSSIES?
XrL4_all_final_sns_norm_filt_CADD_fl_not_gn_br <- XrL4_all_final_sns_norm_filt_CADD[which( XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_flossies & ((XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad == FALSE)&(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo == FALSE))),]

XrL4_all_final_sns_norm_filt_CADD_fl_not_gn_br_cv <- XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_flossies & ((XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad == FALSE)&(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo == FALSE)&(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar == FALSE))),]

#Flossies (all) variants by conseq
ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$flossies_score != 0),], aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(flossies_score))),color=conseq))+geom_jitter(alpha=1,width=0, height = 0.2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Flossies AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#Flossies (all) variants by clinvar
ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$flossies_score != 0),], aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(flossies_score))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Flossies AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#Flossies (unique) variants by conseq
ggplot(XrL4_all_final_sns_norm_filt_CADD_fl_not_gn_br_cv, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(flossies_score))),color=conseq))+geom_jitter(alpha=1,width=0, height = 0.00,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Flossies AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
#Flossies (not in exac or bravo) variants by conseq
ggplot(XrL4_all_final_sns_norm_filt_CADD_fl_not_gn_br, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(flossies_score))),color=conseq))+geom_jitter(alpha=1,width=0, height = 0.00,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Flossies AF (log-10)')+xlab('Functional score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

XrL4_all_final_sns_norm_ns_syn_normex_gam <- XrL4_all_final_sns_norm_ns_syn_normex
XrL4_all_final_sns_norm_ns_syn_normex_gam$lof <- sapply(XrL4_all_final_sns_norm_ns_syn_normex_gam$conseq,assign_lof)

#which points have a conservation score over a given threshold?  -- by functional score?

#PhyloP (mammalian)
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=(as.numeric(as.character(mamPhyloP))),colour=conseq))+geom_jitter(alpha=0.5,width=0, height = 0.0,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('PhyloP score (mammals)')+xlab('Enrichment (post-lib corr., log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
cor(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD$mamPhyloP))),method='spearman')

#fitCons
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=(as.numeric(as.character(fitCons))),colour=conseq))+geom_jitter(alpha=0.5,width=0, height = 0.01,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('fitCons')+xlab('Enrichment (post-lib corr., log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
cor(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD$fitCons))),method='spearman')

#priPhCons
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=(as.numeric(as.character(priPhCons))),colour=conseq))+geom_jitter(alpha=0.5,width=0, height = 0.00,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('priPhCons')+xlab('Enrichment (post-lib corr., log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

cor(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD$priPhCons))),method='spearman')

#Grantham (only missense) colored by ClinVar
ggplot(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',], aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=(as.numeric(as.character(Grantham))),colour=clinvar_simple))+geom_jitter(alpha=1.0,width=0, height = 2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Grantham')+xlab('Enrichment (post-lib corr., log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#Grantham missense corr.
cor(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),'tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm'],(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),'Grantham']))),method='spearman')

#CADD missense corr.
cor(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),'tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm'],(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),'CADD.phred']))),method='spearman')

#polyphen missense corr.
cor(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),'tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm'],(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),'polyphen.val']))),method='spearman')

#sift missense corr.
cor(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),'tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm'],(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),'SIFTval']))),method='spearman')

#CADD synonymous corr.
cor(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),'tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm'],(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),'CADD.phred']))),method='spearman')

#write this out:
write.table(XrL4_all_final_sns_norm, "/mount/SGE/BRCA1/XrL4_all_final_sns_norm_20171226.txt", sep="\t")

#correlation by exon...
XrL4_func_score_cors_X2 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X2'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X2'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X3 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X3'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X3'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X4 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X4'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X4'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X5 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X5'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X5'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X15 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X15'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X15'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X16 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X16'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X16'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X17 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X17'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X17'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X18 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X18'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X18'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X19 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X19'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X19'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X20 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X20'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X20'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X21 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X21'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X21'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X22 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X22'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X22'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
XrL4_func_score_cors_X23 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X23'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X23'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')

XrL4_all_func_score_corss <- c(XrL4_func_score_cors_X2,XrL4_func_score_cors_X3,XrL4_func_score_cors_X4,XrL4_func_score_cors_X5,XrL4_func_score_cors_X15,XrL4_func_score_cors_X16,XrL4_func_score_cors_X17,XrL4_func_score_cors_X18,XrL4_func_score_cors_X19,XrL4_func_score_cors_X20,XrL4_func_score_cors_X21,XrL4_func_score_cors_X22,XrL4_func_score_cors_X23)

XrL4_func_score_corp_X2 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X2'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X2'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X3 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X3'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X3'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X4 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X4'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X4'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X5 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X5'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X5'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X15 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X15'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X15'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X16 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X16'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X16'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X17 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X17'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X17'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X18 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X18'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X18'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X19 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X19'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X19'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X20 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X20'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X20'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X21 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X21'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X21'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X22 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X22'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X22'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')
XrL4_func_score_corp_X23 <- cor(XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X23'),]$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_filt[which(XrL4_all_final_sns_norm_filt$exon == 'X23'),]$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='pearson')

XrL4_all_func_score_corp <- c(XrL4_func_score_corp_X2,XrL4_func_score_corp_X3,XrL4_func_score_corp_X4,XrL4_func_score_corp_X5,XrL4_func_score_corp_X15,XrL4_func_score_corp_X16,XrL4_func_score_corp_X17,XrL4_func_score_corp_X18,XrL4_func_score_corp_X19,XrL4_func_score_corp_X20,XrL4_func_score_corp_X21,XrL4_func_score_corp_X22,XrL4_func_score_corp_X23)

#scratch for poster:
#figure plots position in decreasing order (flips orientation using negative pos, matches gene orientation LtoR)
ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X2'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White') + geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2) 
#expression 
ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X2'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA expression')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4,1))

ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X15'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+ geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2)
#expression 
ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X15'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA expression')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4,1))

ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X16'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge)),colour=clinvar_simple),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+ geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2)
#expression 
ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X16'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA expression')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4,1))

ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X17'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+ geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2)
#expression 
ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X17'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_cDNApos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_cDNApos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA expression')+xlab('clinvarDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4,1))

#exon19 in grid.arrange format (note, not aligned)...
grid.arrange(ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X19'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge)),colour=clinvar_simple),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function (log-2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+ geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2),ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X19'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA expression')+xlab('clinvarDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4,1)),nrow=2)


#some counts:
#calculate missense variants with low RNA
#total
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),])
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm_filt_CADD$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.25),])
#no18
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'NON_SYNONYMOUS'),])
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.25),])
#no18 2-fold drop:
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.5),])
#no18 8-fold drop:
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.125),])

#calculate syn/ss variants with low RNA
#total
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & (XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SYNONYMOUS' | XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SPLICE_SITE')),])
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & (XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SYNONYMOUS' | XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SPLICE_SITE') & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.25),])
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & (XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SYNONYMOUS' | XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SPLICE_SITE') & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.125),])
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & (XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SYNONYMOUS' | XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SPLICE_SITE') & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.50),])

#which ones aren't down 4-fold?
XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & (XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SYNONYMOUS' | XrL4_all_final_sns_norm_filt_EXON_no18$conseq == 'SPLICE_SITE') & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm > 0.25),c('exon','rev_comp','clinvar_CDSpos')]


#exon 16:  SENSE LOF variants:  46/258 LOF;  26/258 under 50%, 15/258 under 25%
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X16' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED'),])

dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X16' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class != 'nana' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED'),])

#exon 16 
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X16' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.50),])

dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X16' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.25),])

#exon 17:  SENSE LOF variants:  
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X17' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED'),])

dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X17' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class != 'nana' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED'),])

#exon 17 
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X17' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.50),])

dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X17' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.25),])

#exon 19: SENSE LOF variants:  55 of 235 LOF (excluding nonsense), 0 with RNA under 50%
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X19' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED'),])

dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X19' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class != 'nana' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED'),])

#exon 19 
dim(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$exon == 'X19' & XrL4_all_final_sns_norm_filt_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_EXON_no18$conseq != 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.50),])

#discordant SNVs:
XrL4_all_final_sns_norm_filt_CADD[which((XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & (XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign' | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Likely benign')) | XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC' & (XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Pathogenic' | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Likely pathogenic')),c('exon','ref','rev_comp','clinvar_CDSpos','cDNApos','pos_alt','protPos','oAA','nAA','Dst2Splice','Dst2SplType','clinvar_simple','XrL4_mmfunc_class','tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm','tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')]

#get min CDSpos (clinvar) for an exon, here 16
min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),c('clinvar_CDSpos')])


##analysis for Fig 1b (indels)
#did this analysis in excel -- excluded variants with indels in primer, filtered on subset > 1 in 1000 reads
X17_cigars_df <- read.table("/mount/SGE/BRCA1/20161201_BRCA1x18_nextseq_SGE_pilots/fastq/Seqprep/merged/no_Ns/sam/BRCA1x18pX459r1_D5_top100.txt", sep="\t", header=TRUE)
which(as.numeric(as.character(X17_cigars_df$d5_rpt)) > 1)

require(ggplot2)
require(gridExtra)
require(mclust)
require(VennDiagram)
library(gam)
library(mixtools)
library(plotROC)
named_conseq_colors_old = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#0B3DB9", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#3885DB", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#9D0012","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "REF" = "#000000")
named_conseq_colors_full = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#0B3DB9", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#CF7700", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#9D0012","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "REF" = "#000000", "absent" = "#BFBCBC","Benign" = "#013DD3","Conflicting interpretations of pathogenicity" = "#FFCB00","Likely benign" = "#2360F9","Likely pathogenic" = "#FA2039","Pathogenic" = "#A40000","Uncertain significance" = "#00A54D" )
named_conseq_colors_full_simple = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#9D0012","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "REF" = "#000000", "absent" = "#BFBCBC","Benign" = "#013DD3","Conflicting interpretations of pathogenicity" = "#FFC927","Likely benign" = "#487CFD","Likely pathogenic" = "#FC3D53","Pathogenic" = "#A40000","Uncertain significance" = "#31947B","Gray" = "#000000" )
named_conseq_colors = c("STOP_GAINED" = "#C11600","CANONICAL_SPLICE" = "#FFB13F","INTRONIC" = "#6793F8", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#002372", "SYNONYMOUS" = "#0C4EE5","STOP_LOST" = "#9D0012","REGULATORY"="#B5A695","5PRIME_UTR"="#909B9F", "REF" = "#000000")
#define maxes and mins and make a global scale for plotting clinvar CDS pos
#global cDNS pos breaks and labels on pos data:
#SxClinVar:
X2_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X2_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X2_CDS_min),]$pos[1]))
X2_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2'),]$pos)))
X2_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X2_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X2_CDS_max),]$pos[1]))
X2_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2'),]$pos)))

X3_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X3' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X3_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X3' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X3_CDS_min),]$pos[1]))
X3_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X3'),]$pos)))
X3_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X3' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X3_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X3' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X3_CDS_max),]$pos[1]))
X3_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X3'),]$pos)))

X4_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X4_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X4_CDS_min),]$pos[1]))
X4_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4'),]$pos)))
X4_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X4_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X4_CDS_max),]$pos[1]))
X4_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4'),]$pos)))

X5_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X5' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X5_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X5' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X5_CDS_min),]$pos[1]))
X5_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X5'),]$pos)))
X5_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X5' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X5_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X5' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X5_CDS_max),]$pos[1]))
X5_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X5'),]$pos)))

X15_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X15' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X15_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X15' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X15_CDS_min),]$pos[1]))
X15_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X15'),]$pos)))
X15_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X15' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X15_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X15' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X15_CDS_max),]$pos[1]))
X15_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X15'),]$pos)))

X16_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X16_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X16_CDS_min),]$pos[1]))
X16_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16'),]$pos)))
X16_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X16_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X16_CDS_max),]$pos[1]))
X16_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16'),]$pos)))

X17_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X17' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X17_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X17' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X17_CDS_min),]$pos[1]))
X17_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X17'),]$pos)))
X17_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X17' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X17_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X17' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X17_CDS_max),]$pos[1]))
X17_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X17'),]$pos)))

X18_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X18' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X18_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X18' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X18_CDS_min),]$pos[1]))
X18_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X18'),]$pos)))
X18_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X18' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X18_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X18' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X18_CDS_max),]$pos[1]))
X18_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X18'),]$pos)))

X19_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X19' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X19_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X19' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X19_CDS_min),]$pos[1]))
X19_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X19'),]$pos)))
X19_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X19' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X19_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X19' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X19_CDS_max),]$pos[1]))
X19_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X19'),]$pos)))

X20_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X20' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X20_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X20' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X20_CDS_min),]$pos[1]))
X20_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X20'),]$pos)))
X20_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X20' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X20_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X20' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X20_CDS_max),]$pos[1]))
X20_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X20'),]$pos)))

X21_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X21_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X21_CDS_min),]$pos[1]))
X21_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21'),]$pos)))
X21_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X21_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X21_CDS_max),]$pos[1]))
X21_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21'),]$pos)))

X22_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X22' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X22_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X22' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X22_CDS_min),]$pos[1]))
X22_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X22'),]$pos)))
X22_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X22' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X22_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X22' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X22_CDS_max),]$pos[1]))
X22_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X22'),]$pos)))

X23_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X23' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X23_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X23' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X23_CDS_min),]$pos[1]))
X23_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X23'),]$pos)))
X23_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X23' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X23_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X23' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X23_CDS_max),]$pos[1]))
X23_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X23'),]$pos)))
#can further engineer this to be a vector of breaks and labels,equal in length, and labels = '', for every 10th base.
#Fig 4a
ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=as.numeric(as.character(pos))+as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=as.numeric(as.character(pos))+as.numeric(as.character(pos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White') + geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2)+scale_x_reverse(breaks=c(X21_CDS_min_pos,X21_CDS_max_pos),labels=c(X21_CDS_min,X21_CDS_max))

global_labels=c(X2_CDS_min,X2_CDS_max,X3_CDS_min,X3_CDS_max,X4_CDS_min,X4_CDS_max,X5_CDS_min,X5_CDS_max,X15_CDS_min,X15_CDS_max,X16_CDS_min,X16_CDS_max,X17_CDS_min,X17_CDS_max,X18_CDS_min,X18_CDS_max,X19_CDS_min,X19_CDS_max,X20_CDS_min,X20_CDS_max,X21_CDS_min,X21_CDS_max,X22_CDS_min,X22_CDS_max,X23_CDS_min,X23_CDS_max)
global_breaks=c(X2_CDS_min_pos,X2_CDS_max_pos,X3_CDS_min_pos,X3_CDS_max_pos,X4_CDS_min_pos,X4_CDS_max_pos,X5_CDS_min_pos,X5_CDS_max_pos,X15_CDS_min_pos,X15_CDS_max_pos,X16_CDS_min_pos,X16_CDS_max_pos,X17_CDS_min_pos,X17_CDS_max_pos,X18_CDS_min_pos,X18_CDS_max_pos,X19_CDS_min_pos,X19_CDS_max_pos,X20_CDS_min_pos,X20_CDS_max_pos,X21_CDS_min_pos,X21_CDS_max_pos,X22_CDS_min_pos,X22_CDS_max_pos,X23_CDS_min_pos,X23_CDS_max_pos)

X2_genomic_labels <- c(X2_CDS_min,'-19','-10',10,20,30,40,50,60,70,X2_CDS_max)
X2_genomic_breaks <- c(X2_CDS_min_pos,X2_CDS_min_pos+19,X2_CDS_min_pos+10,X2_CDS_min_pos-9,X2_CDS_min_pos-19,X2_CDS_min_pos-29,X2_CDS_min_pos-39,X2_CDS_min_pos-49,X2_CDS_min_pos-59,X2_CDS_min_pos-69,X2_CDS_max_pos)
X3_genomic_labels <- c(X3_CDS_min,'81-20','81-10',90,100,110,120,130,X3_CDS_max,'134+10','134+20')
X3_genomic_breaks <- c(X3_CDS_min_pos,X3_CDS_min_pos+20,X3_CDS_min_pos+10,X3_CDS_min_pos-9,X3_CDS_min_pos-19,X3_CDS_min_pos-29,X3_CDS_min_pos-39,X3_CDS_min_pos-49,X3_CDS_max_pos,X3_CDS_max_pos-10,X3_CDS_max_pos-20)
X4_genomic_labels <- c(X4_CDS_min,'135-10',seq(140,200,10),X4_CDS_max,'212+10')
X4_genomic_breaks <- c(X4_CDS_min_pos,X4_CDS_min_pos+10,X4_CDS_min_pos-5,X4_CDS_min_pos-15,X4_CDS_min_pos-25,X4_CDS_min_pos-35,X4_CDS_min_pos-45,X4_CDS_min_pos-55,X4_CDS_min_pos-65,X4_CDS_max_pos,X4_CDS_max_pos-10)
X5_genomic_labels <- c(X5_CDS_min,'213-10',seq(220,290,10),X5_CDS_max,'301+10')
X5_genomic_breaks <- c(X5_CDS_min_pos,X5_CDS_min_pos+10,X5_CDS_min_pos-7,X5_CDS_min_pos-17,X5_CDS_min_pos-27,X5_CDS_min_pos-37,X5_CDS_min_pos-47,X5_CDS_min_pos-57,X5_CDS_min_pos-67,X5_CDS_min_pos-77,X5_CDS_max_pos,X5_CDS_max_pos-10)
X15_genomic_labels <- c(X15_CDS_min,seq(4900,4980,10),X15_CDS_max,'4986+10')
X15_genomic_breaks <- c(X15_CDS_min_pos,X15_CDS_min_pos-9,X15_CDS_min_pos-19,X15_CDS_min_pos-29,X15_CDS_min_pos-39,X15_CDS_min_pos-49,X15_CDS_min_pos-59,X15_CDS_min_pos-69,X15_CDS_min_pos-79,X15_CDS_min_pos-89,X15_CDS_max_pos,X15_CDS_max_pos-10)
X16_genomic_labels <- c('4987-10',X16_CDS_min,seq(4990,5070,10),X16_CDS_max,'5074+10')
X16_genomic_breaks <- c(X16_CDS_min_pos+10,X16_CDS_min_pos,X16_CDS_min_pos-13,X16_CDS_min_pos-23,X16_CDS_min_pos-33,X16_CDS_min_pos-43,X16_CDS_min_pos-53,X16_CDS_min_pos-63,X16_CDS_min_pos-73,X16_CDS_min_pos-83,X16_CDS_max_pos,X16_CDS_max_pos-10)
X17_genomic_labels <- c('5075-10',X17_CDS_min,seq(5080,5140,10),X17_CDS_max,'5152+10')
X17_genomic_breaks <- c(X17_CDS_min_pos+10,X17_CDS_min_pos,X17_CDS_min_pos-5,X17_CDS_min_pos-15,X17_CDS_min_pos-25,X17_CDS_min_pos-35,X17_CDS_min_pos-45,X17_CDS_min_pos-55,X17_CDS_min_pos-65,X17_CDS_max_pos,X17_CDS_max_pos-10)
X18_genomic_labels <- c('5153-30','5153-20','5153-10',X18_CDS_min,seq(5160,5190,10),X18_CDS_max,'5193+10','5193+20','5193+30')
X18_genomic_breaks <- c(X18_CDS_min_pos+30,X18_CDS_min_pos+20,X18_CDS_min_pos+10,X18_CDS_min_pos,X18_CDS_min_pos-7,X18_CDS_min_pos-17,X18_CDS_min_pos-27,X18_CDS_min_pos-37,X18_CDS_max_pos,X18_CDS_max_pos-10,X18_CDS_max_pos-20,X18_CDS_max_pos-30)
X19_genomic_labels <- c('5194-10',X19_CDS_min,seq(5200,5270,10),X19_CDS_max,'5277+10')
X19_genomic_breaks <- c(X19_CDS_min_pos+10,X19_CDS_min_pos,X19_CDS_min_pos-6,X19_CDS_min_pos-16,X19_CDS_min_pos-26,X19_CDS_min_pos-36,X19_CDS_min_pos-46,X19_CDS_min_pos-56,X19_CDS_min_pos-66,X19_CDS_min_pos-76,X19_CDS_max_pos,X19_CDS_max_pos-10)
X20_genomic_labels <- c('5278-20','5278-10',X20_CDS_min,seq(5290,5320,10),X20_CDS_max,'5332+10','5332+20')
X20_genomic_breaks <- c(X20_CDS_min_pos+20,X20_CDS_min_pos+10,X20_CDS_min_pos,X20_CDS_min_pos-12,X20_CDS_min_pos-22,X20_CDS_min_pos-32,X20_CDS_min_pos-42,X20_CDS_max_pos,X20_CDS_max_pos-10,X20_CDS_max_pos-20)
X21_genomic_labels <- c('5333-10',X21_CDS_min,seq(5340,5400,10),X21_CDS_max,'5406+10')
X21_genomic_breaks <- c(X21_CDS_min_pos+10,X21_CDS_min_pos,X21_CDS_min_pos-7,X21_CDS_min_pos-17,X21_CDS_min_pos-27,X21_CDS_min_pos-37,X21_CDS_min_pos-47,X21_CDS_min_pos-57,X21_CDS_min_pos-67,X21_CDS_max_pos,X21_CDS_max_pos-10)
X22_genomic_labels <- c('5407-10',X22_CDS_min,seq(5410,5460,10),X22_CDS_max,'5467+10','5467+20')
X22_genomic_breaks <- c(X22_CDS_min_pos+10,X22_CDS_min_pos,X22_CDS_min_pos-3,X22_CDS_min_pos-13,X22_CDS_min_pos-23,X22_CDS_min_pos-33,X22_CDS_min_pos-43,X22_CDS_min_pos-53,X22_CDS_max_pos,X22_CDS_max_pos-10,X22_CDS_max_pos-20)
X23_genomic_labels <- c('5468-10',X23_CDS_min,seq(5480,5560,10),X23_CDS_max)
X23_genomic_breaks <- c(X23_CDS_min_pos+10,X23_CDS_min_pos,X23_CDS_min_pos-12,X23_CDS_min_pos-22,X23_CDS_min_pos-32,X23_CDS_min_pos-42,X23_CDS_min_pos-52,X23_CDS_min_pos-62,X23_CDS_min_pos-72,X23_CDS_min_pos-82,X23_CDS_min_pos-92,X23_CDS_max_pos)
global_labels_complete=c(X2_genomic_labels,X3_genomic_labels,X4_genomic_labels,X5_genomic_labels,X15_genomic_labels,X16_genomic_labels,X17_genomic_labels,X18_genomic_labels,X19_genomic_labels,X20_genomic_labels,X21_genomic_labels,X22_genomic_labels,X23_genomic_labels)
global_breaks_complete=c(X2_genomic_breaks,X3_genomic_breaks,X4_genomic_breaks,X5_genomic_breaks,X15_genomic_breaks,X16_genomic_breaks,X17_genomic_breaks,X18_genomic_breaks,X19_genomic_breaks,X20_genomic_breaks,X21_genomic_breaks,X22_genomic_breaks,X23_genomic_breaks)

##analysis for HDR rates (Fig.)
HDR_rates.df <- read.csv('/Users/Greg/Documents/BRCA1_SGE_HDR_rates.csv',header = TRUE)
HDR_rates_noX22 <- HDR_rates.df[which(HDR_rates.df$exon != '22'),]
HDR_rates_noX22_by_exon <- HDR_rates_noX22
#order this...by exon, then by cell type
HDR_rates_noX22_by_exon$exon <- factor(HDR_rates_noX22_by_exon$exon, levels = c('2','3','4','5','15','16','17','18','19','20','21','22','23'))
HDR_rates_noX22_by_exon_cells <- HDR_rates_noX22_by_exon
HDR_rates_noX22_by_exon_cells$cells <- factor(HDR_rates_noX22_by_exon_cells$cells, levels = c('WT','LIG4-KO'))

#define colors for this...
ggplot(HDR_rates_noX22_by_exon_cells,aes(fill=cells))+geom_bar(stat="identity",width=0.7,position = position_dodge(width = .7),color="#999999",aes(y=tHDR.ave,x=exon))+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+ylab('HDR rate')+xlab('Exon')+scale_fill_manual(values=c("WT" = "#BEBEBE","LIG4-KO" = "#5ab4ac"))

median(HDR_rates_noX22_by_exon_cells[which(HDR_rates_noX22_by_exon_cells$cells == 'WT'),c('tHDR.ave')])
median(HDR_rates_noX22_by_exon_cells[which(HDR_rates_noX22_by_exon_cells$cells == 'LIG4-KO'),c('tHDR.ave')])
#define fold-difference vector
1/median((HDR_rates_noX22_by_exon_cells[which(HDR_rates_noX22_by_exon_cells$cells == 'WT'),c('tHDR.ave')]/HDR_rates_noX22_by_exon_cells[which(HDR_rates_noX22_by_exon_cells$cells == 'LIG4-KO'),c('tHDR.ave')]))

       
#repeat with X22 (but 0 for WT):
HDR_rates.df_by_exon <- HDR_rates.df
#order this...by exon, then by cell type
HDR_rates.df_by_exon$exon <- factor(HDR_rates.df_by_exon$exon, levels = c('2','3','4','5','15','16','17','18','19','20','21','22','23'))
HDR_rates.df_by_exon_cells <- HDR_rates.df_by_exon
HDR_rates.df_by_exon_cells$cells <- factor(HDR_rates.df_by_exon_cells$cells, levels = c('WT','LIG4-KO'))
#manually replacing WT value for X22 to be 0
HDR_rates.df_by_exon_cells[which(HDR_rates.df_by_exon_cells$exon == '22' & HDR_rates.df_by_exon_cells$cells == 'WT'),7] = 0

#export at 3x6 to PDF
ggplot(HDR_rates.df_by_exon_cells,aes(fill=cells))+geom_bar(stat="identity",width=0.7,position = position_dodge(width = .7),color="#0E0138",aes(y=tHDR.ave,x=exon))+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+ylab('HDR rate')+xlab('Exon')+scale_fill_manual(values=c("WT" = "#BEBEBE","LIG4-KO" = "#210970"))

##FIGURE 2
#define exon:
XrL4_all_final_sns_norm_X17 <- XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$exon == 'X17'),]
#produce the legend from this:
ggplot(XrL4_all_final_sns_norm_X17, aes(x=log10(tHDR_pre_pseudo_freq), y=log10(rL42_tHDR_pre_pseudo_freq),color=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Day 5 freq. R2 (log-10)')+xlab('Day 5 freq. R1 (log-10)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels=c("Nonsense","Canonical splice","Synonymous","Intronic","Splice site","Missense"))+coord_cartesian(xlim=c(-3.7,-2.5),ylim=c(-3.7,-2.5))

#Figure 2 c-f
#metrics of reproducibility for one exon:  pre frequency
fig2c_gg <- ggplot(XrL4_all_final_sns_norm_X17, aes(x=log10(tHDR_pre_pseudo_freq), y=log10(rL42_tHDR_pre_pseudo_freq),fill=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('Day 5 freq. R2 (log10)')+xlab('Day 5 freq. R1 (log10)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_x_continuous(breaks=c(-3.5,-3.0,-2.5))+scale_y_continuous(breaks=c(-3.5,-3.0,-2.5))+coord_cartesian(xlim=c(-3.75,-2.5),ylim=c(-3.7,-2.5))
#metrics of reproducibility for one exon:  post lib frequency
fig2d_gg <-ggplot(XrL4_all_final_sns_norm_X17, aes(x=log2(tHDR_post_lib_ratio), y=log2(rL42_tHDR_post_lib_ratio),fill=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('Day 11 / lib. R2 (log2)')+xlab('Day 11 / lib. R1 (log2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_x_continuous(breaks=c(-2,-1,0,1))+scale_y_continuous(breaks=c(-2,-1,0,1))
#metrics of reproducibility for one exon:  functional score
fig2e_gg <- ggplot(XrL4_all_final_sns_norm_X17, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, y=tHDR_post_lib_loess_e.sns_rL42_sns_norm, fill=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3,1),ylim=c(-3,1))+scale_x_continuous(breaks=c(-3,-2,-1,0,1))+scale_y_continuous(breaks=c(-3,-2,-1,0,1))
#metrics of reproducibility for one exon:  RNA expression
fig2f_gg <- ggplot(XrL4_all_final_sns_norm_X17[which(XrL4_all_final_sns_norm_X17$CDSpos != 'NA'),], aes(x=log2(tHDR_rna_pre_ratio_synnorm), y=log2(rL42_tHDR_rna_pre_ratio_synnorm), fill=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('RNA score R2')+xlab('RNA score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+scale_x_continuous(breaks=c(-4,-3,-2,-1,0,1))+scale_y_continuous(breaks=c(-4,-3,-2,-1,0,1))+coord_cartesian(xlim=c(-5,1),ylim=c(-5,1))

#export as 3x12 PDF and re-size to scale in Illustrator.
grid.arrange(fig2c_gg,fig2d_gg,fig2e_gg,fig2f_gg,nrow=1)

#Saturation by exon...
#define SGE region lengths:
X2_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X2'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X2'),]$pos)))+1)
X3_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X3'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X3'),]$pos)))+1)
X4_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X4'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X4'),]$pos)))+1)
X5_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X5'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X5'),]$pos)))+1)
X15_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X15'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X15'),]$pos)))+1)
X16_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X16'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X16'),]$pos)))+1)
X17_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X17'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X17'),]$pos)))+1)
X18_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X18'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X18'),]$pos)))+1)
X19_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X19'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X19'),]$pos)))+1)
X20_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X20'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X20'),]$pos)))+1)
X21_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X21'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X21'),]$pos)))+1)
X22_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X22'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X22'),]$pos)))+1)
X23_possible_SNVs <- 3*(max(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X23'),]$pos)))-min(as.numeric(as.character(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X23'),]$pos)))+1)

possible_snvs_by_exon <- c(X2_possible_SNVs,X3_possible_SNVs,X4_possible_SNVs,X5_possible_SNVs,X15_possible_SNVs,X16_possible_SNVs,X17_possible_SNVs,X18_possible_SNVs,X19_possible_SNVs,X20_possible_SNVs,X21_possible_SNVs,X22_possible_SNVs,X23_possible_SNVs)
#4035 possible SNVs in region assayed...
sum(possible_snvs_by_exon)

X2_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2'),])[1]
X3_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X3'),])[1]
X4_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4'),])[1]
X5_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X5'),])[1]
X15_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X15'),])[1]
X16_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16'),])[1]
X17_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X17'),])[1]
X18_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X18'),])[1]
X19_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X19'),])[1]
X20_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X20'),])[1]
X21_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21'),])[1]
X22_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X22'),])[1]
X23_filtered_snvs <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X23'),])[1]

#from benchling check length of mutated:
per_exon_length <- c(106,100,100,109,106,108,98,101,104,100,105,100,108)

filtered_snvs_by_exon <- c(X2_filtered_snvs,X3_filtered_snvs,X4_filtered_snvs,X5_filtered_snvs,X15_filtered_snvs,X16_filtered_snvs,X17_filtered_snvs,X18_filtered_snvs,X19_filtered_snvs,X20_filtered_snvs,X21_filtered_snvs,X22_filtered_snvs,X23_filtered_snvs)
snv_coverage_by_exon <- filtered_snvs_by_exon/possible_snvs_by_exon
snv_coverage_exon <- c('2','3','4','5','15','16','17','18','19','20','21','22','23')

XrL4_snv_coverage.df <- data.frame(cbind(snv_coverage_exon,possible_snvs_by_exon,filtered_snvs_by_exon,snv_coverage_by_exon))
#order by exon
XrL4_snv_coverage.df$snv_coverage_exon <- factor(XrL4_snv_coverage.df$snv_coverage_exon, levels = c('2','3','4','5','15','16','17','18','19','20','21','22','23'))

#plot for figure 2 -- coverage by region in LIG4 experiments
ggplot(XrL4_snv_coverage.df)+geom_bar(stat="identity",width=0.7,color="#0E0138",fill="#210970",aes(y=as.numeric(as.character(snv_coverage_by_exon)),x=snv_coverage_exon))+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+ylab('SNV coverage')+xlab('Exon')+scale_y_continuous(limits=c(0,1))+geom_hline(yintercept=1,lty=2)+geom_text(aes(label=filtered_snvs_by_exon,x=snv_coverage_exon,y=as.numeric(as.character(snv_coverage_by_exon))),hjust=0.5, vjust=1.5, size = 3,colour='White')

#plot for figure 3A:
dim(XrL4_all_tHDR_pos_merge_master_df)
dim(XrL4_all_final_sns_norm_filt)
dim(XrL4_all_final_sns_norm_filt_CADD)

conseq_order_histo <- c('STOP_GAINED','CANONICAL_SPLICE','SYNONYMOUS','INTRONIC','SPLICE_SITE','5PRIME_UTR','NON_SYNONYMOUS')
labels_order_histo <- c("Nonsense","Canonical splice","Synonymous","Intronic","Splice site","5' UTR","Missense")
labels_order_histo_short <- c("NS","CS","SYN","INT","SS","5'UTR","MIS")

conseq_ord_XrL4_all_final_sns_norm_filt_CADD <- XrL4_all_final_sns_norm_filt_CADD
conseq_ord_XrL4_all_final_sns_norm_filt_CADD$conseq <- factor(conseq_ord_XrL4_all_final_sns_norm_filt_CADD$conseq, levels = conseq_order_histo)

#Figure 2g -- save as 3 x 5 landscape
ggplot(conseq_ord_XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position=c(0.3,0.7))+ylab('SNVs')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=labels_order_histo)+guides(fill = guide_legend(reverse=T))

#Figure 2h -- geom_bar showing func class by consequence -- 3 x 4
ggplot(conseq_ord_XrL4_all_final_sns_norm_filt_CADD,aes(fill=XrL4_mmfunc_class))+geom_bar(width=0.9,aes(x=conseq))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=9),panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = c(0.3,0.7))+ylab('SNVs')+xlab('Consequence')+scale_fill_manual(values=c("FUNC" = "#5B81D9","INT" = "#8D9486","LOF" = "#292A2C"),labels = c('Functional','Intermediate','Non-functional'))+scale_x_discrete(labels=labels_order_histo_short)


#Figure 3a -- save as 3 x 5 landscape
clinvar_ord_XrL4_all_final_sns_norm_filt_CADD <- XrL4_all_final_sns_norm_filt_CADD
clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple <- factor(clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

ggplot(clinvar_ord_XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SNVs')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

clinvar_ord_XrL4_all_final_sns_norm_filt_CADD_p_b_lp_lb <- clinvar_ord_XrL4_all_final_sns_norm_filt_CADD[which( clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Pathogenic' | clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign' | clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Likely pathogenic' | clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Likely benign'),]

#Figure 3a
fig3a_gg <- ggplot(clinvar_ord_XrL4_all_final_sns_norm_filt_CADD_p_b_lp_lb, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position=c(0.20,0.7))+ylab('SNVs')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+coord_cartesian(xlim=c(-5,1.3))

clinvar_ord_XrL4_all_final_sns_norm_filt_CADD_vus_ci <- clinvar_ord_XrL4_all_final_sns_norm_filt_CADD[which( clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Uncertain significance' | clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Conflicting interpretations of pathogenicity'),]

clinvar_ord_XrL4_all_final_sns_norm_filt_CADD_absent <- clinvar_ord_XrL4_all_final_sns_norm_filt_CADD[which( clinvar_ord_XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent'),]

#Figure 3b
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc <- XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc$D_P.B.LP.LB <- 1
assign_P.B.LP.LB <- function(clinvar_simple){
  if (clinvar_simple == 'Pathogenic' | clinvar_simple == 'Likely pathogenic') return(1) else return(0)}
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc$D_P.B.LP.LB <- sapply(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc$clinvar_simple,assign_P.B.LP.LB)
P.B.LP.LB_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc, aes(d = D_P.B.LP.LB, m = -tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity")
#with AUC:
P.B.LP.LB_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_roc)$AUC, 3)))

#Figure 3c
fig3c_gg <- ggplot(clinvar_ord_XrL4_all_final_sns_norm_filt_CADD_vus_ci, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position=c(0.20,0.7))+ylab('SNVs')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple,labels = c('Uncertain','Conflicting'))+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+coord_cartesian(xlim=c(-5,1.3))

#export stacked 3a/3c for matching as a 6x5 pfg
grid.arrange(fig3a_gg,fig3c_gg,nrow=2)

#fig3d_gg <- ggplot(clinvar_ord_XrL4_all_final_sns_norm_filt_CADD_absent, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position=c(0.20,0.7))+ylab('SNVs')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple,labels = c('Uncertain','Conflicting'))+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+coord_cartesian(xlim=c(-5,1.3))



#Figure 3d -- for legend only
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(ExAC.AF))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.1,size = 0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('ExAC AF (log-10)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+coord_cartesian(xlim=c(-4,1.7))

ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(gnomad_AF))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.2,size = 0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+ylab('gnomAD AF (log-10)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+coord_cartesian(xlim=c(-4,1.7))

#Figure 3e -- export 3 x 5 landscape
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(CADD.phred))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.7)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple != 'absent' & XrL4_all_final_sns_norm_filt_CADD$clinvar_simple != 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.7)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('CADD score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="black")+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")

cor(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD$CADD.raw))),method='spearman')

#Fig
#ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White') + geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2) 
#expression 
#ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X2'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_cDNApos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(cDNApos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4,1))

#Figure 4b
#ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=-as.numeric(as.character(pos))-as.numeric(as.character(pos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_blank(), axis.title.x = element_blank(),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White') + geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2) 
#expression 
#ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-5,1))

XrL4_all_final_sns_norm_filt_CADD_EXON_no18$clinvar_simple <- factor(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))
labels_order_clinvar_short <- c('Pathogenic', 'Likely path.', 'Benign', 'Likely ben.', 'Uncertain sig.','Conflicting int.','absent')

#Figure 5a -- filter on -- no legend.
ggplot(XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$conseq != 'STOP_GAINED'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm), color=clinvar_simple))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.7)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$clinvar_simple != 'absent' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$clinvar_simple != 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.7)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.7)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank(), legend.position = "None")+scale_color_manual(values=named_conseq_colors_full_simple,labels = labels_order_clinvar_short)+geom_hline(yintercept=XrL4_all_f01_thresh,lty=2,color="gray30")+geom_hline(yintercept=XrL4_all_lof_thresh,lty=2,color="gray30")+geom_vline(xintercept=-2,lty=3,color="gray80")+geom_vline(xintercept=log2(0.125),lty=3,color="gray80")


#switch to have the lines be the function of the SNV score
func_colors_tile <- c("FUNC" = "#FFFFFF","INT" = "#8D9486","LOF" = "#292A2C")
named_conseq_colors_func <- c(named_conseq_colors,func_colors_tile)
func_labels <- c('Functional','Intermediate','Non-functional')

#figure 5e #save as 3.5 x 14 and shrink in half in illustrator
ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=XrL4_mmfunc_class),size=0.4,yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_func)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4.5,1))
#figure 5f
ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X19'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=XrL4_mmfunc_class),size=0.4,yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_func)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4.5,1))

#for legend
ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X16'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=XrL4_mmfunc_class),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "bottom")+ylab('RNA score')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_func)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-4.5,1))

#for individual panels:


#figure 5? --> GT donor sites, exons 3, 4, and 17
Fig.5d_X3_gg <- ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X3'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=XrL4_mmfunc_class),size=0.4,yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_func)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-6,1))+scale_x_continuous(limits=c(X3_CDS_max-20.5,X3_CDS_max+0.5),breaks=c(X3_CDS_max-20,X3_CDS_max-15,X3_CDS_max-10,X3_CDS_max-5,X3_CDS_max))
Fig.5d_X4_gg <- ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=XrL4_mmfunc_class),size=0.4,yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_func)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-6,1))+scale_x_continuous(limits=c(X4_CDS_max-20.5,X4_CDS_max+0.5),breaks=c(X4_CDS_max-20,X4_CDS_max-15,X4_CDS_max-10,X4_CDS_max-5,X4_CDS_max))
Fig.5d_X17_gg <- ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X17'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=XrL4_mmfunc_class),size=0.4,yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('cDNA position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_func)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-6,1))+scale_x_continuous(limits=c(X17_CDS_max-20.5,X17_CDS_max+0.5),breaks=c(X17_CDS_max-20,X17_CDS_max-15,X17_CDS_max-10,X17_CDS_max-5,X17_CDS_max))

#export as a 4x14 to edit further in illustrator.
grid.arrange(Fig.5d_X3_gg,Fig.5d_X4_gg,Fig.5d_X17_gg,nrow=1)

#figure 4e?
#expression 
ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X5'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-5,1))

#HEATPLOT OPTIONS (FIGURE 4...) :
### ASSIGN CATEGORIES TO ALL VARIANTS BASED ON THE POSTERIOR OF NS FROM mixtools MM:
func_colors_tile <- c("FUNC" = "#FFFFFF","INT" = "#8D9486","LOF" = "#292A2C")
func_labels <- c('Functional','Intermediate','Non-functional')


#functions to remake scoring systems and assign an additional pos_base_combo for WT...
#use this for score input:  tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm
#if assigned to functional category -- give it one color.
'''assign_func_score_for_plotting <- function(func_score){
  if (func_score >= XrL4_all_f01_thresh) return(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_syn) else if (func_score <=  XrL4_all_lof_thresh) return(XrL4_all_tHDR_post_lib_loess_e.sns_rL41rL42_median_ns) else return(func_score)
}'''
assign_func_score_for_plotting <- function(func_score){
  if (func_score >= XrL4_all_f01_thresh) return(XrL4_all_f01_thresh) else if (func_score <=  XrL4_all_lof_thresh) return(XrL4_all_lof_thresh) else return(func_score)
}

#assign_rna_score_for_plotting <- function(clinvar_cDNApos,rna_score){
#  if (clinvar_cDNApos == 'NA' | is.na(clinvar_cDNApos)) return ('') else if (rna_score <= -3) #return('...') else if (rna_score <= -2) return('..') else if (rna_score <= -1) return('.') else #return('')
#}

assign_rna_score_for_plotting <- function(clinvar_cDNApos,rna_score,exon){
  if (clinvar_cDNApos == 'NA' | is.na(clinvar_cDNApos) | exon == 'X18') return ('') else if (rna_score <= -3) return('||') else if (rna_score <= -2) return('|') else return('')
}

#facet this call on exon and re-roll.

XrL4_all_df_for_tile <- XrL4_all_final_sns_norm_filt_CADD[,]

XrL4_all_df_for_tile$func_score_for_plotting <- sapply(XrL4_all_df_for_tile$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, assign_func_score_for_plotting)
XrL4_all_df_for_tile$rna_score_for_plotting <- mapply(assign_rna_score_for_plotting,XrL4_all_df_for_tile$clinvar_cDNApos,log2(XrL4_all_df_for_tile$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),XrL4_all_df_for_tile$exon)
XrL4_all_df_for_tile$rev_comp <- factor(XrL4_all_df_for_tile$rev_comp, levels = c('T','G','C','A'))
XrL4_all_df_for_tile$exon <- factor(XrL4_all_df_for_tile$exon, levels =  c('X2','X3','X4','X5','X15','X16','X17','X18','X19','X20','X21','X22','X23'))


#remove the labels and resize the axes text:  #5 x 7.5 figure
ggplot(XrL4_all_df_for_tile, aes(y=rev_comp, x=as.numeric(as.character(pos)))) + geom_tile(aes(fill = func_score_for_plotting,color = conseq,width=0.7, height=.9),size=.15)+scale_fill_manual(values = func_colors_tile)+scale_color_manual(values=named_conseq_colors)+theme(panel.background = element_rect(fill = 'white', colour = 'white'),axis.ticks = element_line(size = .3)) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none") + geom_text(aes(label=rna_score_for_plotting),hjust=0.45, vjust=0.50, size = 1.23,color='Red',angle=90,fontface="bold")+scale_x_reverse(breaks=global_breaks_complete,labels=global_labels_complete)+scale_fill_gradient(low = "black", high = "white")+facet_wrap( ~ exon, scales = 'free',ncol=1)+theme(strip.background = element_blank(),strip.text.x = element_blank(),panel.margin = unit(0.1, "lines"),axis.text.y = element_text(size=4.3),axis.text.x = element_text(size=5))+ylab('SNV')+xlab('CDS position')

#try to add the reference base at each position...
assign_rev_comp <- function(base){
  if (base == 'T') return ('A') else if (base == 'A') return ('T') else if (base == 'C') return ('G') else if (base == 'G') return ('C') else return (base)
}
XrL4_all_tHDR_pos_merge_master_df$ref_rc <- sapply(XrL4_all_tHDR_pos_merge_master_df$ref,assign_rev_comp)

XrL4_all_tHDR_pos_merge_master_df$exon <- factor(XrL4_all_tHDR_pos_merge_master_df$exon, levels =  c('X2','X3','X4','X5','X15','X16','X17','X18','X19','X20','X21','X22','X23'))

#remove the labels and resize the axes text:  #5 x 7.5 figure
ggplot()+geom_tile(data=XrL4_all_df_for_tile,aes(x=as.numeric(as.character(pos)),y=rev_comp, fill = func_score_for_plotting,color = conseq,width=0.7, height=.9),size=.15)+scale_fill_manual(values = func_colors_tile)+scale_color_manual(values=named_conseq_colors)+theme(panel.background = element_rect(fill = 'white', colour = 'white'),axis.ticks = element_line(size = .3)) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none") + geom_text(data=XrL4_all_df_for_tile,aes(label=rna_score_for_plotting,x=as.numeric(as.character(pos)),y=rev_comp),hjust=0.45, vjust=0.50, size = 1.23,color='Red',angle=90,fontface="bold")+scale_x_reverse(breaks=global_breaks_complete,labels=global_labels_complete)+scale_fill_gradient(low = "black", high = "white")+theme(strip.background = element_blank(),strip.text.x = element_blank(),panel.margin = unit(0.1, "lines"),axis.text.y = element_text(size=4.3),axis.text.x = element_text(size=5),axis.ticks.x=element_line(size=.10),axis.ticks.y=element_line(size=.15))+ylab('SNV')+xlab('CDS position')+geom_text(data=XrL4_all_tHDR_pos_merge_master_df,aes(label=ref_rc,y=ref_rc,x=as.numeric(as.character(pos))),size=1.5,color='Gray50')+facet_wrap( ~ exon, scales = 'free',ncol=1)


#within functional groups, is there a correlation to CADD scores?
cor(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$CADD.raw))),method='spearman')

#HOW TO write out / read in final data if needed
#write.table(XrL4_all_final_sns_norm_filt_CADD,"/mount/SGE/BRCA1/XrL4_all_final_sns_norm_filt_CADD_20171226.txt", sep="\t")
#XrL4_all_final_sns_norm_filt_CADD <- read.table("/mount/SGE/BRCA1/XrL4_all_final_sns_norm_filt_CADD_20171226.txt", sep="\t", header=TRUE)

#SxOpt:
#WT HAP1 X17 (must have run SGE1 scripts for this)
Xr_all_final_sns_norm_X17 <- Xr_all_final_sns_norm[which(Xr_all_final_sns_norm$exon == 'X17'),]

#WT HAP1 X17:
ggplot(Xr_all_final_sns_norm_X17, aes(x=tHDR_post_lib_loess_e.sns_r1_sns_norm, y=tHDR_post_lib_loess_e.sns_r2_sns_norm, fill=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-6,1),ylim=c(-6,1))
dim(Xr_all_final_sns_norm_X17)
cor(Xr_all_final_sns_norm_X17$tHDR_post_lib_loess_e.sns_r1_sns_norm,Xr_all_final_sns_norm_X17$tHDR_post_lib_loess_e.sns_r2_sns_norm,method='spearman')

#1N-sorted HAP1 X17 (must have run SGE1 scripts for this)
Xr1N_all_final_sns_norm_X17 <- Xr1N_all_final_sns_norm[which(Xr1N_all_final_sns_norm$exon == 'X17'),]

#1N-sorted HAP1:
ggplot(Xr1N_all_final_sns_norm_X17, aes(x=tHDR_post_lib_loess_e.sns_r1_sns_norm, y=tHDR_post_lib_loess_e.sns_r2_sns_norm, fill=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-7,1),ylim=c(-7,1))
dim(Xr1N_all_final_sns_norm_X17)
cor(Xr1N_all_final_sns_norm_X17$tHDR_post_lib_loess_e.sns_r1_sns_norm,Xr1N_all_final_sns_norm_X17$tHDR_post_lib_loess_e.sns_r2_sns_norm,method='spearman')

#LIG4-KO HAP1 X17:
ggplot(XrL4_all_final_sns_norm_X17, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, y=tHDR_post_lib_loess_e.sns_rL42_sns_norm, fill=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-3,1),ylim=c(-3,1))
dim(XrL4_all_final_sns_norm_X17)
cor(XrL4_all_final_sns_norm_X17$tHDR_post_lib_loess_e.sns_rL41_sns_norm,XrL4_all_final_sns_norm_X17$tHDR_post_lib_loess_e.sns_rL42_sns_norm,method='spearman')
#legend only
Xr1N_all_final_sns_norm_X17$conseq <- factor(Xr1N_all_final_sns_norm_X17$conseq, levels = conseq_order_histo)

ggplot(Xr1N_all_final_sns_norm_X17, aes(x=tHDR_post_lib_loess_e.sns_r1_sns_norm, y=tHDR_post_lib_loess_e.sns_r2_sns_norm, fill=conseq))+geom_point(aes(colour=conseq),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'),legend.position = c(.2,.2))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors, labels = labels_order_histo )+coord_cartesian(xlim=c(-7,1),ylim=c(-7,1))



#SxIndels in its own RMD file

#SxPosCor
XrL4_all_tHDR_pos_merge_master_df$exon <- factor(XrL4_all_tHDR_pos_merge_master_df$exon, levels =  c('X2','X3','X4','X5','X15','X16','X17','X18','X19','X20','X21','X22','X23'))

XrL4_all_tHDR_pos_merge_master_df$conseq <- factor(XrL4_all_tHDR_pos_merge_master_df$conseq, levels = conseq_order_histo)
labels_order_histo <- c("Nonsense","Canonical splice","Synonymous","Intronic","Splice region","5' UTR","Missense")
labels_order_histo_short <- c("NS","CS","SYN","INT","SR","5'UTR","MIS")


#rL41rL42
ggplot(XrL4_all_tHDR_pos_merge_master_df)+geom_point(aes(y=log2(tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),x=-as.numeric(as.character(pos)),color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "bottom")+ylab('pre / lib (log-2)')+xlab('Chr. 17 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=9),axis.text.y = element_text(size=10),axis.text.x = element_text(size=7),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels=labels_order_histo)+geom_line(aes(y=rL41rL42_tHDR_pre_lib_pos_effects.x,x=-as.numeric(as.character(pos))))+facet_wrap(~exon,scales="free")

#Fig SxPosCor -- export as 8x12, move legend up with crop. and remove legend
ggplot(XrL4_all_tHDR_pos_merge_master_df)+geom_point(aes(y=log2(tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(pos)),color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'bottom',legend.key.width=unit(1,'cm'),axis.text.x = element_blank())+ylab('Day 5 / lib. (log2)')+xlab('Chr. 17 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10))+ scale_color_manual(values=named_conseq_colors,labels=labels_order_histo)+geom_line(aes(y=rL41rL42_tHDR_pre_lib_pos_effects.x,x=as.numeric(as.character(pos))))+facet_wrap(~exon,scales="free")+scale_x_reverse()

#FigSxPosCor b -- (single exon post/lib before and after correction) export each as 4x9
ggplot(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X3'),])+geom_point(aes(y=log2(tHDR_post_lib_ratio_synnorm),x=as.numeric(as.character(pos)),color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'bottom',legend.key.width=unit(1,'cm'),axis.text.x = element_blank())+ylab('Day 11 / lib. (log2)')+xlab('Chr. 17 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=9))+ scale_color_manual(values=named_conseq_colors,labels=labels_order_histo)+scale_x_reverse()
#c
ggplot(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X3'),])+geom_point(aes(y=rL41_tHDR_post_lib_loess_rL41_e.sns,x=as.numeric(as.character(pos)),color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'bottom',legend.key.width=unit(1,'cm'),axis.text.x = element_blank())+ylab('Day 11 / lib. corrected (log2)')+xlab('Chr. 17 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=9))+ scale_color_manual(values=named_conseq_colors,labels=labels_order_histo)+scale_x_reverse()

#FigSxPosCor d,e  #export as 6 x 5 plot
grid.arrange(ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_lib_ratio_rL41rL42_mean_synnorm), fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position=c(0.25,0.6))+ylab('SNVs')+xlab('Day 11 / lib. (log2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=labels_order_histo)+guides(fill = guide_legend(reverse=T))+coord_cartesian(xlim=c(-6,2.5)),ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=conseq))+geom_histogram(alpha=1,bins=30)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position="None")+ylab('SNVs')+xlab('Day 11 / lib. (log2) position-corrected')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=labels_order_histo)+guides(fill = guide_legend(reverse=T))+coord_cartesian(xlim=c(-6,2.5)),nrow=2)

#edit this plot to make it for SxFiltering!  export each as 5x5 pdf.

XrL4_all_final_sns_norm$conseq <- factor(XrL4_all_final_sns_norm$conseq, levels = conseq_order_histo)
XrL4_all_tHDR_pos_merge_master_df$conseq <- factor(XrL4_all_tHDR_pos_merge_master_df$conseq, levels = conseq_order_histo)

#Sxfiltering (a is flowchart) c,d
ggplot(XrL4_all_tHDR_pos_merge_master_df, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),y=rL41rL42_tHDR_pre_lib_loess_rL41rL42, color=conseq))+geom_point()+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='bottom')+ylab('Day 5 / lib. corrected (log2)')+xlab('Day 11 / day 5 (log2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels=labels_order_histo)+coord_cartesian(ylim = c(-6,2.5),xlim=c(-6,2.5))
ggplot(XrL4_all_final_sns_norm, aes(x=log2(tHDR_post_pre_ratio_rL41rL42_mean_synnorm),y=rL41rL42_tHDR_pre_lib_loess_rL41rL42, color=conseq))+geom_point()+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='bottom')+ylab('Day 5 / lib. corrected (log2)')+xlab('Day 11 / day 5 (log2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels=labels_order_histo)+coord_cartesian(ylim = c(-6,2.5),xlim=c(-6,2.5))

#pre-filtering XrL4_all_tHDR_pos_merge_master_df vs post-filtering XrL4_all_final_sns_norm_filt_CADD for Pathogenic
#histograms

XrL4_all_tHDR_pos_merge_master_df$clinvar_simple <- factor(XrL4_all_tHDR_pos_merge_master_df$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))
 
XrL4_all_final_sns_norm$clinvar_simple <- factor(XrL4_all_final_sns_norm$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

XrL4_all_final_sns_norm_filt_CADD$clinvar_simple <- factor(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple, levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

#Sxfiltering (a is flowchart)
#b - show PAM types for X15 #export this as a 6x3 plot, crop out the legend, use shared legend with above plot
ggplot(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$exon == 'X15'),])+geom_point(aes(y=log2(tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(pos)),color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'bottom',legend.key.width=unit(1,'cm'),axis.text.x = element_blank())+ylab('SNV Day 5 / Lib (log-2)')+xlab('Chr. 17 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=9))+scale_color_manual(values=named_conseq_colors,labels=labels_order_histo)+geom_line(aes(y=rL41rL42_tHDR_pre_lib_pos_effects.x,x=as.numeric(as.character(pos))))+scale_x_reverse(breaks=c(41222940,41222950,41222960))+coord_cartesian(xlim=c(41222940,41222965))+geom_text(aes(y=log2(tHDR_pre_lib_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(pos)),label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

#export all as 4 x 6 plots, and stack into figure Sxfiltering
ggplot(XrL4_all_tHDR_pos_merge_master_df[which(XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Pathogenic' | XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Likely pathogenic' | XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Benign' | XrL4_all_tHDR_pos_merge_master_df$clinvar_simple == 'Likely benign'),], aes(x=rL4_tHDR_post_lib_loess_e.sns_rL41rL42_mean, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='bottom')+ylab('SNVs')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$clinvar_simple == 'Pathogenic' | XrL4_all_final_sns_norm$clinvar_simple == 'Likely pathogenic' | XrL4_all_final_sns_norm$clinvar_simple == 'Benign' | XrL4_all_final_sns_norm$clinvar_simple == 'Likely benign'),], aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='bottom')+ylab('SNVs')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Pathogenic' | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Likely pathogenic' | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign' | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Likely benign'),], aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=clinvar_simple))+geom_histogram(alpha=1,bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='bottom')+ylab('SNVs')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)



#SxMixtureModel:
#a  Distribution of training sets save as 3x5
XrL4_all_final_sns_norm_ns_syn_normex$conseq_simple <- XrL4_all_final_sns_norm_ns_syn_normex$conseq
XrL4_all_final_sns_norm_ns_syn_normex[which(XrL4_all_final_sns_norm_ns_syn_normex$conseq == "SPLICE_SITE"),]$conseq_simple <- 'SYNONYMOUS'
ggplot(XrL4_all_final_sns_norm_ns_syn_normex, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, fill=conseq_simple))+geom_histogram(aes(fill=conseq_simple),alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = c(0.3,0.7))+xlab('Function score')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple,labels=c('Nonsense (N = 141)','Synonymous (N = 493)'))+scale_color_manual(values=named_conseq_colors_full_simple)
#+coord_cartesian(xlim=c(-4,1.2))
#+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+geom_vline(xintercept=XrL4_all_OR_cutoff,lty=2,color="gray")

#ROC curve (b) save as 3x3 pdf
XrL4_all_final_sns_norm_ns_syn_normex_geom_roc <- XrL4_all_final_sns_norm_ns_syn_normex
XrL4_all_final_sns_norm_ns_syn_normex_geom_roc$D_ns_syn_normex <- 1
assign_ns_syn_normex <- function(conseq){
  if (conseq == 'STOP_GAINED') return(1) else return(0)}
XrL4_all_final_sns_norm_ns_syn_normex_geom_roc$D_ns_syn_normex <- sapply(XrL4_all_final_sns_norm_ns_syn_normex_geom_roc$conseq,assign_ns_syn_normex)
XrL4_all_final_sns_norm_ns_syn_normex_roc <- ggplot(XrL4_all_final_sns_norm_ns_syn_normex_geom_roc, aes(d = D_ns_syn_normex, m = -tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)) +geom_roc(n.cuts=0)+style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC: ###EXPORT AS 3x3 PDF for SxMixModb
XrL4_all_final_sns_norm_ns_syn_normex_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(ns_syn_normex_roc)$AUC, 4)))

#c  P(NS) as a function of funcitonal score (all data) save as 3x6 PDF
ggplot(XrL4_all_final_sns_norm, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=posterior_ns))+geom_jitter(aes(colour=conseq),alpha=0.3,height=0.00)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'none')+ylab('P(non-functional)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=max(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns > 0.99),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+geom_vline(xintercept=min(XrL4_all_final_sns_norm[which(XrL4_all_final_sns_norm$posterior_ns < 0.01),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')]),lty=2)+coord_cartesian(ylim=c(-0.12,1),xlim=c(-1.7,-0.5))

XrL4_all_final_sns_norm_filt_CADD$conseq <- factor(XrL4_all_final_sns_norm_filt_CADD$conseq, levels = conseq_order_histo)

#SxMixMod d, save as 3x5 PDF
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, color=conseq,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(alpha=0.5,size=0.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors, labels = labels_order_histo) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_OR_cutoff,lty=2,slope = -1,color="gray")

#MODEL APPLIED ACCROSS EACH EXON INDIVIDUALLY
#candidate figure for supplemental showing exons, both replicates...
XrL4_all_final_sns_norm_filt_CADD_exon_ordered$conseq <- factor( XrL4_all_final_sns_norm_filt_CADD_exon_ordered$conseq, levels = conseq_order_histo)

#SxMixMod e export as 3.5 x 9 and shrink to 7 inches wide for figure
ggplot(XrL4_all_final_sns_norm_filt_CADD_exon_ordered, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(aes(color=conseq),alpha=0.5,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position=c(.93,.23))+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=9),axis.text.y = element_text(size=9),axis.text.x = element_text(size=9),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors,labels = labels_order_histo) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1,color="Gray") + facet_wrap( ~ exon, nrow=2)



levels_clinvar_simple <- c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent")
XrL4_all_final_sns_norm_filt_CADD$clinvar_simple <- factor(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple, levels = levels_clinvar_simple)


#for clinvar_categorization by functional class:
labels_order_clinvar <- c("Pathogenic","Likely pathogenic","Likely benign","Benign","Uncertain significance","Conflicting interpretations of pathogenicity","absent")
#+guides(fill = guide_legend(reverse=T))

#should fix order of labels here and elsewhere to be consistent.
#export as 6 x 8 PDF for SxClinvarConseq

conseq_paired_labels <- list(
  'STOP_GAINED'="Nonsense",
  'CANONICAL_SPLICE'="Canonical splice",
  'SYNONYMOUS'="Synonymous",
  'INTRONIC'="Intronic",
  'SPLICE_SITE'="Splice region",
  '5PRIME_UTR'="5' UTR",
  'NON_SYNONYMOUS'="Missense"
)

conseq_labeller <- function(variable,value){
  return(conseq_paired_labels[value])
}

ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position=c(.93,.23))+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=9),axis.text.y = element_text(size=9),axis.text.x = element_text(size=9),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_full_simple) + geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1,color='Gray')+facet_wrap( ~ conseq, nrow=2,labeller=conseq_labeller)

#SxPop -- BRAVO -- legend for scatter 4x4 PDF with ClinVar colors
ggplot(XrL4_all_final_sns_norm_filt_CADD_bravo, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(bravo_AF))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.1,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position=c(.3,.8))+ylab('Bravo AF (log-10)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#export as 4x4 pdf for SxPopSeq
ggplot(XrL4_all_final_sns_norm_filt_CADD_bravo, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(bravo_AF))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Bravo AF (log10)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

cor.test(XrL4_all_final_sns_norm_filt_CADD_bravo$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_bravo$bravo_AF)),method='spearman')

#SNVs in FLOSSIES:
#Flossies (all) variants by clinvar  #export as 4x4 pdf for SxPopSeq
ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$flossies_score != 0),], aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(flossies_score))),color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.2,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Flossies AF (log-10)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)+scale_x_continuous(limits=c(-2,0.7))

cor.test(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$flossies_score != 0),]$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$flossies_score != 0),]$flossies_score)),method='spearman')


#Flossies (unique) variants by conseq (gray in above figure)
ggplot(XrL4_all_final_sns_norm_filt_CADD_fl_not_gn_br_cv, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(flossies_score))),color=conseq))+geom_jitter(alpha=1,width=0, height = 0.00,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Flossies AF (log-10)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)
#Flossies (not in exac or bravo) variants by conseq
ggplot(XrL4_all_final_sns_norm_filt_CADD_fl_not_gn_br, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y=log10(as.numeric(as.character(flossies_score))),color=conseq))+geom_jitter(alpha=1,width=0, height = 0.00,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Flossies AF (log-10)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_vline(xintercept=XrL4_all_lof_thresh,lty=2)+geom_vline(xintercept=XrL4_all_f01_thresh,lty=2)

#venn diagram showing overlap between ClinVar, Bravo and gnomad
require(VennDiagram)
#intersect variants in clinvar, bravo and gnomad!!
#length of in any of the three:
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad))
#clinvar + gnomad
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad))
#clinvar + bravo
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo))
#gnomad or bravo
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo))
#gnomad AND bravo
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo))
#gnomad AND bravo AND clinvar
#length of in any of the three:
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad))

draw.triple.venn(area1 = length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar)),area2 = length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad)),area3 = length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo)),n12 = length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad)),n13 = length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo)),n23 = length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo)),n123 = length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo & XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad)),category = c("ClinVar","gnomAD","Bravo"),fill = c('red','blue','yellow'),alpha=0.5,euler.d = TRUE,scaled=TRUE)

XrL4_all_clinvar_mis_not_in_gn_br_points <- ggplot(XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br[XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br$conseq == 'NON_SYNONYMOUS',], aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, y=tHDR_post_lib_loess_e.sns_rL42_sns_norm,color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.0,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1)+coord_cartesian(xlim=c(-5,1),ylim=c(-5,1))

XrL4_all_clinvar_mis_in_gn_br_points <- ggplot(XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br[XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br$conseq == 'NON_SYNONYMOUS',], aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, y=tHDR_post_lib_loess_e.sns_rL42_sns_norm,color=clinvar_simple))+geom_jitter(alpha=1,width=0, height = 0.0,size = 1.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score R2')+xlab('Function score R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_abline(intercept=2*XrL4_all_lof_thresh,lty=2,slope = -1) + geom_abline(intercept=2*XrL4_all_f01_thresh,lty=2,slope = -1)+coord_cartesian(xlim=c(-5,1),ylim=c(-5,1))

#export as 4" x 8.5" for SxPopSeq c/d
grid.arrange(XrL4_all_clinvar_mis_in_gn_br_points,XrL4_all_clinvar_mis_not_in_gn_br_points,nrow=1)
#stats for this:


XrL4_cv_mis_not_gn.br_LOF <- dim(XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br[XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br$XrL4_mmfunc_class == 'LOF' ,])
XrL4_cv_mis_not_gn.br_total <- dim(XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br[XrL4_all_final_sns_norm_filt_CADD_cv_not_gn_br$conseq == 'NON_SYNONYMOUS',])

XrL4_cv_mis_in_gn.br_LOF <- dim(XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br[XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br$conseq == 'NON_SYNONYMOUS' & XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br$XrL4_mmfunc_class == 'LOF' ,])
XrL4_cv_mis_in_gn.br_total <- dim(XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br[XrL4_all_final_sns_norm_filt_CADD_cv_in_gn_br$conseq == 'NON_SYNONYMOUS',])


contingency_matrix5 <- cbind(c(XrL4_cv_mis_not_gn.br_LOF,XrL4_cv_mis_not_gn.br_total-XrL4_cv_mis_not_gn.br_LOF),c(XrL4_cv_mis_in_gn.br_LOF,XrL4_cv_mis_in_gn.br_total-XrL4_cv_mis_in_gn.br_LOF))
fisher.test(contingency_matrix5)$p.value



#SxComp  CADD (main text, PhyloP, SIFT, Grantham,)
#notes -- DO PhyloP on ALL, then rest only missense, including CADD again.
#export as 4x4 pdf for SxComp
ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(mamPhyloP))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent'),], aes(color=clinvar),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple != 'absent'),], aes(color=clinvar),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign'),], aes(color=clinvar),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('PhyloP score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")
XrL4_PhyloP_SPEARMAN <- cor(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD$mamPhyloP)),method='spearman')
cor.test(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD$mamPhyloP)),method='spearman')$p.value

## export to Fig SxComp
## ROC CADD:
P.B.LP.LB_CADD_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(CADD.phred)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B.LP.LB_CADD_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_CADD_roc)$AUC, 3)))

# ROC PhyloP
P.B.LP.LB_PhyloP_roc <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(mamPhyloP)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC: #export as 4x4 pdf for SxComp
P.B.LP.LB_PhyloP_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_PhyloP_roc)$AUC, 3)))

#export this as 3x9 and bring in ClinVar legend.
grid.arrange(ggplot(XrL4_all_final_sns_norm_filt_CADD, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(mamPhyloP))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent'),], aes(color=clinvar),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='none')+ylab('PhyloP score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black"),P.B.LP.LB_CADD_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_CADD_roc)$AUC, 3))),P.B.LP.LB_PhyloP_roc + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_PhyloP_roc)$AUC, 3))),nrow=1)

XrL4_all_final_sns_norm_filt_CADD_MIS <- XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',]
XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS <- XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc[XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc$conseq == 'NON_SYNONYMOUS',]
XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc_MIS <- XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS[which(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS$clinvar_simple == 'Pathogenic' | XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS$clinvar_simple == 'Benign'),]

#CADD MIS
edf9d <- ggplot(XrL4_all_final_sns_norm_filt_CADD_MIS, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(CADD.phred))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.6)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.6)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.6)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('CADD score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")
XrL4_CADD_MIS_SPEARMAN <- cor(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$CADD.phred)),method='spearman')
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$CADD.phred)),method='spearman')


edf9e <- ggplot(XrL4_all_final_sns_norm_filt_CADD_MIS, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(mamPhyloP))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.6)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.6)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.6)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('PhyloP score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")

XrL4_PhyloP_MIS_SPEARMAN <- cor(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$mamPhyloP)),method='spearman')
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$mamPhyloP)),method='spearman')

#SIFT
ggplot(XrL4_all_final_sns_norm_filt_CADD_MIS, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(SIFTval))))+geom_jitter(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.6,height=0.03)+geom_jitter(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.6,height=0.03)+geom_jitter(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.6,height=0.03)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SIFT score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")

#alternative plotting of SIFT to show 0 to 0.05 range better:
ggplot(XrL4_all_final_sns_norm_filt_CADD_MIS, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=log10(as.numeric(as.character(SIFTval)))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent' & XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval != 0),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent' & XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval != 0),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign' & XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval != 0),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_jitter(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent' & XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval == 0),], aes(color=clinvar_simple),alpha=1,size=0.5,height=.005)+geom_jitter(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent' & XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval == 0),], aes(color=clinvar_simple),alpha=1,size=0.5,height=.005)+geom_jitter(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign' & XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval == 0),], aes(color=clinvar_simple),alpha=1,size=0.5,height=.005)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SIFT score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")

XrL4_PhyloP_MIS_SIFT <- cor(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval)),method='spearman')
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval)),method='spearman')

ggplot(XrL4_all_final_sns_norm_filt_CADD_MIS, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(polyphen.val))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('PolyPhen-2 score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")

#Grantham deviation only
ggplot(XrL4_all_final_sns_norm_filt_CADD_MIS, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(Grantham))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Grantham deviation score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")

#plot vs. aGVGD GV-GD
edf9f <- ggplot(XrL4_all_final_sns_norm_filt_CADD_MIS, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(GV.GD))))+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('aGVGD (GV-GD)')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")

#plot vs. aGVGD class
edf9g <- ggplot(XrL4_all_final_sns_norm_filt_CADD_MIS, aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(aGVGD_class))))+geom_jitter(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'absent'),], aes(color=clinvar_simple),alpha=0.5,size=0.5,height=8)+geom_jitter(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple != 'absent'),], aes(color=clinvar_simple),alpha=1,size=0.5,height=8)+geom_point(data=XrL4_all_final_sns_norm_filt_CADD_MIS[which(XrL4_all_final_sns_norm_filt_CADD_MIS$clinvar_simple == 'Benign'),], aes(color=clinvar_simple),alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('aGVGD Class')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="Gray") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")+scale_y_continuous(breaks=c(0,15,25,35,45,55,65),labels = c('C0','C15','C25','C35','C45','C55','C65'))

#export as 3 x 12 and resize in illustrator, add clinvar legend
grid.arrange(edf9d,edf9e,edf9f,edf9g,nrow=1)

XrL4_CADD_SPEARMAN <- cor(XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD$CADD.phred)),method='spearman')

cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$CADD.phred)),method='spearman')$p.value
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$mamPhyloP)),method='spearman')$p.value
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$SIFTval)),method='spearman')$p.value
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$polyphen.val)),method='spearman')$p.value
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$Grantham)),method='spearman')$p.value
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$GV.GD)),method='spearman')$p.value
cor.test(XrL4_all_final_sns_norm_filt_CADD_MIS$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_MIS$aGVGD_class)),method='spearman')$p.value

#debug this to show the RNA scores by aGVGD class, by functional class
#or plot only missense and use the color or fill / shading to show the RNA.
ggplot(XrL4_all_final_sns_norm_filt_EXON_no18[which(XrL4_all_final_sns_norm_filt_EXON_no18$aGVGD_class != 'NA'),], aes(fill=XrL4_mmfunc_class))+geom_boxplot(aes(x=as.factor(aGVGD_class), y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Functional category')+ylab('RNA expression')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=c("FUNC" = "#5B81D9","INT" = "#8D9486","LOF" = "#292A2C"),labels = c('Functional','Intermediate','Non-functional'))+scale_x_discrete(breaks=c(0,15,25,35,45,55,65))
# 1380 for ROCs vs. aGVGD
# 1566 for BRAVO and FLOSSIES

#ROC curves for missense only, including LP/LB:

#SGE data
P.B.LP.LB_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = -tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B.LP.LB_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_roc_MIS)$AUC, 3)))
P.B.LP.LB_roc_MIS_anno <- P.B.LP.LB_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_roc_MIS)$AUC, 3)))

#CADD
P.B.LP.LB_CADD_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(CADD.phred)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B.LP.LB_CADD_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_CADD_roc_MIS)$AUC, 3)))
P.B.LP.LB_CADD_roc_MIS_anno <- P.B.LP.LB_CADD_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_CADD_roc_MIS)$AUC, 3)))

#PhyloP
P.B.LP.LB_PhyloP_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(mamPhyloP)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B.LP.LB_PhyloP_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_PhyloP_roc_MIS)$AUC, 3)))
P.B.LP.LB_PhyloP_roc_MIS_anno <- P.B.LP.LB_PhyloP_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_PhyloP_roc_MIS)$AUC, 3)))

#GV.GD
P.B.LP.LB_GV.GD_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = -as.numeric(as.character(GV.GD)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B.LP.LB_GV.GD_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_GV.GD_roc_MIS)$AUC, 3)))
P.B.LP.LB_GV.GD_roc_MIS_anno <- P.B.LP.LB_GV.GD_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_GV.GD_roc_MIS)$AUC, 3)))

#GV.GD
P.B.LP.LB_aGVGD_class_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B.LP.LB_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(aGVGD_class)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B.LP.LB_aGVGD_class_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_aGVGD_class_roc_MIS)$AUC, 3)))
P.B.LP.LB_aGVGD_class_roc_MIS_anno <- P.B.LP.LB_aGVGD_class_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B.LP.LB_aGVGD_class_roc_MIS)$AUC, 3)))

XrL4_P.B.LP.LB_MIS_roc_curve_comparison <- grid.arrange(P.B.LP.LB_roc_MIS_anno,P.B.LP.LB_CADD_roc_MIS_anno,P.B.LP.LB_PhyloP_roc_MIS_anno,P.B.LP.LB_GV.GD_roc_MIS_anno,P.B.LP.LB_aGVGD_class_roc_MIS_anno,nrow=1)

#ROC curves for missense only NOT including LP/LB:
#SGE data
P.B_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = -tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm)) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_roc_MIS)$AUC, 3)))
P.B_roc_MIS_anno <- P.B_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_roc_MIS)$AUC, 3)))

#CADD
P.B_CADD_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(CADD.phred)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B_CADD_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_CADD_roc_MIS)$AUC, 3)))
P.B_CADD_roc_MIS_anno <- P.B_CADD_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_CADD_roc_MIS)$AUC, 3)))

#PhyloP
P.B_PhyloP_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(mamPhyloP)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B_PhyloP_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_PhyloP_roc_MIS)$AUC, 3)))
P.B_PhyloP_roc_MIS_anno <- P.B_PhyloP_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_PhyloP_roc_MIS)$AUC, 3)))

#GV.GD
P.B_GV.GD_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = -as.numeric(as.character(GV.GD)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B_GV.GD_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_GV.GD_roc_MIS)$AUC, 3)))
P.B_GV.GD_roc_MIS_anno <- P.B_GV.GD_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_GV.GD_roc_MIS)$AUC, 3)))

#GV.GD
P.B_aGVGD_class_roc_MIS <- ggplot(XrL4_all_final_sns_norm_filt_CADD_P.B_geom_roc_MIS, aes(d = D_P.B.LP.LB, m = as.numeric(as.character(aGVGD_class)))) +geom_roc(n.cuts=0,labels=FALSE) + style_roc(xlab = "1 - Specificity",major.breaks=c(0.0,0.1,0.5,0.9,1.0))
#with AUC:  #export as 4x4 pdf for SxComp
P.B_aGVGD_class_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_aGVGD_class_roc_MIS)$AUC, 3)))
P.B_aGVGD_class_roc_MIS_anno <- P.B_aGVGD_class_roc_MIS + annotate("text", x = .75, y = .25, label = paste("AUC =", round(calc_auc(P.B_aGVGD_class_roc_MIS)$AUC, 3)))

#export this as a 20x4 PDF 
XrL4_P.B_MIS_roc_curve_comparison <- grid.arrange(P.B_roc_MIS_anno,P.B_CADD_roc_MIS_anno,P.B_PhyloP_roc_MIS_anno,P.B_GV.GD_roc_MIS_anno,P.B_aGVGD_class_roc_MIS_anno,nrow=1)

#SxClinVar  how many points pass filters vs. don't? get same variants out?
#Lig4.KO data -- plotting benign on top export as 4x4 for SxCLinVar
XrL4_9_exons_P.LP.B.LB_R1_R2_gg<- ggplot(XrL4_all_final_sns_norm_filt_CADD[which((XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Pathogenic" | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Likely pathogenic" | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Benign" | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Likely benign" )&(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X2' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X3' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X5' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X16' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X17' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X19' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X23')),], aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, color=clinvar_simple,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(alpha=1)+geom_point(data = XrL4_all_final_sns_norm_filt_CADD[which(( XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Benign"  )&(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X2' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X3' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X5' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X16' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X17' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X19' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X23')),])+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function R2')+xlab('Function R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)
#with labels -- to ID points
XrL4_9_exons_P.LP.B.LB_R1_R2_gg+geom_text(aes(label=paste(exon,pos)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black',alpha=0.25)

#export Xr for SxClinVar b: 4x4 pdf
ggplot(Xr_all_final_sns_norm_filt_CADD[which((Xr_all_final_sns_norm_filt_CADD$clinvar_simple == "Pathogenic" | Xr_all_final_sns_norm_filt_CADD$clinvar_simple == "Likely pathogenic" | Xr_all_final_sns_norm_filt_CADD$clinvar_simple == "Benign" | Xr_all_final_sns_norm_filt_CADD$clinvar_simple == "Likely benign" )&(Xr_all_final_sns_norm_filt_CADD$exon == 'X4' | Xr_all_final_sns_norm_filt_CADD$exon == 'X2' | Xr_all_final_sns_norm_filt_CADD$exon == 'X3' | Xr_all_final_sns_norm_filt_CADD$exon == 'X5' | Xr_all_final_sns_norm_filt_CADD$exon == 'X16' | Xr_all_final_sns_norm_filt_CADD$exon == 'X17' | Xr_all_final_sns_norm_filt_CADD$exon == 'X19' | Xr_all_final_sns_norm_filt_CADD$exon == 'X21' | Xr_all_final_sns_norm_filt_CADD$exon == 'X23')),], aes(x=tHDR_post_lib_loess_e.sns_r1_sns_norm, color=clinvar_simple,y=tHDR_post_lib_loess_e.sns_r2_sns_norm))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function R2')+xlab('Function R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#scratch with labels to ID points#
grid.arrange(ggplot(XrL4_all_final_sns_norm_filt_CADD[which((XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Pathogenic" | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Likely pathogenic" | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Benign" | XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == "Likely benign" )&(XrL4_all_final_sns_norm_filt_CADD$exon == 'X4' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X2' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X3' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X5' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X16' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X17' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X19' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' | XrL4_all_final_sns_norm_filt_CADD$exon == 'X23')),], aes(x=tHDR_post_lib_loess_e.sns_rL41_sns_norm, color=clinvar_simple,y=tHDR_post_lib_loess_e.sns_rL42_sns_norm))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function R2')+xlab('Function R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=paste(exon,pos)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black',alpha=0.25),ggplot(Xr_all_final_sns_norm_filt_CADD[which((Xr_all_final_sns_norm_filt_CADD$clinvar_simple == "Pathogenic" | Xr_all_final_sns_norm_filt_CADD$clinvar_simple == "Likely pathogenic" | Xr_all_final_sns_norm_filt_CADD$clinvar_simple == "Benign" | Xr_all_final_sns_norm_filt_CADD$clinvar_simple == "Likely benign" )&(Xr_all_final_sns_norm_filt_CADD$exon == 'X4' | Xr_all_final_sns_norm_filt_CADD$exon == 'X2' | Xr_all_final_sns_norm_filt_CADD$exon == 'X3' | Xr_all_final_sns_norm_filt_CADD$exon == 'X5' | Xr_all_final_sns_norm_filt_CADD$exon == 'X16' | Xr_all_final_sns_norm_filt_CADD$exon == 'X17' | Xr_all_final_sns_norm_filt_CADD$exon == 'X19' | Xr_all_final_sns_norm_filt_CADD$exon == 'X21' | Xr_all_final_sns_norm_filt_CADD$exon == 'X23')),], aes(x=tHDR_post_lib_loess_e.sns_r1_sns_norm, color=clinvar_simple,y=tHDR_post_lib_loess_e.sns_r2_sns_norm))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function R2')+xlab('Function R1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=paste(exon,pos)),hjust=0.5, vjust=0.5, size = 1.4,colour='Black',alpha=0.25),nrow=1)

#SxClinVar X21:
X21_CDS_min <- min(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X21_CDS_min_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X21_CDS_min),]$pos[1]))
X21_pos_min <- min(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21'),]$pos)))
X21_CDS_max <- max(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos != 'NA'),]$clinvar_CDSpos)
X21_CDS_max_pos <- as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21' & XrL4_all_final_sns_norm_filt_CADD$clinvar_CDSpos == X21_CDS_max),]$pos[1]))
X21_pos_max <- max(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21'),]$pos)))
#can further engineer this to be a vector of breaks and labels,equal in length, and labels = '', for every 10th base.
#Fig 4a
ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21'),], aes(y=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,x=as.numeric(as.character(pos))+as.numeric(as.character(pos_nudge))))+geom_segment(aes(yend=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,xend=as.numeric(as.character(pos))+as.numeric(as.character(pos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White') + geom_hline(yintercept = XrL4_all_lof_thresh, lty=2)+geom_hline(yintercept = XrL4_all_f01_thresh , lty=2)+scale_x_reverse(breaks=X21_genomic_breaks,labels=X21_genomic_labels)
#expression -- export as 3 x 6.34 to match proportion of Function scores, but not needed as figure
ggplot(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$exon == 'X21'),], aes(y=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),x=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge))))+geom_segment(aes(yend=log2(tHDR_rna_pre_ratio_rL41rL42_mean_synnorm),xend=as.numeric(as.character(clinvar_CDSpos))+as.numeric(as.character(CDSpos_nudge)),colour=conseq),yend=0)+geom_point(aes(colour=conseq),alpha=1,size=1.3)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=rev_comp),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim = c(-3,0.5))

#Function vs. CADD, canonical splice only:
ggplot(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE',], aes(x=tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,y=as.numeric(as.character(CADD.phred))))+geom_point(aes(color=clinvar_simple),alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('CADD score')+xlab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_full_simple) + geom_vline(xintercept=XrL4_all_f01_thresh,lty=2,color="black") + geom_vline(xintercept=XrL4_all_lof_thresh,lty=2,color="black")

#FINAL number calculations:
XrL4_total_SNVs_assayed <- dim(XrL4_all_final_sns_norm_filt_CADD)[1]
XrL4_total_SNVs_assayed
XrL4_final_CADD_median_syn <- median(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
XrL4_final_CADD_median_syn
XrL4_final_CADD_median_stop <- median(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'STOP_GAINED'),c('tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm')])
XrL4_final_CADD_median_stop
XrL4_stop_gained_assayed <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'STOP_GAINED'),])[1]
XrL4_stop_gained_assayed
XrL4_stop_gained_below_n1.25 <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < -1.25),])[1]
XrL4_syn_assayed <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),])[1]
XrL4_syn_assayed
XrL4_syn_below_n1.25 <- dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS' & XrL4_all_final_sns_norm_filt_CADD$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm < -1.25),])[1]
#fraction syn below -1.25:  
XrL4_syn_below_n1.25/XrL4_syn_assayed
#syn SNVs within 0.57 (log2) i.e. 1 SD of median
dim(XrL4_all_final_sns_norm_syn_norm_exp)

#How many ClinVar SNVs in the region?
XrL4_all_prog_map_all_df <- rbind(X2rL41_prog_map_all_df,X3rL41_prog_map_all_df,X4rL41_prog_map_all_df,X5rL41_prog_map_all_df,X15rL41_prog_map_all_df,X16rL41_prog_map_all_df,X17rL41_prog_map_all_df,X18rL41_prog_map_all_df,X19rL41_prog_map_all_df,X20rL41_prog_map_all_df,X21rL41_prog_map_all_df,X22rL41_prog_map_all_df,X23rL41_prog_map_all_df)
dim(XrL4_all_prog_map_all_df[XrL4_all_prog_map_all_df$clinvar_simple != 'absent',])[1]
dim(XrL4_all_prog_map_all_df[XrL4_all_prog_map_all_df$clinvar_simple == 'Uncertain significance',])[1]
dim(XrL4_all_prog_map_all_df[XrL4_all_prog_map_all_df$clinvar_simple == 'Conflicting interpretations of pathogenicity',])[1]

#proportion non-func (same)
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD)[1]
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$posterior_ns > 0.99),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD)[1]
#proportion func (same)
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD)[1]
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$posterior_ns < 0.01),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD)[1]
#proportion int
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD)[1]

#missense LOF
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',])[1]
#missense FUNC
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',])[1]
#missense INT
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS',])[1]
#not missense INT
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq != 'NON_SYNONYMOUS'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq != 'NON_SYNONYMOUS',])[1]
contingency_matrix1 <- cbind(c(dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq != 'NON_SYNONYMOUS'),])[1],dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class != 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq != 'NON_SYNONYMOUS'),])[1]),c(dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),])[1],dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class != 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),])[1]))
fisher.test(contingency_matrix1)$p.value

#canonical splice:
#CS LOF
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE',])[1]
#CS FUNC
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE',])[1]
#CS INT
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'CANONICAL_SPLICE',])[1]

#splice region:
#SR LOF
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE',])[1]
#SR FUNC
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE',])[1]
#SR INT
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE',])[1]

#INTRONIC:
# LOF
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC',])[1]
# FUNC
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC',])[1]
# INT
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'INTRONIC',])[1]

#5PRIME_UTR:
# LOF
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR',])[1]
# FUNC
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR',])[1]
# INT
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == '5PRIME_UTR',])[1]

#SYN:
# LOF
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS',])[1]
# FUNC
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS',])[1]
# INT
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS',])[1]

#clinvar counts:
XrL4_total_PATH <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Pathogenic'))
XrL4_total_PATH_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'Pathogenic'))
XrL4_total_PATH_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'Pathogenic'))
XrL4_total_PATH_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'Pathogenic'))
XrL4_total_PATH_FUNC/XrL4_total_PATH
XrL4_total_PATH_LOF/XrL4_total_PATH
XrL4_total_PATH_INT/XrL4_total_PATH

XrL4_total_LP <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Likely pathogenic'))
XrL4_total_LP_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'Likely pathogenic'))
XrL4_total_LP_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'Likely pathogenic'))
XrL4_total_LP_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'Likely pathogenic'))
XrL4_total_LP_FUNC/XrL4_total_LP
XrL4_total_LP_LOF/XrL4_total_LP
XrL4_total_LP_INT/XrL4_total_LP

XrL4_total_BEN <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Benign'))
XrL4_total_BEN_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'Benign'))
XrL4_total_BEN_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'Benign'))
XrL4_total_BEN_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'Benign'))
XrL4_total_BEN_FUNC/XrL4_total_BEN
XrL4_total_BEN_LOF/XrL4_total_BEN
XrL4_total_BEN_INT/XrL4_total_BEN

XrL4_total_LB <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Likely benign'))
XrL4_total_LB_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'Likely benign'))
XrL4_total_LB_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'Likely benign'))
XrL4_total_LB_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'Likely benign'))
XrL4_total_LB_FUNC/XrL4_total_LB
XrL4_total_LB_LOF/XrL4_total_LB
XrL4_total_LB_INT/XrL4_total_LB


#how many VUS or Conflicting are re-assigned to classes vs. not:
XrL4_total_VUS <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Uncertain significance'))
XrL4_total_VUS_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'Uncertain significance'))
XrL4_total_VUS_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'Uncertain significance'))
XrL4_total_VUS_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'Uncertain significance'))
XrL4_total_VUS_FUNC/XrL4_total_VUS
XrL4_total_VUS_LOF/XrL4_total_VUS
XrL4_total_VUS_INT/XrL4_total_VUS

XrL4_total_CI <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Conflicting interpretations of pathogenicity'))
XrL4_total_CI_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'Conflicting interpretations of pathogenicity'))
XrL4_total_CI_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'Conflicting interpretations of pathogenicity'))
XrL4_total_CI_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'Conflicting interpretations of pathogenicity'))
XrL4_total_CI_FUNC/XrL4_total_CI
XrL4_total_CI_LOF/XrL4_total_CI
XrL4_total_CI_INT/XrL4_total_CI

#MISSENSE VUS
XrL4_total_VUS_MIS <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Uncertain significance' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'))
XrL4_total_VUS_MIS_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Uncertain significance' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS' ),]$XrL4_mmfunc_class == 'FUNC'))
XrL4_total_VUS_MIS_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Uncertain significance' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS' ),]$XrL4_mmfunc_class == 'LOF'))
XrL4_total_VUS_MIS_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'Uncertain significance' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS' ),]$XrL4_mmfunc_class == 'INT'))
XrL4_total_VUS_MIS_FUNC/XrL4_total_VUS_MIS
XrL4_total_VUS_MIS_LOF/XrL4_total_VUS_MIS
XrL4_total_VUS_MIS_INT/XrL4_total_VUS_MIS

XrL4_total_ABS <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent'))
XrL4_total_ABS_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'FUNC'),]$clinvar_simple == 'absent'))
XrL4_total_ABS_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF'),]$clinvar_simple == 'absent'))
XrL4_total_ABS_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'INT'),]$clinvar_simple == 'absent'))
XrL4_total_ABS_FUNC/XrL4_total_ABS
XrL4_total_ABS_LOF/XrL4_total_ABS
XrL4_total_ABS_INT/XrL4_total_ABS

#MISSENSE ABSENT:
#MISSENSE ABS
XrL4_total_ABS_MIS <- length(which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'))
XrL4_total_ABS_MIS_FUNC <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS' ),]$XrL4_mmfunc_class == 'FUNC'))
XrL4_total_ABS_MIS_LOF <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS' ),]$XrL4_mmfunc_class == 'LOF'))
XrL4_total_ABS_MIS_INT <- length(which(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$clinvar_simple == 'absent' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS' ),]$XrL4_mmfunc_class == 'INT'))
XrL4_total_ABS_MIS_FUNC/XrL4_total_ABS_MIS
XrL4_total_ABS_MIS_LOF/XrL4_total_ABS_MIS
XrL4_total_ABS_MIS_INT/XrL4_total_ABS_MIS

contingency_matrix2 <- cbind(c(XrL4_total_ABS_MIS_LOF,XrL4_total_ABS_MIS-XrL4_total_ABS_MIS_LOF),c(XrL4_total_VUS_MIS_LOF,XrL4_total_VUS_MIS-XrL4_total_VUS_MIS_LOF))
fisher.test(contingency_matrix2)$p.value

#XrL4 final variants in gnomad
length(which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad))

#wilcoxon rank sum test on gnomad alleles
cor.test(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad),]$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm,as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad),]$gnomad_AF)),method='spearman')

wilcox.test(x = XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad),]$tHDR_post_lib_loess_e.sns_rL41rL42_sns_norm, y = as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad),]$gnomad_AF)),paired = TRUE)

#make a gnomad DB:
XrL4_all_final_sns_norm_filt_CADD_gnomad <- XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$gnomad_AF != 'NA'),]
XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF <- XrL4_all_final_sns_norm_filt_CADD_gnomad[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_gnomad$gnomad_count)) == 1),]
XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF <- XrL4_all_final_sns_norm_filt_CADD_gnomad[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_gnomad$gnomad_count)) > 1),]

dim(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF$XrL4_mmfunc_class != 'LOF'),])[1]
dim(XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF$XrL4_mmfunc_class != 'LOF'),])[1]
dim(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF$XrL4_mmfunc_class == 'LOF'),])[1]
dim(XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF$XrL4_mmfunc_class == 'LOF'),])[1]

contingency_matrix3 <- cbind(c(dim(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF$XrL4_mmfunc_class == 'LOF'),])[1],dim(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_notminAF$XrL4_mmfunc_class != 'LOF'),])[1]),c(dim(XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF$XrL4_mmfunc_class == 'LOF'),])[1],dim(XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF[which(XrL4_all_final_sns_norm_filt_CADD_gnomad_minAF$XrL4_mmfunc_class != 'LOF'),])[1]))
fisher.test(contingency_matrix3)$p.value

#make a bravo DB:
XrL4_all_final_sns_norm_filt_CADD_bravo <- XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$bravo_AF != 'NA'),]
XrL4_all_final_sns_norm_filt_CADD_bravo_minAF <- XrL4_all_final_sns_norm_filt_CADD_bravo[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_bravo$bravo_count)) == 1),]
XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF <- XrL4_all_final_sns_norm_filt_CADD_bravo[which(as.numeric(as.character(XrL4_all_final_sns_norm_filt_CADD_bravo$bravo_count)) > 1),]

dim(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF$XrL4_mmfunc_class != 'LOF'),])[1]
dim(XrL4_all_final_sns_norm_filt_CADD_bravo_minAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_minAF$XrL4_mmfunc_class != 'LOF'),])[1]
dim(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF$XrL4_mmfunc_class == 'LOF'),])[1]
dim(XrL4_all_final_sns_norm_filt_CADD_bravo_minAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_minAF$XrL4_mmfunc_class == 'LOF'),])[1]

contingency_matrix4 <- cbind(c(dim(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF$XrL4_mmfunc_class == 'LOF'),])[1],dim(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_notminAF$XrL4_mmfunc_class != 'LOF'),])[1]),c(dim(XrL4_all_final_sns_norm_filt_CADD_bravo_minAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_minAF$XrL4_mmfunc_class == 'LOF'),])[1],dim(XrL4_all_final_sns_norm_filt_CADD_bravo_minAF[which(XrL4_all_final_sns_norm_filt_CADD_bravo_minAF$XrL4_mmfunc_class != 'LOF'),])[1]))
fisher.test(contingency_matrix4)$p.value

#how many variants unreported in any of these databases:
dim(XrL4_all_final_sns_norm_filt_CADD)[1]-dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_clinvar | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_bravo | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_gnomad | XrL4_all_final_sns_norm_filt_CADD$pos_alt %in% variants_in_flossies),])[1]

#Proportion of missense with low RNA:
dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS' & log2(XrL4_all_final_sns_norm_filt_CADD$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm) < -2 ),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'NON_SYNONYMOUS'),])[1]

dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS' & log2(XrL4_all_final_sns_norm_filt_CADD$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm) < -2 ),])[1]/dim(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),])[1]

#median SYN LOF RNA
median(log2(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS'),]$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm))

median(log2(XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & (XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS' | (XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE' & XrL4_all_final_sns_norm_filt_CADD$clinvar_cDNApos != 'NA'))),]$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm))
#clinvar_status on these?
XrL4_all_final_sns_norm_filt_CADD[which(XrL4_all_final_sns_norm_filt_CADD$XrL4_mmfunc_class == 'LOF' & (XrL4_all_final_sns_norm_filt_CADD$conseq == 'SYNONYMOUS' | (XrL4_all_final_sns_norm_filt_CADD$conseq == 'SPLICE_SITE' & XrL4_all_final_sns_norm_filt_CADD$clinvar_cDNApos != 'NA'))),c('clinvar_simple','tHDR_rna_pre_ratio_rL41rL42_mean_synnorm')]

#exon counts for low RNA SNVs:
#exon 16 
dim(XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$conseq != 'STOP_GAINED'),])

dim(XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$conseq != 'STOP_GAINED'),])

dim(XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$conseq != 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.50),])

dim(XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$exon == 'X16' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$conseq != 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.25),])

#exon 19 
dim(XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$exon == 'X19' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$conseq != 'STOP_GAINED'),])

dim(XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$exon == 'X19' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$conseq != 'STOP_GAINED'),])

dim(XrL4_all_final_sns_norm_filt_CADD_EXON_no18[which(XrL4_all_final_sns_norm_filt_CADD_EXON_no18$exon == 'X19' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$XrL4_mmfunc_class == 'LOF' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$conseq != 'STOP_GAINED' & XrL4_all_final_sns_norm_filt_CADD_EXON_no18$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm < 0.50),])


```


